diff -wbBur synfig-0.62.01/src/modules/mod_png/mptr_png.cpp synfig-0.62.01.my/src/modules/mod_png/mptr_png.cpp
--- synfig-0.62.01/src/modules/mod_png/mptr_png.cpp	2010-05-31 18:12:14.000000000 +0000
+++ synfig-0.62.01.my/src/modules/mod_png/mptr_png.cpp	2010-06-03 15:43:39.000000000 +0000
@@ -309,7 +309,7 @@
 				float b=gamma().b_U8_to_F32((unsigned char)png_ptr->palette[row_pointers[y][x]].blue);
 				float a=1.0;
 				if(info_ptr->valid & PNG_INFO_tRNS)
-				    a = (float)(unsigned char)png_ptr->trans[row_pointers[y][x]]*(1.0/255.0);
+				    a = (float)(unsigned char)png_ptr->trans_alpha[row_pointers[y][x]]*(1.0/255.0);
 				surface_buffer[y][x]=Color(
 					r,
 					g,
