--- ./src/explorer.c	2006-06-22 01:02:02.000000000 -0500
+++ ./src/explorer.c.new	2008-07-09 07:29:34.000000000 -0500
@@ -381,6 +381,7 @@
     gchar *filename = NULL;
 
 #if (GTK_CHECK_VERSION(2, 4, 0))
+    explorer_force_pause(self);
     dialog = gtk_file_chooser_dialog_new ("Open Image Parameters",
 		                          GTK_WINDOW (glade_xml_get_widget (self->xml, "explorer_window")),
 					  GTK_FILE_CHOOSER_ACTION_OPEN,
@@ -402,8 +403,9 @@
 	    g_free (file_location);
 	file_location = gtk_file_chooser_get_current_folder (GTK_FILE_CHOOSER (dialog));
     }
+    explorer_restore_pause(self);
 #else
-    dialog = gtk_file_selection_new ("Open Image Parameters");
+    dialog = gtk_file_selection_new ("Open zImage Parameters");
 
     if (gtk_dialog_run (GTK_DIALOG (dialog)) == GTK_RESPONSE_OK) {
 	filename = gtk_file_selection_get_filename (GTK_FILE_SELECTION (dialog));
@@ -436,6 +438,7 @@
     gchar *filename = NULL;
 
 #if (GTK_CHECK_VERSION(2, 4, 0))
+    explorer_force_pause(self);
     dialog = gtk_file_chooser_dialog_new ("Save Image",
 		                          GTK_WINDOW (glade_xml_get_widget (self->xml, "explorer_window")),
 					  GTK_FILE_CHOOSER_ACTION_SAVE,
@@ -453,6 +456,7 @@
             g_free (file_location);
 	file_location = gtk_file_chooser_get_current_folder (GTK_FILE_CHOOSER (dialog));
     }
+    explorer_restore_pause(self);
 #else
     dialog = gtk_file_selection_new ("Save Image");
     gtk_file_selection_set_filename (GTK_FILE_SELECTION (dialog), "rendering.png");
@@ -491,6 +495,7 @@
     gchar *filename = NULL;
 
 #if (GTK_CHECK_VERSION(2, 4, 0))
+    explorer_force_pause(self);
     dialog = gtk_file_chooser_dialog_new ("Save OpenEXR Image",
 		                          GTK_WINDOW (glade_xml_get_widget (self->xml, "explorer_window")),
 					  GTK_FILE_CHOOSER_ACTION_SAVE,
@@ -509,6 +514,7 @@
 	    g_free (file_location);
 	file_location = gtk_file_chooser_get_current_folder (GTK_FILE_CHOOSER (dialog));
     }
+    explorer_restore_pause(self);
 #else
     dialog = gtk_file_selection_new ("Save OpenEXR Image");
     gtk_file_selection_set_filename (GTK_FILE_SELECTION (dialog), "rendering.exr");
