--- Makefile.am.orig	2009-11-20 19:21:08.000000000 +0100
+++ Makefile.am	2009-11-20 19:20:29.000000000 +0100
@@ -46,13 +46,13 @@
 	voacapl/itshfbc/run/voacapx.out
 
 install-exec-hook:
-	sed  -i "s|__PREFIX__|@prefix@|" $(DEST)/$(bindir)/makeitshfbc
+	sed  -i "s|__PREFIX__|@prefix@|" $(DESTDIR)/$(bindir)/makeitshfbc
 
 install-data-hook:
-	coeffbin $(datadir)/voacapl/itshfbc/coeffs &&\
-	coefbinw $(datadir)/voacapl/itshfbc/coeffs &&\
-	fof2bin $(datadir)/voacapl/itshfbc/coeffs &&\
-	fof2binw $(datadir)/voacapl/itshfbc/coeffs &&\
+	$(DESTDIR)/$(bindir)/coeffbin $(DESTDIR)/$(datadir)/voacapl/itshfbc/coeffs &&\
+	$(DESTDIR)/$(bindir)/coefbinw $(DESTDIR)/$(datadir)/voacapl/itshfbc/coeffs &&\
+	$(DESTDIR)/$(bindir)/fof2bin $(DESTDIR)/$(datadir)/voacapl/itshfbc/coeffs &&\
+	$(DESTDIR)/$(bindir)/fof2binw $(DESTDIR)/$(datadir)/voacapl/itshfbc/coeffs &&\
 	echo "" &&\
 	echo "" &&\
 	echo "***************************************************" &&\
