--- debug.cc.orig	2007-02-19 21:46:01.000000000 +0100
+++ debug.cc	2007-02-19 21:46:12.000000000 +0100
@@ -20,7 +20,7 @@
 #if (__GNUC__ == 3 && __GNUC_MINOR__ < 4)
 #include <bits/stl_alloc.h>
 #else
-extern "C" char* getenv(char const* name);	// Needed before including ext/pool_allocator.h.
+extern "C" char* getenv(char const* name) throw ();	// Needed before including ext/pool_allocator.h.
 #include <ext/pool_allocator.h>
 #endif
 #undef private
