From: hasufell <julian.ospald@googlemail.com>
Date: Sun Apr  8 18:48:34 UTC 2012

add proper install rules

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -32,6 +32,14 @@
 option(USE_MPG123 "use mpg123 + libsndfile for sound" ON)
 
 find_program(DPKG_PROGRAM dpkg DOC "dpkg program of Debian-based systems")
+
+# paths
+set(BINDIR "bin" CACHE PATH "Where to install binaries")
+set(DATAROOTDIR "${CMAKE_INSTALL_PREFIX}/share" CACHE PATH "Sets the root of data directories to a non-default location")
+set(DATADIR "${CMAKE_INSTALL_PREFIX}/share/openmw" CACHE PATH "Sets the openmw data directories to a non-default location")
+set(DOCDIR "${DATAROOTDIR}/doc/openmw" CACHE PATH "Sets the doc directory to a non-default location.")
+set(MANDIR "${DATAROOTDIR}/man" CACHE PATH "Where to install manpages")
+set(SYSCONFDIR "" CACHE PATH "Set config dir")
 
 # Location of morrowind data files
 if(DPKG_PROGRAM)
@@ -42,8 +50,8 @@
         set(MORROWIND_DATA_FILES "./data" CACHE PATH "location of Morrowind data files")
         set(MORROWIND_RESOURCE_FILES "./resources" CACHE PATH "location of OpenMW resources files")
     else()
-        set(MORROWIND_DATA_FILES "data" CACHE PATH "location of Morrowind data files")
-        set(MORROWIND_RESOURCE_FILES "resources" CACHE PATH "location of OpenMW resources files")
+        set(MORROWIND_DATA_FILES "/usr/share/games/openmw/data/" CACHE PATH "location of Morrowind data files")
+        set(MORROWIND_RESOURCE_FILES "${DATADIR}/resources" CACHE PATH "location of OpenMW resources files")
     endif(APPLE)
 endif(DPKG_PROGRAM)
 
@@ -574,3 +582,23 @@
 set(CMAKE_CXX_FLAGS "-arch i386")
 
 endif (APPLE)
+
+
+## Linux building
+
+# Install binaries
+INSTALL(PROGRAMS "${OpenMW_BINARY_DIR}/openmw" DESTINATION ${BINDIR} )
+INSTALL(PROGRAMS "${OpenMW_BINARY_DIR}/omwlauncher" DESTINATION ${BINDIR} )
+INSTALL(PROGRAMS "${OpenMW_BINARY_DIR}/esmtool" DESTINATION ${BINDIR} )
+
+# Install icon
+INSTALL(FILES "${OpenMW_SOURCE_DIR}/apps/launcher/resources/images/openmw.png" DESTINATION "share/pixmaps/")
+
+# Install global configuration files
+INSTALL(FILES "${OpenMW_BINARY_DIR}/openmw.cfg.install" DESTINATION ${SYSCONFDIR} RENAME "openmw.cfg" )
+INSTALL(FILES "${OpenMW_BINARY_DIR}/plugins.cfg" DESTINATION ${SYSCONFDIR} )
+INSTALL(FILES "${OpenMW_SOURCE_DIR}/files/launcher.cfg" DESTINATION ${SYSCONFDIR} )
+
+# Install resources
+INSTALL(DIRECTORY "${OpenMW_BINARY_DIR}/resources" DESTINATION ${DATADIR} )
+INSTALL(FILES "${OpenMW_BINARY_DIR}/launcher.qss" DESTINATION "${DATADIR}/resources" )
