--- luvit-0.2.0/src/lconstants.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/lconstants.h	2012-03-29 16:46:40.000000000 +0200
@@ -37,8 +37,8 @@
 # include <openssl/ssl.h>
 #endif
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
 #include "utils.h"
 
 LUALIB_API int luaopen_constants(lua_State *L);
--- luvit-0.2.0/src/lenv.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/lenv.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,8 +18,8 @@
 #ifndef LENV
 #define LENV
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
 #include "utils.h"
 
 LUALIB_API int luaopen_env (lua_State *L);
--- luvit-0.2.0/src/lhttp_parser.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/lhttp_parser.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,8 +18,8 @@
 #ifndef LHTTP_PARSER
 #define LHTTP_PARSER
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
 #include "utils.h"
 
 LUALIB_API int luaopen_http_parser (lua_State *L);
--- luvit-0.2.0/src/los.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/los.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,8 +18,8 @@
 #ifndef LOS
 #define LOS
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
 
 LUALIB_API int luaopen_os_binding (lua_State *L);
 
--- luvit-0.2.0/src/luv.c	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv.c	2012-03-29 16:46:40.000000000 +0200
@@ -16,7 +16,7 @@
  */
 
 #include "luv.h"
-#include "uv.h"
+#include <uv.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <string.h>
--- luvit-0.2.0/src/luv_dns.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_dns.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,10 +18,10 @@
 #ifndef LUV_DNS
 #define LUV_DNS
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
-#include "ares.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
+#include <ares.h>
 #include "utils.h"
 
 /* Wrapped functions exposed to lua */
--- luvit-0.2.0/src/luv_fs.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_fs.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_FS
 #define LUV_FS
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 
 void luv_push_stats_table(lua_State* L, struct stat* s);
--- luvit-0.2.0/src/luv_fs_watcher.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_fs_watcher.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_FS_WATCHER
 #define LUV_FS_WATCHER
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_handle.h"
 
--- luvit-0.2.0/src/luv.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,8 +18,8 @@
 #ifndef LUV
 #define LUV
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
 #include "utils.h"
 
 LUALIB_API int luaopen_uv_native (lua_State *L);
--- luvit-0.2.0/src/luv_handle.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_handle.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_HANDLE
 #define LUV_HANDLE
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 
 typedef struct {
--- luvit-0.2.0/src/luvit_init.c	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luvit_init.c	2012-03-29 16:46:40.000000000 +0200
@@ -20,12 +20,12 @@
 #include <stdlib.h>
 #include <limits.h> /* PATH_MAX */
 
-#include "lua.h"
-#include "lualib.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lualib.h>
+#include <luajit-2.0/lauxlib.h>
 
 #include "luvit.h"
-#include "uv.h"
+#include <uv.h>
 #include "utils.h"
 #include "los.h"
 #include "luv.h"
--- luvit-0.2.0/src/luvit_main.c	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luvit_main.c	2012-03-29 16:46:40.000000000 +0200
@@ -20,14 +20,14 @@
 #include <stdlib.h>
 #include <limits.h> /* PATH_MAX */
 
-#include "lua.h"
-#include "lualib.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lualib.h>
+#include <luajit-2.0/lauxlib.h>
 
 #ifndef WIN32
 #include <pthread.h>
 #endif
-#include "uv.h"
+#include <uv.h>
 
 #include "luvit.h"
 #include "luv.h"
--- luvit-0.2.0/src/luv_misc.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_misc.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_MISC
 #define LUV_MISC
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 
 int luv_activate_signal_handler(lua_State* L);
--- luvit-0.2.0/src/luv_pipe.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_pipe.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_PIPE
 #define LUV_PIPE
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_stream.h"
 
--- luvit-0.2.0/src/luv_process.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_process.h	2012-03-29 16:46:41.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_PROCESS
 #define LUV_PROCESS
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_handle.h"
 
--- luvit-0.2.0/src/luv_stream.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_stream.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_STREAM
 #define LUV_STREAM
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_handle.h"
 
--- luvit-0.2.0/src/luv_tcp.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_tcp.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_TCP
 #define LUV_TCP
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_stream.h"
 
--- luvit-0.2.0/src/luv_timer.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_timer.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_TIMER
 #define LUV_TIMER
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_handle.h"
 
--- luvit-0.2.0/src/luv_tty.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_tty.h	2012-03-29 16:46:41.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_TTY
 #define LUV_TTY
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_stream.h"
 
--- luvit-0.2.0/src/luv_udp.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/luv_udp.h	2012-03-29 16:46:41.000000000 +0200
@@ -18,9 +18,9 @@
 #ifndef LUV_UDP
 #define LUV_UDP
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
 #include "utils.h"
 #include "luv_handle.h"
 
--- luvit-0.2.0/src/lyajl.c	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/lyajl.c	2012-03-29 16:46:40.000000000 +0200
@@ -22,9 +22,9 @@
 
 #include "lyajl.h"
 #include "utils.h"
-#include "yajl/yajl_parse.h"
-#include "yajl/yajl_gen.h"
-#include "yajl/yajl_version.h"
+#include <yajl_parse.h>
+#include <yajl_gen.h>
+#include <yajl_version.h>
 
 static void* yjajl_js_null;
 
--- luvit-0.2.0/src/lyajl.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/lyajl.h	2012-03-29 16:46:41.000000000 +0200
@@ -18,8 +18,8 @@
 #ifndef LYAJL
 #define LYAJL
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
 #include "utils.h"
 
 LUALIB_API int luaopen_yajl (lua_State *L);
--- luvit-0.2.0/src/utils.h	2012-02-25 07:02:11.000000000 +0100
+++ luvit-0.2.0/src/utils.h	2012-03-29 16:46:40.000000000 +0200
@@ -18,10 +18,10 @@
 #ifndef LUV_UTILS
 #define LUV_UTILS
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "uv.h"
-#include "ares.h"
+#include <luajit-2.0/lua.h>
+#include <luajit-2.0/lauxlib.h>
+#include <uv.h>
+#include <ares.h>
 
 /* C doesn't have booleans on it's own */
 #ifndef FALSE
