--- profile-sync-daemon
+++ profile-sync-daemon
@@ -5,7 +5,7 @@
 # Inspired by some code originally  written by Colin Verot
 
 BROWSERS="chromium google-chrome midori mozilla opera opera-next" # all supported browsers
-. /etc/psd.conf
+. /etc/conf.d/psd
 
 set_which() {
 	homedir="$(su -c 'echo $HOME' $user)"
@@ -108,17 +108,17 @@
 		;;
 	check)
 		# do not run if the daemon is running
-		[[ ! -f /run/daemons/psd ]] && check
+		[[ ! -f $PIDFILE ]] && check
 		;;
 	sync)
 		sync
 		;;
 	unsync)
 		# make sure the daemon is running
-		[[ -f /run/daemons/psd ]] && unsync
+		[[ -f $PIDFILE ]] && unsync
 		;;
 	*)
 		# keep user from messing with this directly
-		[[ -f /run/daemons/psd ]] && echo "You should not call this script directly, let the /etc/rc.d/psd do it for you!" && exit 1
+		[[ -f $PIDFILE ]] && echo "You should not call this script directly, let the /etc/rc.d/psd do it for you!" && exit 1
 esac
 exit 0
--- psd.conf
+++ psd.conf
@@ -2,6 +2,9 @@
 # /etc/psd.conf
 #
 # For documentation, see: https://wiki.archlinux.org/index.php/Profile-sync-daemon
+
+# path to PIDFILE to use
+PIDFILE=/var/run/psd.pid
 
 # List users separated by spaces whose browser(s) profile(s) will get symlinked 
 # and sync'ed to tmpfs.
