--- profile-sync-daemon.old	2012-01-09 21:01:57.000000000 +0100
+++ profile-sync-daemon	2012-01-10 00:33:34.368059351 +0100
@@ -2,6 +2,8 @@
 # By graysky <graysky AT archlinux DOT us>
 # Main part of this script adapted from work by Colin Verot
 
+PIDFILE=/var/run/psd.pid
+
 . /etc/psd.conf
 
 set_which() {
@@ -84,17 +86,17 @@
 case "$1" in
 	check)
 		# do not run if the daemon is running
-		[[ ! -f /run/daemons/psd ]] && check
+		[[ ! -f $PIDFILE ]] && check
 		;;
 	sync)
 		sync
 		;;
 	unsync)
 		# make sure the daemon is running
-		[[ -f /run/daemons/psd ]] && unsync
+		[[ -f $PIDFILE ]] && unsync
 		;;
 	*)
 		# keep user from messing with this directly
-		[[ -f /run/daemons/psd ]] && echo "You should not call this script directly, let the /etc/rc.d/psd do it for you!" && exit 1
+		[[ -f $PIDFILE ]] && echo "You should not call this script directly, let the /etc/rc.d/psd do it for you!" && exit 1
 esac
 exit 0
