## Fixes jack depend and reduces daemons permissions
--- /usr/src/rivendell-1.2.1/rivendell-gentoo	2007-10-05 09:31:21.000000000 -0500
+++ /etc/init.d/rivendell	2009-01-10 07:24:04.226186550 -0600
@@ -26,17 +26,16 @@
 ##
 
 depend() {
-        need jackd
         use mysql
 }
 
 start() {
         ebegin "Starting Rivendell"
         ipcrm -M 0x5005 2> /dev/null
-        start-stop-daemon --start --quiet --exec /usr/bin/caed
-        start-stop-daemon --start --quiet --exec /usr/bin/ripcd
+        start-stop-daemon --start --quiet --chuid rivendell --exec /usr/bin/caed
+        start-stop-daemon --start --quiet --chuid rivendell --exec /usr/bin/ripcd
         sleep 1
-        start-stop-daemon --start --quiet --exec /usr/bin/rdcatchd
+        start-stop-daemon --start --quiet --chuid rivendell --exec /usr/bin/rdcatchd
         eend $?
 }
 
