diff -Naur sql2class-1.9.1.orig/Parse.h sql2class-1.9.1/Parse.h
--- sql2class-1.9.1.orig/Parse.h	2006-08-06 12:04:04.000000000 +0200
+++ sql2class-1.9.1/Parse.h	2006-08-06 12:04:31.000000000 +0200
@@ -28,25 +28,25 @@
 class Parse
 {
 public:
-	Parse::Parse();
-	Parse::Parse(char *);
-	Parse::Parse(char *,char *);
-	Parse::~Parse();
-	short Parse::issplit(char);
-	void Parse::getsplit(void);
-	void Parse::getsplit(char *);
-	void Parse::getword(void);
-	void Parse::getword(char *);
-	void Parse::getword(char *,char *,int);
-	void Parse::getrest(char *);
-	long Parse::getvalue(void);
-	void Parse::setbreak(char);
-	int Parse::getwordlen(void);
-	int Parse::getrestlen(void);
-	void Parse::enablebreak(char c) {
+	Parse();
+	Parse(char *);
+	Parse(char *,char *);
+	~Parse();
+	short issplit(char);
+	void getsplit(void);
+	void getsplit(char *);
+	void getword(void);
+	void getword(char *);
+	void getword(char *,char *,int);
+	void getrest(char *);
+	long getvalue(void);
+	void setbreak(char);
+	int getwordlen(void);
+	int getrestlen(void);
+	void enablebreak(char c) {
 		pa_enable = c;
 	}
-	void Parse::disablebreak(char c) {
+	void disablebreak(char c) {
 		pa_disable = c;
 	}
 
