--- src/freenet/siteinserter.cpp	2009-01-04 16:20:56.000000000 +0100
+++ src/freenet/siteinserter.cpp.new	2009-01-13 19:37:36.000000000 +0100
@@ -4,6 +4,7 @@
 #include <Poco/DateTime.h>
 #include <Poco/Timespan.h>
 #include <Poco/DateTimeFormatter.h>
+#include <cstdio>
 
 #ifdef XMEM
 	#include <xmem.h>
--- src/http/fmshttprequesthandlerfactory.cpp	2008-12-03 19:17:26.000000000 +0100
+++ src/http/fmshttprequesthandlerfactory.cpp.new	2009-01-13 19:38:32.000000000 +0100
@@ -23,6 +23,7 @@
 #include "../../include/http/pages/forumthreadspage.h"
 #include "../../include/http/pages/forumviewthreadpage.h"
 #include "../../include/http/pages/forumcreatepostpage.h"
+#include <cstdio>
 
 FMSHTTPRequestHandlerFactory::FMSHTTPRequestHandlerFactory()
 {
--- src/http/pages/showimagepage.cpp	2008-12-01 18:22:38.000000000 +0100
+++ src/http/pages/showimagepage.cpp.new	2009-01-13 19:40:59.000000000 +0100
@@ -1,4 +1,5 @@
 #include "../../../include/http/pages/showimagepage.h"
+#include <cstdio>
 
 #ifdef XMEM
 	#include <xmem.h>
--- src/nntp/uwildmat.cpp	2008-05-02 08:39:44.000000000 +0200
+++ src/nntp/uwildmat.cpp.new	2009-01-13 19:52:18.000000000 +0100
@@ -67,6 +67,7 @@
 #include "../../include/nntp/uwildmat.h"
 #include <string>
 #include <cstring>
+#include <stdint.h>
 
 #define ABORT -1
 
--- src/nntp/mime/Mime.cpp	2008-05-02 08:38:38.000000000 +0200
+++ src/nntp/mime/Mime.cpp.new	2009-01-13 20:01:33.000000000 +0100
@@ -13,6 +13,7 @@
 #include <stdlib.h>
 #include <time.h>
 #include <cstring>
+#include <stdio.h>
 
 #ifndef _WIN32
 	#define stricmp strcasecmp
