--- configure.ac.orig	2010-04-25 14:18:36.000000000 -0300
+++ configure.ac	2010-04-25 14:20:46.000000000 -0300
@@ -83,45 +83,8 @@
 if test $ac_cv_header_xpm_h = yes; then
 	MH_XLIBS="$MH_XLIBS -lXpm"
 fi
-dnl ---------- allow --enable-debug to compile in debug mode ---------
-AC_ARG_ENABLE(debug,
-	[  --enable-debug          turn on debugging],
-	[with_debug=$enableval],
-	[with_debug=no],
-)
-cflags_g="`echo $CFLAGS | grep -c '\-g'`"
-cflags_O="`echo $CFLAGS | grep -c '\-O'`"
-
-if test "$with_debug" = yes; then
-	if test "$cflags_g" = "0"; then
-		CFLAGS="${CFLAGS} -g"
-	fi
-	if test "$cflags_O" != "0"; then
-		CFLAGS="`echo ${CFLAGS} | sed -e s/-O.//`"
-	fi
-	CFLAGS="${CFLAGS} -DPDCDEBUG"
-else
-	if test "$cflags_O" = "0"; then
-		CFLAGS="${CFLAGS} -O"
-	fi
-	if test "$cflags_g" != "0"; then
-		CFLAGS="`echo ${CFLAGS} | sed -e s/-g//`"
-	fi
-fi
-if test "$ac_cv_prog_CC" = "gcc"; then
-	if test "$with_debug" = yes; then
-		CFLAGS="${CFLAGS} -Wall"
-	else
-		CFLAGS="-O2 -Wall -fomit-frame-pointer"
-	fi
-fi
-if test "$on_qnx" = yes; then
-	if test "$with_debug" = yes; then
-		CFLAGS="-g"
-	else
-		CFLAGS="-Otax"
-	fi
-fi
+
+CFLAGS="${CFLAGS} -Wall"
 
 dnl --------------- check for wide character support -----------------
 dnl allow --enable-widec to include wide character support
