Index: src/ticonv.c
===================================================================
--- src/ticonv.c	(Revision 2632)
+++ src/ticonv.c	(Revision 2663)
@@ -447,7 +447,7 @@
 
 	if(model == CALC_TI84P_USB)
 	{
-		ti = ticonv_varname_detokenize(model, src);
+		ti = ticonv_varname_detokenize(CALC_TI84P, src);
 
 		utf16 = ticonv_charset_ti_to_utf16(CALC_TI84P, ti);
 		g_free(ti);
@@ -465,7 +465,7 @@
 	else
 		dst = g_strdup(src);
 	
-	return g_strdup(src);
+	return dst;
 }
 
 /**
Index: src/tokens.c
===================================================================
--- src/tokens.c	(Revision 2632)
+++ src/tokens.c	(Revision 2663)
@@ -355,7 +355,8 @@
 	else 
 		return v-1; 
 }
-
+
+//FIXME: does not work with named list because we should pass the vartype, too
 TIEXPORT char* TICALL ticonv_varname_tokenize(CalcModel model, const char *src_)
 {
 	const unsigned char *src = (const unsigned char *)src_;
Index: src/charset.c
===================================================================
--- src/charset.c	(Revision 2632)
+++ src/charset.c	(Revision 2663)
@@ -322,7 +322,8 @@
         || (*p>=32 && *p<=126)
         || (*p>=161 && *p<=167)
         || (*p>=169 && *p<=172)
-        || (*p>=174 && *p<=186)
+        || (*p>=174 && *p<=183)
+        || (*p>=185 && *p<=187)
         || (*p>=191 && *p<=255)) {
       *(q++)=(unsigned char)*(p++);
     } else switch (*(p++)) {
