--- Makefile.in	2009-09-28 16:09:43.000000000 +0200
+++ Makefile.in.as-needed	2009-10-03 16:42:19.207824900 +0200
@@ -37,8 +37,8 @@
 		$(srcdir)/TAGS
 
 setup: setup.o gettext.o
-	$(CC) $(LDFLAGS) $(LIBOBJS) $(LIBS) $(SETUP_LIBS) \
-	gettext.o setup.o -o setup
+	$(CC) $(LDFLAGS) $(LIBOBJS) gettext.o setup.o -o setup \
+	$(LIBS) $(SETUP_LIBS)
 
 webcit: webserver.o context_loop.o ical_dezonify.o \
 	cookie_conversion.o locate_host.o floors.o summary.o \
@@ -55,7 +55,7 @@
 	decode.o modules_init.o paramhandling.o utils.o \
 	ical_maps.o ical_subst.o static.o \
 	$(LIBOBJS)
-	$(CC) $(LDFLAGS) $(LIBOBJS) $(LIBS) \
+	$(CC) $(LDFLAGS) $(LIBOBJS) -o webcit \
 	webserver.o context_loop.o cookie_conversion.o marchlist.o \
 	webcit.o auth.o tcp_sockets.o mainmenu.o serv_func.o who.o listsub.o \
 	roomops.o roomlist.o messages.o msg_renderers.o userlist.o paging.o sysmsgs.o \
@@ -68,7 +68,6 @@
 	groupdav_put.o http_datestring.o setup_wizard.o fmt_date.o modules_init.o \
 	gettext.o downloads.o addressbook_popup.o pushemail.o sysdep.o decode.o \
 	paramhandling.o utils.o ical_maps.o ical_subst.o static.o \
-	-o webcit \
 	$(LIBS)
 
 .c.o:
