--- examples/hotplug.c.orig	2006-08-22 17:58:31.000000000 -0500
+++ examples/hotplug.c	2006-08-22 17:59:56.000000000 -0500
@@ -23,7 +23,7 @@
   extern int optind;
   extern char *optarg;
   char *udev_action = NULL;
-  char default_udev_action[] = "SYMLINK+=\"libmtp-%k\", MODE=\"666\"";
+  char default_udev_action[] = "SYMLINK+=\"libmtp-%k\", GROUP=\"plugdev\", MODE=\"660\"";
 
   while ( (opt = getopt(argc, argv, "ua:")) != -1 ) {
     switch (opt) {
