diff -urNd apt-cacher-ng-0.3.8.orig/Makefile apt-cacher-ng-0.3.8/Makefile
--- apt-cacher-ng-0.3.8.orig/Makefile	2009-04-11 13:25:45.000000000 +0400
+++ apt-cacher-ng-0.3.8/Makefile	2009-04-11 13:27:53.000000000 +0400
@@ -47,12 +47,12 @@
 # small build, even replacing libstdc++ with supc++ and gcc linking
 in.acng: $(OBJSIN)
 	$(CC) -o $@ $(OBJSIN) -lsupc++ -Wl,--as-needed
-	strip $@
+#	strip $@
 
 apt-cacher-ng: $(OBJS)
 	$(CXX) -o $@ $(OBJS) -lpthread $(LDFLAGS) -lz -lbz2
 	@-test -d ../unstripped && cp $@ ../unstripped/$@.$$(dpkg-parsechangelog |grep ^Version: | cut -f2 -d' ') || true
-	strip $@
+#	strip $@
 
 else
 
@@ -60,7 +60,7 @@
 
 in.acng: $(OBJSIN) $(OBJS)
 	$(CXX) -o $@ acbuf.o client.o acfg.o filereader.o meta.o md5.o sha1.o aclogger.o -lz -lbz2 -Wl,--as-needed
-	strip $@
+#	strip $@
 
 apt-cacher-ng: $(OBJS)
 	$(CXX) -o $@ $(OBJS) -lpthread $(LDFLAGS) -lz -lbz2
@@ -69,7 +69,7 @@
 acngfs: $(OBJSFS)
 	$(CXX) -o acngfs $(OBJSFS) -Wl,--as-needed `pkg-config fuse --libs`
 	@-test -d ../unstripped && cp $@ ../unstripped/$@.$$(dpkg-parsechangelog |grep ^Version: | cut -f2 -d' ') || true
-	strip $@
+#	strip $@
 
 doc: doc/src/README.but doc/src/manpage.but doc/src/acngfs.but
 	mkdir -p doc/.build doc/html doc/man
