diff -dupr Mupen64Plus-1-5-src.orig/pre.mk Mupen64Plus-1-5-src/pre.mk
--- Mupen64Plus-1-5-src.orig/pre.mk	2009-08-01 17:27:41.415292353 +0200
+++ Mupen64Plus-1-5-src/pre.mk	2009-08-01 17:32:29.038293121 +0200
@@ -151,10 +151,10 @@ CC      = gcc
 CXX     = g++
 LD      = g++
 ifeq ($(OS),LINUX)
-STRIP	= strip -s
+STRIP	= true
 endif
 ifeq ($(OS),OSX)
-STRIP	= strip -x 
+STRIP	= true
 endif
 RM      = rm
 RM_F    = rm -f
@@ -185,19 +185,10 @@ else 
 endif
 
 # set base CFLAGS and LDFLAGS
-CFLAGS += -pipe -O3 -ffast-math -funroll-loops -fexpensive-optimizations -fno-strict-aliasing
 CORE_LDFLAGS += -lz -lm -lpng -lfreetype -ldl
 
 # set special flags per-system
 ifeq ($(CPU), X86)
-  ifeq ($(ARCH), 64BITS)
-    CFLAGS += -march=athlon64
-  else
-    CFLAGS += -march=i686 -mtune=pentium-m -mmmx -msse
-    ifneq ($(PROFILE), 1)
-      CFLAGS += -fomit-frame-pointer
-    endif
-  endif
   # tweak flags for 32-bit build on 64-bit system
   ifeq ($(ARCH), 64BITS_32)
     CFLAGS += -m32
@@ -205,7 +196,7 @@ ifeq ($(CPU), X86)
   endif
 endif
 ifeq ($(CPU), PPC)
-  CFLAGS += -mcpu=powerpc -D_BIG_ENDIAN
+  CFLAGS += -D_BIG_ENDIAN
 endif
 
 # set CFLAGS, LIBS, and LDFLAGS for external dependencies
