--- nsis-2.19-src~/SCons/Config/gnu	2006-09-28 16:29:57.000000000 +0200
+++ nsis-2.19-src/SCons/Config/gnu	2006-09-28 17:30:37.000000000 +0200
@@ -72,6 +72,7 @@
 if not defenv['DEBUG']:
 	stub_env.Append(LINKFLAGS = '-s')               # strip
 stub_env.Append(LINKFLAGS = '-mwindows')          # build windows executables
+stub_env.Append(LINKFLAGS = '-L/usr/mingw32/usr/lib')
 stub_env.Append(LINKFLAGS = '$NODEFLIBS_FLAG')    # no standard libraries
 stub_env.Append(LINKFLAGS = '$ALIGN_FLAG')        # 512 bytes align
 stub_env.Append(LINKFLAGS = '-Wl,-e,_WinMain@16') # entry point
