diff -ur nufw-2.0.3~/configure nufw-2.0.3/configure
--- nufw-2.0.3~/configure	2006-06-28 01:34:15.000000000 +0200
+++ nufw-2.0.3/configure	2006-06-28 01:39:09.000000000 +0200
@@ -20041,59 +20041,59 @@
 # Check whether --with-prelude-log or --without-prelude-log was given.
 if test "${with_prelude_log+set}" = set; then
   withval="$with_prelude_log"
-  enable_prelude_log="yes"
+  enable_prelude_log=$withval
 else
-  enable_prelude_log=""
+  enable_prelude_log="no"
 fi;
 
 # Check whether --with-mysql-log or --without-mysql-log was given.
 if test "${with_mysql_log+set}" = set; then
   withval="$with_mysql_log"
-  enable_mysql_log="yes"
+  enable_mysql_log=$withval
 else
-  enable_mysql_log=""
+  enable_mysql_log="no"
 fi;
 
 # Check whether --with-pgsql-log or --without-pgsql-log was given.
 if test "${with_pgsql_log+set}" = set; then
   withval="$with_pgsql_log"
-  enable_pgsql_log="yes"
+  enable_pgsql_log=$withval
 else
-  enable_pgsql_log=""
+  enable_pgsql_log="no"
 fi;
 
 
 # Check whether --with-system-auth or --without-system-auth was given.
 if test "${with_system_auth+set}" = set; then
   withval="$with_system_auth"
-  enable_system_auth="yes"
+  enable_system_auth=$withval
 else
-  enable_system_auth=""
+  enable_system_auth="no"
 fi;
 
 
 # Check whether --with-ldap or --without-ldap was given.
 if test "${with_ldap+set}" = set; then
   withval="$with_ldap"
-  ldap="yes"
+  ldap=$withval
 else
-  ldap=""
+  ldap="no"
 fi;
 
 # Check whether --with-gdbm or --without-gdbm was given.
 if test "${with_gdbm+set}" = set; then
   withval="$with_gdbm"
-  gdbm="yes"
+  gdbm=$withval
 else
-  gdbm=""
+  gdbm="no"
 fi;
 
 # Check whether --with-ident or --without-ident was given.
 if test "${with_ident+set}" = set; then
   withval="$with_ident"
-  ident="yes"
+  ident=$withval
 else
-  ident=""
+  ident="no"
 fi;
 
 
@@ -20101,9 +20101,9 @@
 # Check whether --with-nfqueue or --without-nfqueue was given.
 if test "${with_nfqueue+set}" = set; then
   withval="$with_nfqueue"
-  use_nfqueue="yes"
+  use_nfqueue=$withval
 else
-  use_nfqueue=""
+  use_nfqueue="no"
 fi;
 
 
@@ -20121,7 +20121,7 @@
 # Check whether --with-fixedtimeout or --without-fixedtimeout was given.
 if test "${with_fixedtimeout+set}" = set; then
   withval="$with_fixedtimeout"
-  have_conntrack_fixedtimeout="yes"
+  have_conntrack_fixedtimeout=${withval}
 fi;
 
 
@@ -20130,7 +20130,7 @@
   enableval="$enable_debug"
   debug=$enableval
 else
-  debug=""
+  debug="no"
 fi;
 # Check whether --enable-nufw or --disable-nufw was given.
 if test "${enable_nufw+set}" = set; then
