--- libass-0.9.4/configure.ac.orig	2008-03-16 09:23:39.000000000 +0100
+++ libass-0.9.4/configure.ac	2008-03-16 10:42:17.000000000 +0100
@@ -11,13 +11,29 @@
 AC_PROG_CPP
 
 # Checks for libraries.
-AC_CHECK_LIB([png], [png_create_write_struct], AC_DEFINE(HAVE_LIBPNG, 1, [use libpng]))
-AC_SEARCH_LIBS([iconv_open], [iconv], AC_DEFINE(USE_ICONV, 1, [use iconv]))
-AC_CHECK_LIB([enca], [enca_analyser_alloc], [AC_DEFINE(HAVE_ENCA, 1, [use enca])  LIBS="$LIBS -lenca"])
+
+
+
+AC_ARG_WITH([png], AS_HELP_STRING([--without-png], [Build without png library (default: test)]))
+if test "x$with_png" != "xno"; then
+	AC_CHECK_LIB([png], [png_create_write_struct], AC_DEFINE(HAVE_LIBPNG, 1, [use libpng]))
+fi
+
+AC_ARG_WITH([iconv], AS_HELP_STRING([--without-iconv], [Build without iconv library (default: test)]))
+if test "x$with_iconv" != "xno"; then
+	AC_SEARCH_LIBS([iconv_open], [iconv], AC_DEFINE(USE_ICONV, 1, [use iconv]))
+	AC_CHECK_HEADER([iconv.h])
+fi
+
+AC_ARG_WITH([enca], AS_HELP_STRING([--without-enca], [Build without enca library (default: test)]))
+if test "x$with_enca" != "xno"; then
+	AC_CHECK_LIB([enca], [enca_analyser_alloc], [AC_DEFINE(HAVE_ENCA, 1, [use enca])  LIBS="$LIBS -lenca"])
+	AC_CHECK_HEADER([enca.h])
+fi
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS([inttypes.h stdint.h stdlib.h string.h sys/time.h unistd.h iconv.h enca.h])
+AC_CHECK_HEADERS([inttypes.h stdint.h stdlib.h string.h sys/time.h unistd.h])
 
 PKG_CHECK_MODULES([FONTCONFIG], fontconfig >= 2.2.0, [
     CFLAGS="$CFLAGS $FONTCONFIG_CFLAGS"
