--- src/fusepod_ipod.cpp	2006-08-08 23:28:23.000000000 +0200
+++ src/fusepod_ipod.cpp.new	2007-01-17 20:58:17.000000000 +0100
@@ -417,7 +417,7 @@
 
     stats << "FUSEPod Version: " << PACKAGE_VERSION << endl;
     stats << "ITunesDB Version: " << this->ipod->version << endl;
-    stats << "Real Mountpoint: " << this->ipod->mountpoint << endl;
+    stats << "Real Mountpoint: " << itdb_get_mountpoint(this->ipod) << endl;
 
     Playlist * mpl = itdb_playlist_mpl (ipod);
     stats << "iPod Name: " << (mpl && mpl->name ? string(mpl->name) : "Unknown") << endl; 
