diff -Nur incron-0.5.5/Makefile incron-0.5.5-r1/Makefile
--- incron-0.5.5/Makefile	2007-02-14 15:54:33.000000000 +0100
+++ incron-0.5.5-r1/Makefile	2007-02-23 09:57:16.000000000 +0100
@@ -1,9 +1,10 @@
 
-PREFIX = /usr/local
-USERDATADIR = /var/spool/incron
-SYSDATADIR = /etc/incron.d
-CFGDIR = /etc
-MANPATH = /usr/share/man
+PREFIX = $(DESTDIR)usr
+USERDATADIR = $(DESTDIR)var/spool/incron
+SYSDATADIR = $(DESTDIR)etc/incron.d
+CFGDIR = $(DESTDIR)etc
+MANPATH = $(DESTDIR)usr/share/man
+INITDIR = $(DESTDIR)etc/init.d
 RELEASE = incron-`cat VERSION`
 RELEASEDIR = /tmp/$(RELEASE)
 
@@ -18,7 +19,7 @@
 CXXAUX = -pipe
 
 CXXFLAGS = $(OPTIMIZE) $(DEBUG) $(WARNINGS) $(CXXAUX)
-LDFLAGS = $(WARNINGS)
+LDFLAGS += $(WARNINGS)
 
 PROGRAMS = incrond incrontab
 
@@ -45,11 +46,15 @@
 
 install:	all install-man
 	[ -d $(PREFIX) ]
+	$(INSTALL) -m 0755 -d $(PREFIX)/bin
+	$(INSTALL) -m 0755 -d $(PREFIX)/sbin
+	$(INSTALL) -m 0755 -d $(INITDIR)/
 	$(INSTALL) -m 04755 -o $(USER) incrontab $(PREFIX)/bin/
 	$(INSTALL) -m 0755 incrond $(PREFIX)/sbin/
 	$(INSTALL) -m 0755 -o $(USER) -d $(USERDATADIR)
 	$(INSTALL) -m 0755 -o $(USER) -d $(SYSDATADIR)
 	$(INSTALL) -m 0644 -o $(USER) incron.conf.example $(CFGDIR)
+	$(INSTALL) -m 0755 incrond-init.d $(INITDIR)/incrond
 
 install-man:	incrontab.1 incrontab.5 incrond.8 incron.conf.5
 	$(INSTALL) -m 0755 -d $(MANPATH)/man1
diff -Nur incron-0.5.5/incrond-init.d incron-0.5.5-r1/incrond-init.d
--- incron-0.5.5/incrond-init.d	1970-01-01 01:00:00.000000000 +0100
+++ incron-0.5.5-r1/incrond-init.d	2007-02-23 09:53:59.000000000 +0100
@@ -0,0 +1,22 @@
+#!/sbin/runscript
+# Copyright 1999-2004 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header:  $
+
+
+depend() {
+	need localmount
+	use logger
+}
+
+start() {
+	ebegin "Starting incrond"
+	/usr/sbin/incrond 
+	eend ${?}
+}
+
+stop() {
+	ebegin "Stopping incrond"
+	/usr/sbin/incrond -k
+	eend ${?}
+}
