--- igmpproxy.orig/src/igmpproxy.c	2008-09-17 10:02:40.000000000 +0200
+++ igmpproxy/src/igmpproxy.c	2008-09-17 10:03:00.000000000 +0200
@@ -87,9 +87,6 @@
     // Set the default config Filepath...
     char* configFilePath = IGMPPROXY_CONFIG_FILEPATH;
 
-    // Display version 
-    fputs( Version, stderr );
-
     // Parse the commandline options and setup basic settings..
     int i = 1;
     while (i < ArgCn) {
