--- install.cmake
+++ install.cmake
@@ -2,11 +2,12 @@
 ############### Prepare the installation of the compiled stuff here!
 
 IF(BUILD_TYPE STREQUAL WIN32)
-SET(SHAREDIR CGenius)
-SET(APPDIR CGenius)
+SET(SHAREDIR CGenius CACHE PATH "")
+SET(APPDIR CGenius CACHE PATH "")
 ELSE(BUILD_TYPE STREQUAL WIN32)
-SET(SHAREDIR share/CommanderGenius)
-SET(APPDIR games)
+SET(SHAREDIR share/CommanderGenius CACHE PATH "")
+SET(APPDIR games CACHE PATH "")
+SET(DOCDIR ${SHAREDIR} CACHE PATH "")
 ENDIF(BUILD_TYPE STREQUAL WIN32)
 
 # This will install the application itself
@@ -19,11 +20,11 @@
 
 # This will copy the readme file. 
 INSTALL(FILES readme.txt 
-	DESTINATION ${SHAREDIR})
+	DESTINATION ${DOCDIR})
 
 # This will copy the readme file. 
 INSTALL(FILES changelog.txt 
-	DESTINATION ${SHAREDIR})
+	DESTINATION ${DOCDIR})
 
 # Windows might not have those dlls so we ship them with the packages
 IF(BUILD_TYPE STREQUAL WIN32)
