--- squidclamav~/Makefile	2006-06-30 18:17:32.000000000 +0200
+++ squidclamav/Makefile	2006-06-30 18:38:02.000000000 +0200
@@ -1,9 +1,9 @@
 # Makefile for SquidClamav
 #
 
-OFILES = squidclamav.o main.o config.o log.o lists.o regex.o
+OFILES = squidclamav.o main.o config.o log.o lists.o
 
-HFILES = squidclamav.h paths.h lists.h log.h regex.h
+HFILES = squidclamav.h paths.h lists.h log.h
 
 CC=gcc
 #CC=checkergcc
@@ -13,7 +13,7 @@
 
 BINARIES = squidclamav
 
-CFLAGS = -O3 -Wall -funroll-loops -fno-builtin-log
+CFLAGS ?= -O3 -Wall -funroll-loops -fno-builtin-log
 #CFLAGS = -Wall -g
 
 all:	$(BINARIES)
