diff -Naur palantir-2.6/server/Makefile palantir-2.6.new/server/Makefile
--- palantir-2.6/server/Makefile	2005-09-02 14:33:50.000000000 +0200
+++ palantir-2.6.new/server/Makefile	2006-07-09 00:08:24.000000000 +0200
@@ -8,7 +8,7 @@
 
 # most installation directories are relative to this one.
 # Some might want to choose "/usr" instead.
-BASE_DIR=	/usr/local
+BASE_DIR=	/usr
 
 # the owner of the system directory and of the named pipe contained therein.
 # Note that since pipe has permission 600, all processes which interface to
@@ -21,12 +21,12 @@
 SYS_DIR=	$(BASE_DIR)/share/palantir
 
 # location of default config file
-CONFIG_FILE=	$(SYS_DIR)/palantir.conf
+CONFIG_FILE=	/etc/palantir/palantir.conf
 
 # named pipe used as a supplementary I/O channel.
 # For Palantir to use it, its correct path must be specified as a
 # NamedPipe directive in the configuration file
-NAMED_PIPE=	$(SYS_DIR)/telmu_pipe
+#NAMED_PIPE=	$(SYS_DIR)/telmu_pipe
 
 # directory where to install executables in
 BIN_DIR=	$(BASE_DIR)/bin
@@ -35,7 +35,7 @@
 MAN_DIR=	$(BASE_DIR)/man
 
 # location of default log file
-LOG_FILE=	/var/log/palantir
+LOG_FILE=	/var/log/palantir.log
 
 # set to 1 to enable capture monitoring in a stand-alone window
 # (requires Imlib)
@@ -46,9 +46,7 @@
 
 .PHONY : tools man
 
-
-CFLAGS= -Wall -D_REENTRANT -O3 -g
-CC= gcc
+CFLAGS+= -D_REENTRANT
 libjpeg= -ljpeg
 
 ifeq ($(TARGET),mips)
