diff -u /var/tmp/portage/app-arch/fsarchiver-0.6.1/work/fsarchiver-0.6.1/configure.ac /tmp/buffer-content-5978Hkj
--- a/configure.ac	2009-10-04 04:45:19.000000000 -0400
+++ b/configure.ac	2009-11-21 15:57:20.000000000 -0500
@@ -48,40 +48,43 @@
 AC_CHECK_HEADERS(bzlib.h)
 
 dnl option to disable lzma support (for people who don't have xz installed)
-AC_ARG_ENABLE(lzma,
-    [  --disable-lzma          don't compile the support for lzma compression (which requires liblzma)],
-  ,
-  [AC_DEFINE([OPTION_LZMA_SUPPORT], 1, [Define to 1 to enable the support for lzma compression])
+AC_ARG_ENABLE([lzma],
+    [AS_HELP_STRING([--disable-lzma], [don't compile the support for lzma compression (which requires liblzma)])],
+    [enable_lzma=$enableval],
+    [enable_lzma=yes])
+if test "x$enable_lzma" = "xyes"; then
+  AC_DEFINE([OPTION_LZMA_SUPPORT], 1, [Define to 1 to enable the support for lzma compression])
   AC_CHECKING([for liblzma (library and header files)])
   AC_CHECK_LIB([lzma], [lzma_easy_encoder], [LIBS="$LIBS -llzma"],
      AC_MSG_ERROR([*** lzma library (liblzma) not found: you can either install xz-4.999.9beta.tar.gz (or newer) or disable lzma support using --disable-lzma]))
   AC_CHECK_HEADERS(lzma.h)
-  ]
-)
+fi
 
 dnl option to disable lzo support (for people who don't have liblzo2 installed)
-AC_ARG_ENABLE(lzo,
-    [  --disable-lzo           don't compile the support for lzo compression (which requires liblzo2)],
-  ,
-  [AC_DEFINE([OPTION_LZO_SUPPORT], 1, [Define to 1 to enable the support for lzo compression])
+AC_ARG_ENABLE([lzo],
+    [AS_HELP_STRING([--disable-lzo], [don't compile the support for lzo compression (which requires liblzo2)])],
+    [enable_lzo=$enableval],
+    [enable_lzo=yes])
+if test "x$enable_lzo" = "xyes"; then
+  AC_DEFINE([OPTION_LZO_SUPPORT], 1, [Define to 1 to enable the support for lzo compression])
   AC_CHECKING([for liblzo2 (library and header files)])
   AC_CHECK_LIB([lzo2], [lzo1x_1_compress], [LIBS="$LIBS -llzo2"],
      AC_MSG_ERROR([*** lzo library (liblzo2) not found: please install liblzo (you may also have to install lzo-devel) or disable lzo support using --disable-lzo]))
   AC_CHECK_HEADERS(lzo/lzo1x.h)
-  ]
-)
+fi
 
 dnl option to disable crypto support (for people who don't have libgcrypt installed)
-AC_ARG_ENABLE(crypto,
-    [  --disable-crypto        don't compile the support for encryption (which requires libgcrypt)],
-  ,
-  [AC_DEFINE([OPTION_CRYPTO_SUPPORT], 1, [Define to 1 to enable the support for archive encryption using libgcrypt])
+AC_ARG_ENABLE([crypto],
+    [AS_HELP_STRING([--disable-crypto], [don't compile the support for encryption (which requires libgcrypt)])],
+    [enable_crypto=$enableval],
+    [enable_crypto=yes])
+if test "x$enable_crypto" = "xyes"; then
+  AC_DEFINE([OPTION_CRYPTO_SUPPORT], 1, [Define to 1 to enable the support for archive encryption using libgcrypt])
   AC_CHECKING([for libgcrypt (library and header files)])
   AC_CHECK_LIB([gcrypt], [gcry_cipher_encrypt], [LIBS="$LIBS -lgcrypt -lgpg-error"],
      AC_MSG_ERROR([*** libgcrypt not found: you can either install libgcrypt (and the development libgcrypt package) or disable the support for encryption using --disable-crypto]))
   AC_CHECK_HEADERS(gcrypt.h)
-  ]
-)
+fi
 
 dnl check e2fsprogs and its libs
 AC_CHECKING([for e2fsprogs-1.39 or newer])
@@ -122,13 +125,19 @@
     AC_SUBST(LOGDIR)
 
 dnl options for developers
-AC_ARG_ENABLE(devel,
-    [  --enable-devel          enable options for developers (debug, ...)],
-    AC_DEFINE([OPTION_DEVEL_SUPPORT], 1, [Define to 1 to enable options for development]),)
+AC_ARG_ENABLE([devel],
+    [AS_HELP_STRING([--enable-devel],[enable options for developers (debug, ...)])],
+    [enable_devel=$enableval],
+    [enable_devel=no])
+if test "x$enable_devel" = "xyes"; then
+   AC_DEFINE([OPTION_DEVEL_SUPPORT], 1, [Define to 1 to enable options for development])
+fi
 
 dnl static executables?
-AC_ARG_ENABLE(static,
-    [  --enable-static         build static binaries], allstatic=$enableval, allstatic=no)
+AC_ARG_ENABLE([static],
+    [AS_HELP_STRING([--enable-static],[build static binaries])], 
+    [allstatic=$enableval], 
+    [allstatic=no])
 
 if test "$allstatic" = "yes"
 then

Diff finished.  Sat Nov 21 15:57:20 2009
