diff -urN freej-0.7-orig/src/include/blitter.h freej-0.7/src/include/blitter.h
--- freej-0.7-orig/src/include/blitter.h	2004-08-29 23:48:18.000000000 +0200
+++ freej-0.7/src/include/blitter.h	2006-06-04 18:52:11.000000000 +0200
@@ -114,7 +114,7 @@
   bool fade_value(int step, int val); ///< fade to a new blit value
   bool pulse_value(int step, int val); ///< pulse it to a value and come back
   bool set_kernel(short *krn); /// set the convolution kernel
-  bool Blitter::set_colorkey(int x,int y);
+  bool set_colorkey(int x,int y);
   Linklist blitlist; ///< list of available blits
 
   /* ==== geometrical transformations */
diff -urN freej-0.7-orig/src/include/linklist.h freej-0.7/src/include/linklist.h
--- freej-0.7-orig/src/include/linklist.h	2004-09-23 12:40:00.000000000 +0200
+++ freej-0.7/src/include/linklist.h	2006-06-04 18:52:20.000000000 +0200
@@ -50,7 +50,7 @@
   Entry *search(char *name);
   int *completion(char *needle);
 
-  Entry *Linklist::selected();
+  Entry *selected();
 
   Entry *operator[](int pos) { return pick(pos); };
 
