--- src/freej.cpp.old	2004-11-20 00:27:32.301971896 +0900
+++ src/freej.cpp	2004-11-20 00:27:44.193164160 +0900
@@ -25,6 +25,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <assert.h>
+#include <cerrno>
 
 #include <context.h>
 
--- src/console.cpp.old	2004-11-20 00:31:48.678996664 +0900
+++ src/console.cpp	2004-11-20 00:32:06.817239232 +0900
@@ -23,6 +23,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <dirent.h>
+#include <cerrno>
 
 #include <signal.h>
 #include <slang.h>
--- src/scroll_layer.cpp.old	2004-11-20 00:40:18.799446568 +0900
+++ src/scroll_layer.cpp	2004-11-20 00:41:16.402689544 +0900
@@ -23,6 +23,7 @@
  */
 
 #include <stdlib.h>
+#include <cerrno>
 
 #include <scroll_layer.h>
 #include <context.h>
