diff -urN rscds-0.9.5.orig/inc/drivers_ldap.php rscds-0.9.5/inc/drivers_ldap.php
--- rscds-0.9.5.orig/inc/drivers_ldap.php	2008-06-20 09:55:09.000000000 +0200
+++ rscds-0.9.5/inc/drivers_ldap.php	2008-06-20 09:56:42.000000000 +0200
@@ -105,7 +105,7 @@
       if (!ldap_bind($this->connect,$config['bindDN'],$config['passDN'])){
           $bindDN = isset($config['bindDN']) ? $config['bindDN'] : 'anonymous';
           $passDN = isset($config['passDN']) ? $config['passDN'] : 'anonymous';
-          dbg_error_log( "LDAP", i18n('drivers_ldap : Failed to bind to host %1$s on port %2$s with bindDN of %3$s'), $host, $port $bindDN );
+          dbg_error_log( "LDAP", i18n('drivers_ldap : Failed to bind to host %1$s on port %2$s with bindDN of %3$s'), $host, $port, $bindDN );
           $c->messages[] = i18n( 'drivers_ldap : Unable to bind to LDAP - check your configuration for bindDN and passDN, and that your LDAP server is reachable');
           $this->valid=false;
           return ;
