--- bugle-0.0.20061109.orig/Makefile.am	2006-11-09 13:35:00.000000000 +0100
+++ bugle-0.0.20061109/Makefile.am	2006-11-09 13:35:13.000000000 +0100
@@ -122,7 +122,7 @@
 # ccache doesn't play nice with -fdump-translation-unit, so we disable it
 src/data/gl.tu: src/data/gl.c src/data/overrides.h
 	$(INSTALL) -d src/data
-	CCACHE_DISPLAY=1 $(COMPILE) -fdump-translation-unit -c $(srcdir)/src/data/gl.c -o src/data/gl.o
+	CCACHE_DISABLE=1 $(COMPILE) -fdump-translation-unit -c $(srcdir)/src/data/gl.c -o src/data/gl.o
 	rm -f src/data/gl.o
 	mv gl.c.`test -f gl.c.t00.tu && echo t00.`tu src/data/gl.tu
 
