--- solenv/bin/make_installer.pl.orig	2009-03-14 14:28:15.000000000 +0500
+++ solenv/bin/make_installer.pl	2009-03-14 14:29:33.000000000 +0500
@@ -1388,11 +1388,11 @@
 			# 1. copy all files that need to be stripped locally 
 			# 2. strip all these files
 
-			if ( $installer::globals::strip )
-			{
-				installer::strip::strip_libraries($filesinpackage, $languagestringref);
-				if ( $installer::globals::globallogging ) { installer::files::save_array_of_hashes($loggingdir . $packagename ."_files.log", $filesinpackage); }
-			}
+			#if ( $installer::globals::strip )
+			#{
+			#	installer::strip::strip_libraries($filesinpackage, $languagestringref);
+			#	if ( $installer::globals::globallogging ) { installer::files::save_array_of_hashes($loggingdir . $packagename ."_files.log", $filesinpackage); }
+			#}
 			
 			###############################################################
 			# Searching for files in $filesinpackage with flag LINUXLINK
