--- bin/apply.pl.orig	2009-05-30 14:14:39.000000000 +0600
+++ bin/apply.pl	2009-05-30 14:14:53.000000000 +0600
@@ -918,7 +918,7 @@
     my $ver_line;
     my $is_old = 1;
 
-    open ($Patch, "/usr/local/bin/gpatch --version|") || die "Can't run patch: $!";
+    open ($Patch, "patch --version|") || die "Can't run patch: $!";
     $ver_line = <$Patch>;
     $ver_line =~ m/\s+(\d+)\.(\d+)\.(\d+)/ || die "Can't get patch version\n";
     if ($1 >= 2 && $2 >= 5 && $3 >= 9) {
@@ -1280,7 +1280,7 @@
 
     $opts = join ' ', @arguments;
     $patch_args = " -l -p0 $opts -d $dest_dir";
-    $patch_cmd = "/usr/local/bin/gpatch";
+    $patch_cmd = "patch";
     if ($pieces) {
 	# nasty path mess
 	$patch_cmd = "$patch_dir/../../bin/sloppypatch.pl";
--- bin/enable-dict.orig	2009-05-30 13:44:33.000000000 +0600
+++ bin/enable-dict		2009-05-30 13:44:40.000000000 +0600
@@ -6,7 +6,7 @@
 
 test -f $ARC || exit 1
 cd $WRKDIR/dictionaries || exit 2
-gtar xjf $ARC
+tar xjf $ARC
 grep -q $LNG prj/build.lst && exit 0
 line="di      dictionaries\\$LNG      nmake   -       all     di_$LNG NULL"
 echo "$line" >> prj/build.lst
