--- patches/dev300/apply.orig	2009-06-28 13:42:16.000000000 +0600
+++ patches/dev300/apply	2009-06-28 13:58:45.000000000 +0600
@@ -151,6 +151,12 @@
 
 InfraUnix:  LinuxCommon, CJK, OOXML, CustomUserConfig, PostgreSQL, InfraPatches, NotInfraLinux, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, IntegrateExtensions
 InfraLinux:  UbuntuJaunty, CustomUserConfig, InfraPatches, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, InfraLinuxOnly
+InfraGentooBase: Gentoo, SVGImport, UpdaterRemoval, Layout, LayoutDialogs, CJK, OOXML, Split, Fpickers, QuickStarter, CustomUserConfig, NotInfraLinux, InfraPatches, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, CustomKeyBinding, InfraLinuxOnly
+InfraGentoo: InfraGentooBase
+InfraGentooPG: InfraGentooBase, InfraUnixPatchesPG, PostgreSQL
+InfraDictru: InfraDictRU
+InfraDictuk: InfraDictUK
+InfraDicttr: InfraDictTR
 InfraSolaris:  LinuxCommon, CustomUserConfig, InfraPatches, NotInfraLinux, Infra1252toLocale, InfraSolaris, NotDebian, IntegrateExtensions
 InfraWin32: Win32Common, CustomUserConfig, InfraPatches, NotInfraLinux, InfraWin32Patches, Infra1252toLocale, InfraNotSolaris, IntegrateExtensions
 InfraFake: QuickStarter, WMF, SVGImport, UnitTesting, Layout, CJK, EMFPlus, Fpickers, LayoutDialogs, UnitBootstrap, AutoCorrectCapsLock
@@ -1601,7 +1607,7 @@
 sc-export-shape-macro-bindings.diff, n#304739, noelpwer
 
 # export of hlink bindings for shapes
-# sc-export-shape-hlink-bindings.diff, n#304739, noelpwer
+sc-export-shape-hlink-bindings.diff, n#304739, noelpwer
 
 # attributes present in the vba streams can override the default shape name
 # used for the control in the binary format
@@ -1768,6 +1774,8 @@
 # fix crash when attempting to use geometry attributes of
 # control on sheet/document
 vba-fix-control-shape-geom-fix.diff, n#500006
+#must ask kendy about this ( noelp )
+vba-parallel-build.diff, #no-upstrea
 [ VBAUntested ]
 SectionOwner => noelpwer
 # needs more extensive testing, disable for 3.1
@@ -2970,7 +2978,7 @@
 xlsx-shared-import-and-export.diff
 
 xlsx-arabic-export-crash.diff, n#497419, janneke
-xlsx-export-notes-avoid-dereferencing-0-svx.diff, n#497417, janneke
+#xlsx-export-notes-avoid-dereferencing-0-svx.diff, n#497417, janneke
 xlsx-export-set-xml-2007-flavour.diff, n#502090, janneke
 xlsx-export-simple-autofilter.diff, n#497559, janneke
 # hack to ignore writerfilter when odf-converter is present
@@ -3065,15 +3073,22 @@
 sc-toogle-grid-menubar.diff
 infra-palette.diff
 
+[InfraDictRU]
+dictionaries-ru-RU.diff
+
+[InfraDictUK]
+dict-uk.diff
+
+[InfraDictTR]
+extensions-zemberek.diff
+
 [NotInfraLinux]
 build-xterm-title.diff
-no-ure-sdk.diff
+#no-ure-sdk.diff
 enable-quickstart.diff
 updates.diff
 build-javainstaller2.diff
 icons-galaxy-as-default.diff
-dictionaries-ru-RU.diff
-dict-uk.diff
 extras-add-educate-officorr-misc-offimisc.diff
 reportbuilder-no-license-infra.diff
 wikipublisher-no-license-infra.diff
@@ -3081,7 +3096,6 @@
 scp2_extensions-remove-SUN.diff
 extensions-dont-install-by-default.diff
 extensions-separate-java-based.diff
-extensions-zemberek.diff
 officecfg-create-backups.diff
 
 [InfraLinuxOnly]
@@ -3090,8 +3104,10 @@
 sfx2-remove-check-update-on-fileload.diff, kohei
 default-system-fpicker.diff, michael
 
-[InfraUnixPatches]
+[InfraUnixPatchesPG]
 connectivity-freebsd-libs.diff
+
+[InfraUnixPatches]
 freebsd-store-alloca.diff
 freebsd-gperf.diff, i#85469, rail
 freebsd-writerfilter.diff
