--- configure.in.orig	2008-10-26 12:57:20.000000000 +0500
+++ configure.in	2008-10-26 12:57:57.000000000 +0500
@@ -808,6 +813,10 @@
   --disable-verbose       Decrease build verbosity.
 ],,)
 
+AC_ARG_ENABLE(gstreamer,
+[  --enable-gstreamer      Enables GStreamer avmedia backend
+],,)
+
 BUILD_TYPE="OOo"
 
 dnl ===================================================================
@@ -3367,12 +3376,12 @@
       for a in "$RPM" rpmbuild rpm; do
          $a --usage >/dev/null 2> /dev/null
          if test $? -eq 0; then
-            RPM=$a
+#            RPM=$a
             break
          else
             $a --version >/dev/null 2> /dev/null
             if test $? -eq 0; then
-               RPM=$a
+#               RPM=$a
                break
             fi 
          fi
@@ -3385,7 +3394,8 @@
        fi
    fi
    if echo "$PKGFORMAT" | $EGREP deb 2>&1 >/dev/null; then
-      AC_PATH_PROG(DPKG, dpkg, no)
+#      AC_PATH_PROG(DPKG, dpkg, no)
+	  DPKG=no
       if test "$DPKG" = "no"; then
          AC_MSG_ERROR([dpkg needed for deb creation. Install dpkg.])
       fi
@@ -4171,7 +4181,7 @@
 AC_MSG_CHECKING([whether to build Mozilla addressbook connectivity])
 if test "$enable_mozilla" = "no"; then
    AC_MSG_RESULT([no])
-elif test "$with_system_mozilla" = "yes"; then
+elif test "$with_system_mozilla" = "yes" -o "$enable_kdeab" = "yes"; then
    AC_MSG_RESULT([no, not possible with system-mozilla])
 else
    AC_MSG_RESULT([yes])
