--- source/CMakeLists.txt	2008-02-02 15:06:06.000000000 +0100
+++ source/CMakeLists.txt	2008-02-02 15:26:12.000000000 +0100
@@ -845,8 +845,6 @@
 
 add_executable(vavoom ${MAIN_SOURCES})
 if(UNIX)
-set_target_properties(vavoom PROPERTIES OUTPUT_NAME ../vavoom.i686)
-else(UNIX)
 set_target_properties(vavoom PROPERTIES OUTPUT_NAME ../vavoom)
 endif(UNIX)
 if (MAIN_COMPILE_FLAGS)
@@ -857,20 +855,6 @@
 
 install(TARGETS vavoom DESTINATION ${BINDIR})
 
-install(FILES vavoom.png DESTINATION ${DATADIR})
-
-if(UNIX)
-	file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/vavoom
-"#!/bin/sh
-# Needed to make symlinks/shortcuts work.
-# the binaries must run with correct working directory
-cd \"${CMAKE_INSTALL_PREFIX}/${DATADIR}\"
-\"${CMAKE_INSTALL_PREFIX}/${BINDIR}/vavoom.i686\" $* ${IWADDIR}
-exit $?
-")
-	install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/vavoom DESTINATION ${BINDIR})
-endif(UNIX)
-
 endif(ENABLE_CLIENT)
 
 #---------------------------------------
@@ -889,9 +873,7 @@
 
 add_executable(vavoom-dedicated ${SERVER_SOURCES})
 if(UNIX)
-set_target_properties(vavoom-dedicated PROPERTIES OUTPUT_NAME ../vavoom-dedicated.i686)
-else(UNIX)
-set_target_properties(vavoom-dedicated PROPERTIES OUTPUT_NAME ../vavoom-dedicated)
+set_target_properties(vavoom-dedicated PROPERTIES OUTPUT_NAME ../vavoom-ded)
 endif(UNIX)
 set_target_properties(vavoom-dedicated PROPERTIES COMPILE_FLAGS -DSERVER)
 target_link_libraries(vavoom-dedicated glbsp ${ZLIB_LIBRARY} ${NET_LIBRARIES})
@@ -899,18 +881,6 @@
 
 install(TARGETS vavoom-dedicated DESTINATION ${BINDIR})
 
-if(UNIX)
-	file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/vavoom-dedicated
-"#!/bin/sh
-# Needed to make symlinks/shortcuts work.
-# the binaries must run with correct working directory
-cd \"${CMAKE_INSTALL_PREFIX}/${DATADIR}\"
-\"${CMAKE_INSTALL_PREFIX}/${BINDIR}/vavoom-dedicated.i686\" $* ${IWADDIR}
-exit $?
-")
-	install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/vavoom-dedicated DESTINATION ${BINDIR})
-endif(UNIX)
-
 endif(ENABLE_SERVER)
 
 # # ---------------------------------------
