--- source/Makefile.am	2007-06-21 21:25:49.000000000 +0200
+++ source/Makefile.am	2007-06-21 21:27:27.000000000 +0200
@@ -19,12 +19,6 @@
 
 if ENABLE_CLIENT
 bin_PROGRAMS += ../$(MAIN_EXE)
-if MAIN_PLATFORM_UNIX_ALLEGRO
-bin_SCRIPTS += vavoom
-endif
-if MAIN_PLATFORM_UNIX_SDL
-bin_SCRIPTS += vavoom
-endif
 endif
 
 _____MAIN_EXE__CPPFLAGS = @MAIN_CPPFLAGS@
@@ -405,25 +399,8 @@
 	gl_win32.cpp
 endif
 
-vavoom: Makefile
-	echo "#!/bin/sh" > $@
-	echo "# Needed to make symlinks/shortcuts work." >> $@
-	echo "# the binaries must run with correct working directory" >> $@
-	echo "cd \"$(datadir)/vavoom\"" >> $@
-	echo "\"$(bindir)/$(MAIN_EXE)\" \$$* $(IWADDIR_PARM)" >> $@
-	echo "exit \$$?" >> $@
-
-#---------------------------------------
-#
-#	Dedicated server
-#
-#---------------------------------------
-
 if ENABLE_SERVER
 bin_PROGRAMS += ../$(SERVER_EXE)
-if SERVER_PLATFORM_UNIX
-bin_SCRIPTS += vavoom-dedicated
-endif
 endif
 
 _____SERVER_EXE__CPPFLAGS = @SV_CPPFLAGS@ -DSERVER
@@ -534,14 +511,6 @@
 	vavoom.rc
 endif
 
-vavoom-dedicated: Makefile
-	echo "#!/bin/sh" > $@
-	echo "# Needed to make symlinks/shortcuts work." >> $@
-	echo "# the binaries must run with correct working directory" >> $@
-	echo "cd \"$(datadir)/vavoom\"" >> $@
-	echo "\"$(bindir)/$(SERVER_EXE)\" \$$* $(IWADDIR_PARM)" >> $@
-	echo "exit \$$?" >> $@
-
 # ---------------------------------------
 #
 #	Compilation of Windows resources
--- source/Makefile.in	2007-06-21 21:25:59.000000000 +0200
+++ source/Makefile.in	2007-06-21 21:30:49.000000000 +0200
@@ -46,8 +46,6 @@
 #
 #---------------------------------------
 @ENABLE_CLIENT_TRUE@am__append_1 = ../$(MAIN_EXE)
-@ENABLE_CLIENT_TRUE@@MAIN_PLATFORM_UNIX_ALLEGRO_TRUE@am__append_2 = vavoom
-@ENABLE_CLIENT_TRUE@@MAIN_PLATFORM_UNIX_SDL_TRUE@am__append_3 = vavoom
 
 #-- i386 assembler files --
 @MAIN_ASM_I386_TRUE@am__append_4 = \
@@ -199,7 +197,6 @@
 #
 #---------------------------------------
 @ENABLE_SERVER_TRUE@am__append_21 = ../$(SERVER_EXE)
-@ENABLE_SERVER_TRUE@@SERVER_PLATFORM_UNIX_TRUE@am__append_22 = vavoom-dedicated
 @SERVER_PLATFORM_DJGPP_TRUE@am__append_23 = \
 @SERVER_PLATFORM_DJGPP_TRUE@	mplib.c \
 @SERVER_PLATFORM_DJGPP_TRUE@	mplpc.c \
@@ -828,7 +825,6 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = timidity
-bin_SCRIPTS = $(am__append_2) $(am__append_3) $(am__append_22)
 _____MAIN_EXE__CPPFLAGS = @MAIN_CPPFLAGS@
 _____MAIN_EXE__CXXFLAGS = @MAIN_CXXFLAGS@
 _____MAIN_EXE__DEFS = 
@@ -5184,22 +5180,6 @@
 	uninstall-dist_baseDATA uninstall-info-am
 
 
-vavoom: Makefile
-	echo "#!/bin/sh" > $@
-	echo "# Needed to make symlinks/shortcuts work." >> $@
-	echo "# the binaries must run with correct working directory" >> $@
-	echo "cd \"$(datadir)/vavoom\"" >> $@
-	echo "\"$(bindir)/$(MAIN_EXE)\" \$$* $(IWADDIR_PARM)" >> $@
-	echo "exit \$$?" >> $@
-
-vavoom-dedicated: Makefile
-	echo "#!/bin/sh" > $@
-	echo "# Needed to make symlinks/shortcuts work." >> $@
-	echo "# the binaries must run with correct working directory" >> $@
-	echo "cd \"$(datadir)/vavoom\"" >> $@
-	echo "\"$(bindir)/$(SERVER_EXE)\" \$$* $(IWADDIR_PARM)" >> $@
-	echo "exit \$$?" >> $@
-
 # ---------------------------------------
 #
 #	Compilation of Windows resources
