diff -Naur bombono-dvd-0.5.2.orig/debian/control bombono-dvd-0.5.2/debian/control
--- bombono-dvd-0.5.2.orig/debian/control	2009-11-01 22:04:07.000000000 +0300
+++ bombono-dvd-0.5.2/debian/control	2009-10-30 10:19:25.000000000 +0300
@@ -8,7 +8,7 @@
 
 Package: bombono-dvd
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, dvdauthor, mjpegtools, dvd+rw-tools, scons, toolame
+Depends: ${shlibs:Depends}, ${misc:Depends}, dvdauthor, mjpegtools, dvd+rw-tools, scons, twolame
 Suggests: totem
 Description: DVD authoring program with nice and clean GUI
  Bombono DVD is easy to use program for making DVD-Video.
diff -Naur bombono-dvd-0.5.2.orig/resources/scons_authoring/SConstruct bombono-dvd-0.5.2/resources/scons_authoring/SConstruct
--- bombono-dvd-0.5.2.orig/resources/scons_authoring/SConstruct	2009-11-01 22:04:07.000000000 +0300
+++ bombono-dvd-0.5.2/resources/scons_authoring/SConstruct	2009-10-28 13:08:34.000000000 +0300
@@ -16,23 +16,10 @@
 
 # звук тишины для неподвижных меню
 def GetkHzArgForToolame(kHz):
-    import os
-    import re
-    #s = "tooLAME version 0.2l"
-    s = os.popen("toolame -v").read()
-    #print s
-
-    m = re.compile("tooLAME version 0\.2(.)", re.M).search(s)
-
-    res = kHz
-    if m and m.group(1) == 'l':
-        pass # classic toolame
-    else:
-        res = res * 1000 # twolame or similar
-    return res
+    return kHz * 1000
 
 argHz = GetkHzArgForToolame(48)
-Command('Silent.mp2', [], "dd if=/dev/zero bs=4 count=1920 | toolame -b 128 -s %d /dev/stdin $TARGET" % argHz)
+Command('Silent.mp2', [], "dd if=/dev/zero bs=4 count=1920 | twolame -b 128 -s %d -r - $TARGET" % argHz)
 
 # основная функция создания меню
 import ADVD
diff -Naur bombono-dvd-0.5.2.orig/tools/scripts/bombono-dvd.spec bombono-dvd-0.5.2/tools/scripts/bombono-dvd.spec
--- bombono-dvd-0.5.2.orig/tools/scripts/bombono-dvd.spec	2009-10-08 13:25:07.000000000 +0400
+++ bombono-dvd-0.5.2/tools/scripts/bombono-dvd.spec	2009-11-01 22:06:40.000000000 +0300
@@ -16,7 +16,7 @@
 Packager:	Ilya Murav'jov <muravev@yandex.ru>
 Source:		%{name}-%{version}.tar.bz2
 BuildRequires:	scons libdvdread-devel gtkmm2-devel libxml++-devel mjpegtools-devel libGraphicsMagick++-devel
-Requires:       dvdauthor mjpegtools dvd+rw-tools scons toolame
+Requires:       dvdauthor mjpegtools dvd+rw-tools scons twolame
 BuildRoot:	%{_tmppath}/%{name}-%{version}-build
 
 # :TODO:
