--- bin/apply.pl.orig	2008-10-18 03:32:19.000000000 +0600
+++ bin/apply.pl	2008-10-25 21:31:42.000000000 +0600
@@ -767,7 +767,7 @@
     my $ver_line;
     my $is_old = 1;
 
-    open ($Patch, "gpatch --version|") || die "Can't run patch: $!";
+    open ($Patch, "patch --version|") || die "Can't run patch: $!";
     $ver_line = <$Patch>;
     $ver_line =~ m/\s+(\d+)\.(\d+)\.(\d+)/ || die "Can't get patch version\n";
     if ($1 >= 2 && $2 >= 5 && $3 >= 9) {
@@ -1118,7 +1118,7 @@
 
     $opts = join ' ', @arguments;
     my $base_args = " -l -p0 $opts -d $dest_dir";
-    $base_cmd = "gpatch";
+    $base_cmd = "patch";
     if ($pieces) {
 	# nasty path mess
 	$base_cmd = "$patch_dir/../../bin/sloppypatch.pl";
--- bin/enable-dict.orig	2008-10-18 11:48:50.000000000 +0600
+++ bin/enable-dict	2008-10-25 21:31:52.000000000 +0600
@@ -6,7 +6,7 @@
 
 test -f $ARC || exit 1
 cd $WRKDIR/dictionaries || exit 2
-gtar xjf $ARC
+tar xjf $ARC
 grep -q $LNG prj/build.lst && exit 0
 line="di      dictionaries\\$LNG      nmake   -       all     di_$LNG NULL"
 echo "$line" >> prj/build.lst
