diff -r -u spyview-2009-08-28-23_55.orig/spyview/spyview.C spyview-2009-08-28-23_55/spyview/spyview.C
--- spyview-2009-08-28-23_55.orig/spyview/spyview.C	2009-08-28 23:46:45.000000000 +0200
+++ spyview-2009-08-28-23_55/spyview/spyview.C	2009-09-10 09:46:58.000000000 +0200
@@ -127,7 +127,7 @@
    }
   for(subdirs_t::iterator i = subdirs.begin(); i != subdirs.end(); i++)
     {
-      if(*i == "./" || *i == "../")
+      if(*i == "./" || *i == "../" || *i == "") // the last case SHOULD not occur but does...
 	continue;
       find_cmaps(path+*i,pretty_path+*i);
     }
