diff -r -u spyview-2009-12-14-09_17.orig/spyview/Fl_Listbox.C spyview-2009-12-14-09_17/spyview/Fl_Listbox.C
--- spyview-2009-12-14-09_17.orig/spyview/Fl_Listbox.C	2008-09-05 16:23:20.000000000 +0200
+++ spyview-2009-12-14-09_17/spyview/Fl_Listbox.C	2010-01-05 15:54:46.000000000 +0100
@@ -1,4 +1,5 @@
 #include "Fl_Listbox.H"
+#include <stdio.h>
 
 Fl_Listbox::Fl_Listbox(int x, int y, int w, int h, const char *l) : Fl_Table(x,y,w,h,l) 
 { 
diff -r -u spyview-2009-12-14-09_17.orig/spyview/ImageData.H spyview-2009-12-14-09_17/spyview/ImageData.H
--- spyview-2009-12-14-09_17.orig/spyview/ImageData.H	2009-12-14 09:15:33.000000000 +0100
+++ spyview-2009-12-14-09_17/spyview/ImageData.H	2010-01-04 22:59:38.000000000 +0100
@@ -5,6 +5,7 @@
 #include <FL/Fl_Output.H>
 
 #include <string>
+#include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 #include <vector>
diff -r -u spyview-2009-12-14-09_17.orig/spyview/gp_to_mat.C spyview-2009-12-14-09_17/spyview/gp_to_mat.C
--- spyview-2009-12-14-09_17.orig/spyview/gp_to_mat.C	2009-08-28 23:46:45.000000000 +0200
+++ spyview-2009-12-14-09_17/spyview/gp_to_mat.C	2010-01-05 16:47:03.000000000 +0100
@@ -5,6 +5,7 @@
 #include <map>
 #include <math.h>
 #include <string.h>
+#include <stdio.h>
 #include "../config.h"
 
 using namespace std;
