diff -r -U 8 spyview-2009-09-10-00_02.orig/configure.ac spyview-2009-09-10-00_02/configure.ac
--- spyview-2009-09-10-00_02.orig/configure.ac	2009-09-09 13:20:18.000000000 +0200
+++ spyview-2009-09-10-00_02/configure.ac	2009-09-10 17:03:43.000000000 +0200
@@ -56,17 +56,17 @@
   AC_CHECK_LIB(wsock32,main,[],AC_MSG_ERROR([cannot find libwsock32]))
   AC_CHECK_LIB(uuid,main,[],AC_MSG_ERROR([cannot find libuuid]))
 fi
 
 AC_LANG_PUSH([C++])
 AC_CHECK_LIB([z], [main], , AC_MSG_ERROR([libz not found]))
 AC_CHECK_LIB([png], [main], , AC_MSG_ERROR([libpng not found]))
 AC_CHECK_LIB([fltk], [main], , AC_MSG_ERROR([libfltk not found]))
-AC_CHECK_LIB([fltk_jpeg], [main], , AC_MSG_ERROR([libz not found]))
+AC_CHECK_LIB([fltk_jpeg], [main], , AC_MSG_WARN([libfltk_jpeg not found - assuming you use system jpeg and continuing anyway]))
 AC_CHECK_LIB([fltk_images], [main], , AC_MSG_ERROR([libfltk_images not found]))
 AC_CHECK_LIB([fltk_gl], [main], ,AC_MSG_ERROR([libfltk_gl not found]))
 AC_CHECK_LIB([boost_serialization],[main],,AC_CHECK_LIB([boost_serialization-mt],[main],,AC_MSG_ERROR([libboost-serialization-dev library not found.])))
 AC_CHECK_LIB([boost_regex],[main],,AC_CHECK_LIB([boost_regex-mt],[main],,AC_MSG_ERROR([libboost-regex-dev library not found.])))
 AC_LANG_POP
 
 #AC_CHECK_LIB([gnuplot_i], [gnuplot_init], , AC_MSG_ERROR([libgnuplot_i not found]))
 #AC_CHECK_LIB([pgm], [main], , AC_MSG_ERROR([libpgm not found]))
