From 80ca7351a20286aff2006300a837933633a87447 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <gentoo@mgorny.alt.pl>
Date: Tue, 7 Sep 2010 22:33:11 +0200
Subject: [PATCH 1/6] Append miniupnpc-specific flags to CFLAGS.

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index ca59031..af257a8 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ OS = $(shell uname -s)
 CC ?= gcc
 #AR = gar
 #CFLAGS = -O -Wall -g -DDEBUG
-CFLAGS ?= -O -Wall -DNDEBUG -DMINIUPNPC_SET_SOCKET_TIMEOUT
+CFLAGS += -Wall -DNDEBUG -DMINIUPNPC_SET_SOCKET_TIMEOUT
 INSTALL = install
 SH = /bin/sh
 JAVA = java
-- 
1.7.2.2

