From bbd74061406e1972820c24982a39c4f1770d0ed5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <gentoo@mgorny.alt.pl>
Date: Tue, 7 Sep 2010 22:34:20 +0200
Subject: [PATCH 2/6] Respect LDFLAGS.

---
 Makefile |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index af257a8..427315c 100644
--- a/Makefile
+++ b/Makefile
@@ -141,16 +141,16 @@ $(LIBRARY):	$(LIBOBJS)
 
 $(SHAREDLIBRARY):	$(LIBOBJS)
 ifeq ($(OS), Darwin)
-	$(CC) -dynamiclib -Wl,-install_name,$(SONAME) -o $@ $^ 
+	$(CC) -dynamiclib $(LDFLAGS) -Wl,-install_name,$(SONAME) -o $@ $^ 
 else
-	$(CC) -shared -Wl,-soname,$(SONAME) -o $@ $^
+	$(CC) -shared $(LDFLAGS) -Wl,-soname,$(SONAME) -o $@ $^
 endif
 
 upnpc-static:	upnpc.o $(LIBRARY) $(LDLIBS)
-	$(CC) -o $@ $^
+	$(CC) $(LDFLAGS) -o $@ $^
 
 upnpc-shared:	upnpc.o $(SHAREDLIBRARY) $(LDLIBS)
-	$(CC) -o $@ $^
+	$(CC) $(LDFLAGS) -o $@ $^
 
 testminixml:	$(TESTMINIXMLOBJS)
 
-- 
1.7.2.2

