From 5aafd1a45c10dc9c06cf3655ec488339d292e58f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <gentoo@mgorny.alt.pl>
Date: Tue, 7 Sep 2010 22:50:37 +0200
Subject: [PATCH 4/6] Move minixml validation test to 'check' target.

---
 Makefile |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 199be58..85b9737 100644
--- a/Makefile
+++ b/Makefile
@@ -89,9 +89,12 @@ FILESTOINSTALL := $(FILESTOINSTALL) $(SHAREDLIBRARY)
 endif
 
 
-.PHONY:	install clean depend all everything installpythonmodule
+.PHONY:	install clean depend all check everything installpythonmodule \
+	validateminixml
 
-all:	validateminixml $(LIBRARY) $(EXECUTABLES)
+all:	$(LIBRARY) $(EXECUTABLES)
+
+check:	validateminixml
 
 everything:	all $(EXECUTABLES_ADDTESTS)
 
-- 
1.7.2.2

