From 332ad8565374f67e380f19168f8586bf57e01ff7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <gentoo@mgorny.alt.pl>
Date: Tue, 7 Sep 2010 23:05:49 +0200
Subject: [PATCH 6/6] Support disabling static library install through LIBRARY=.

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index a201c9b..b3007e5 100644
--- a/Makefile
+++ b/Makefile
@@ -123,7 +123,7 @@ install:	$(FILESTOINSTALL)
 	$(INSTALL) -d $(INSTALLDIRINC)
 	$(INSTALL) -m 644 $(HEADERS) $(INSTALLDIRINC)
 	$(INSTALL) -d $(INSTALLDIRLIB)
-	$(INSTALL) -m 644 $(LIBRARY) $(INSTALLDIRLIB)
+	[ -z "$(LIBRARY)" ] || $(INSTALL) -m 644 $(LIBRARY) $(INSTALLDIRLIB)
 ifneq ($(OS), AmigaOS)
 	$(INSTALL) -m 644 $(SHAREDLIBRARY) $(INSTALLDIRLIB)/$(SONAME)
 	ln -fs $(SONAME) $(INSTALLDIRLIB)/$(SHAREDLIBRARY)
-- 
1.7.2.2

