diff -dupr miniupnpc-1.4.orig/Makefile miniupnpc-1.4/Makefile
--- miniupnpc-1.4.orig/Makefile	2009-12-09 22:21:25.347370694 +0100
+++ miniupnpc-1.4/Makefile	2009-12-09 22:22:38.932621003 +0100
@@ -12,7 +12,7 @@ OS = $(shell uname -s)
 CC ?= gcc
 #AR = gar
 #CFLAGS = -fPIC -O -Wall -g -DDEBUG
-CFLAGS ?= -fPIC -O -Wall -DNDEBUG -DMINIUPNPC_SET_SOCKET_TIMEOUT
+CFLAGS += -fPIC -DNDEBUG -DMINIUPNPC_SET_SOCKET_TIMEOUT
 INSTALL = install
 SH = /bin/sh
 #following libs are needed on Solaris
