--- gengetopt-2.16~/tests/Makefile.am	2006-07-02 23:50:55.000000000 +0200
+++ gengetopt-2.16/tests/Makefile.am	2006-07-03 20:14:03.000000000 +0200
@@ -120,20 +120,20 @@
 test_redef_help_SOURCES = test_redef_help_cmd.ggo test_redef_help.c
 more_cmdparsers_SOURCES = more_cmdparsers.c test_all_opts_cmd.ggo test_conf_parser_cmd.ggoconf
 
-BUILT_SOURCES = no_strings_cmd.c no_strings_cmd.h \
-no_unamed_cmd.c no_unamed_cmd.h \
-no_optgiven_cmd.c no_optgiven_cmd.h \
-more_than_once_cmd.c more_than_once_cmd.h \
-canonize-names-cmd.c canonize-names-cmd.h \
-default_values_cmd.c default_values_cmd.h \
-cmdline.c cmdline.h \
-test_all_opts_cmd.c test_all_opts_cmd.h \
-test_groups_cmd.c test_groups_cmd.h \
-test_conf_parser_cmd.c test_conf_parser_cmd.h \
-test_multiple_cmd.c test_multiple_cmd.h \
+BUILT_SOURCES = no_strings_cmd.c \
+no_unamed_cmd.c \
+no_optgiven_cmd.c \
+more_than_once_cmd.c \
+canonize-names-cmd.c \
+default_values_cmd.c \
+cmdline.c \
+test_all_opts_cmd.c \
+test_groups_cmd.c \
+test_conf_parser_cmd.c \
+test_multiple_cmd.c \
 test_sections_cmd.c \
-test_values_cmd.c test_values_cmd.h \
-test_redef_help_cmd.c test_redef_help_cmd.h
+test_values_cmd.c \
+test_redef_help_cmd.c
 
 GENERATED_TESTS	= no_strings_cmd.c.test no_strings_cmd.h.test \
 no_unamed_cmd.c.test no_unamed_cmd.h.test \
