diff -ur orig/acpi-support-0.94/acpi-support work/acpi-support-0.94/acpi-support
--- orig/acpi-support-0.94/acpi-support	2006-09-15 12:47:55.000000000 -0400
+++ work/acpi-support-0.94/acpi-support	2007-04-12 20:00:17.000000000 -0400
@@ -67,3 +67,7 @@
 # Switch to laptop-mode on battery power - off by default as it causes odd
 # hangs on some machines
 ENABLE_LAPTOP_MODE=false
+
+# Disable wake-on-LAN for wired Ethernet devices. WOL is on by default on some
+# laptops. Disabling it can save power during sleep. (Requires ethtool)
+DISABLE_WOL=false
diff -ur orig/acpi-support-0.94/debian/init.d work/acpi-support-0.94/debian/init.d
--- orig/acpi-support-0.94/debian/init.d	2006-03-28 11:26:44.000000000 -0500
+++ work/acpi-support-0.94/debian/init.d	2007-04-12 10:24:25.000000000 -0400
@@ -1,39 +1,37 @@
-#!/bin/bash
+#!/sbin/runscript
 # INIT script to check whether we're on batteries, and so start with laptop 
 # mode etc enabled.
 
 # BUGS: unless we start *really* late, we have no way of throttling 
 # xscreensaver, since it won't be there to command.
-. /usr/share/acpi-support/power-funcs
-
-test -f /lib/lsb/init-functions || exit 1
-. /lib/lsb/init-functions
-
-test -d /var/lib/acpi-support || exit 0
-
-shopt -s nullglob
-
-case "$1" in
-  start)
-    log_begin_msg "Checking battery state..."
-    if [ `echo /proc/acpi/ac_adapter/*` ]; then 
-	/etc/acpi/power.sh
-    fi
-    # Source everything in /etc/acpi/start.d/
-    for SCRIPT in /etc/acpi/start.d/*.sh; do
-	. $SCRIPT
-    done
-    log_end_msg 0
-    ;;
-  stop)
-    log_begin_msg "Disabling power management..."
-    if [ `echo /proc/acpi/ac_adapter/*` ]; then
-	/etc/acpi/power.sh stop
-    fi
-    log_end_msg 0
-    ;;
-  *)
-  ;;
-esac
-        
 
+depend() {
+	need acpid
+}
+
+start() {
+	ebegin "Starting acpi-support"
+	. /usr/share/acpi-support/power-funcs
+
+	test -d /var/lib/acpi-support || eend 0
+
+	shopt -s nullglob
+
+	ebegin "Checking battery state..."
+	if [ `echo /proc/acpi/ac_adapter/*` ]; then
+		/etc/acpi/power.sh
+	fi
+	# Source everything in /etc/acpi/start.d/
+	for SCRIPT in /etc/acpi/start.d/*.sh; do
+		. $SCRIPT
+	done
+	eend 0
+}
+
+stop() {
+	ebegin "Disabling power management..."
+	if [ `echo /proc/acpi/ac_adapter/*` ]; then
+		/etc/acpi/power.sh stop
+	fi
+	eend 0
+}
diff -ur orig/acpi-support-0.94/hibernatebtn.sh work/acpi-support-0.94/hibernatebtn.sh
--- orig/acpi-support-0.94/hibernatebtn.sh	2006-02-06 09:54:47.000000000 -0500
+++ work/acpi-support-0.94/hibernatebtn.sh	2007-04-12 10:25:25.000000000 -0400
@@ -1,3 +1,3 @@
 #!/bin/bash
-. /usr/share/acpi-support/key-constants
-acpi_fakekey $KEY_SUSPEND 
+
+. /etc/acpi/hibernate.sh
diff -ur orig/acpi-support-0.94/lib/policy-funcs work/acpi-support-0.94/lib/policy-funcs
--- orig/acpi-support-0.94/lib/policy-funcs	2006-05-19 16:21:58.000000000 -0400
+++ work/acpi-support-0.94/lib/policy-funcs	2007-04-12 10:27:15.000000000 -0400
@@ -1,6 +1,6 @@
 CheckPolicy() {
 	if pidof gnome-power-manager kpowersave > /dev/null ||
-	   (pidof dcopserver > /dev/null && test -x /usr/bin/dcop && /usr/bin/dcop kded kded loadedModules | grep -q klaptopdaemon) ; then
+	   (pidof dcopserver > /dev/null && test -x /usr/kde/3.5/bin/dcop && /usr/kde/3.5/bin/dcop kded kded loadedModules | grep -q klaptopdaemon) ; then
 		echo 0;
 	else
 		echo 1;
diff -ur orig/acpi-support-0.94/lib/power-funcs work/acpi-support-0.94/lib/power-funcs
--- orig/acpi-support-0.94/lib/power-funcs	2005-11-29 12:50:02.000000000 -0500
+++ work/acpi-support-0.94/lib/power-funcs	2007-04-12 10:27:51.000000000 -0400
@@ -6,9 +6,9 @@
 POWERSTATE="/var/lib/acpi-support/powerstate"
 
 getXuser() {
-        user=`finger| grep -m1 ":$displaynum " | awk '{print $1}'`
+        user=`who| grep -m1 ":$displaynum " | awk '{print $1}'`
 	if [ x"$user" = x"" ]; then
-		user=`finger| grep -m1 ":$displaynum" | awk '{print $1}'`
+		user=`who| grep -m1 ":$displaynum" | awk '{print $1}'`
 	fi
 	if [ x"$user" != x"" ]; then
         	userhome=`getent passwd $user | cut -d: -f6`
diff -ur orig/acpi-support-0.94/resume.d/13-855-resolution-set.sh work/acpi-support-0.94/resume.d/13-855-resolution-set.sh
--- orig/acpi-support-0.94/resume.d/13-855-resolution-set.sh	2005-09-26 20:29:27.000000000 -0400
+++ work/acpi-support-0.94/resume.d/13-855-resolution-set.sh	2007-04-12 19:34:31.000000000 -0400
@@ -1,7 +1,6 @@
 #!/bin/bash
 
 if [ -x /usr/sbin/855resolution ]; then
-    . /etc/default/855resolution
     if [ "$MODE" != "" ] && [ "$XRESO" != "" ] && [ "$YRESO" != "" ]; then
 	/etc/init.d/855resolution start;
     fi
diff -ur orig/acpi-support-0.94/resume.d/40-infra-red.sh work/acpi-support-0.94/resume.d/40-infra-red.sh
--- orig/acpi-support-0.94/resume.d/40-infra-red.sh	2006-04-21 10:12:54.000000000 -0400
+++ work/acpi-support-0.94/resume.d/40-infra-red.sh	2007-04-12 10:28:33.000000000 -0400
@@ -3,7 +3,6 @@
 # Restart IR if necessary
 if [ -f /var/run/irdadev ] && [ x$RESTART_IRDA = xtrue ]; then
     rm /var/run/irdadev;
-    /etc/init.d/irda-setup start;
-    /etc/init.d/irda-utils start;
+    /etc/init.d/irda start;
 fi;
 
diff -ur orig/acpi-support-0.94/resume.d/49-855-resolution-set.sh work/acpi-support-0.94/resume.d/49-855-resolution-set.sh
--- orig/acpi-support-0.94/resume.d/49-855-resolution-set.sh	2005-09-26 20:29:27.000000000 -0400
+++ work/acpi-support-0.94/resume.d/49-855-resolution-set.sh	2007-04-12 19:34:31.000000000 -0400
@@ -1,7 +1,6 @@
 #!/bin/bash
 
 if [ -x /usr/sbin/855resolution ]; then
-    . /etc/default/855resolution
     if [ "$MODE" != "" ] && [ "$XRESO" != "" ] && [ "$YRESO" != "" ]; then
 	/etc/init.d/855resolution start;
     fi
diff -ur orig/acpi-support-0.94/resume.d/62-ifup.sh work/acpi-support-0.94/resume.d/62-ifup.sh
--- orig/acpi-support-0.94/resume.d/62-ifup.sh	2006-04-21 10:13:34.000000000 -0400
+++ work/acpi-support-0.94/resume.d/62-ifup.sh	2007-04-12 10:29:11.000000000 -0400
@@ -3,6 +3,6 @@
 # Bring up the interfaces (this should probably be left up to some policy
 # manager, but at the moment we just bring back whatever we ifdowned)
 for x in $INTERFACES; do
-    ifup $x &
+    /etc/init.d/net.$x start &
 done
 
diff -ur orig/acpi-support-0.94/sleepbtn.sh work/acpi-support-0.94/sleepbtn.sh
--- orig/acpi-support-0.94/sleepbtn.sh	2006-02-06 09:55:00.000000000 -0500
+++ work/acpi-support-0.94/sleepbtn.sh	2007-04-12 10:29:48.000000000 -0400
@@ -1,3 +1,3 @@
 #!/bin/bash
-. /usr/share/acpi-support/key-constants
-acpi_fakekey $KEY_SLEEP 
+
+. /etc/acpi/sleep.sh
diff -ur orig/acpi-support-0.94/suspend.d/50-irda-stop.sh work/acpi-support-0.94/suspend.d/50-irda-stop.sh
--- orig/acpi-support-0.94/suspend.d/50-irda-stop.sh	2006-04-21 11:05:47.000000000 -0400
+++ work/acpi-support-0.94/suspend.d/50-irda-stop.sh	2007-04-12 10:30:31.000000000 -0400
@@ -2,7 +2,7 @@
 
 # Stop IRDA if it's running
 if [ -f /var/run/irattach.pid ]; then
-    /etc/init.d/irda-utils stop
+    /etc/init.d/irda stop
     killall -9 irattach
 fi
 
Only in work/acpi-support-0.94/suspend.d: 52-disable-wol.sh
diff -ur orig/acpi-support-0.94/suspend.d/55-down-interfaces.sh work/acpi-support-0.94/suspend.d/55-down-interfaces.sh
--- orig/acpi-support-0.94/suspend.d/55-down-interfaces.sh	2006-05-16 07:57:43.000000000 -0400
+++ work/acpi-support-0.94/suspend.d/55-down-interfaces.sh	2007-04-12 10:30:52.000000000 -0400
@@ -10,7 +10,7 @@
 
 # And shut them down
 for x in $INTERFACES; do
-    ifdown $x;
+    /etc/init.d/net.$x stop;
     ifconfig $x down;
 done
 
diff -ur orig/acpi-support-0.94/vbesave work/acpi-support-0.94/vbesave
--- orig/acpi-support-0.94/vbesave	2006-05-03 16:31:51.000000000 -0400
+++ work/acpi-support-0.94/vbesave	2007-04-12 10:31:29.000000000 -0400
@@ -3,8 +3,6 @@
 test -x /usr/sbin/vbetool || exit 0
 set -e
 
-. /lib/lsb/init-functions
-
 test -f /etc/default/rcS && . /etc/default/rcS
 test -f /etc/default/acpi-support && . /etc/default/acpi-support
 test -f /usr/share/acpi-support/device-funcs && . /usr/share/acpi-support/device-funcs
@@ -20,20 +18,16 @@
 	fi
 
         DeviceConfig
-	log_begin_msg "Saving VESA state..."
 	if [ "$SAVE_VBE_STATE" = "true" ]; then
 	    if [ "$VERBOSE" = no ]; then 
 		if ! vbetool vbestate save > $VBESTATE 2>/dev/null; then
-		    log_end_msg $?
 		    exit 1
 		fi
 	    else
 		if ! vbetool vbestate save > $VBESTATE ; then
-		    log_end_msg $?
 		    exit 1
 		fi
 	    fi
-	    log_end_msg 0
         fi
         ;;
  *)
