--- dictionaries/ru_RU/dictionary.lst.orig	2008-10-28 20:45:36.000000000 +0500
+++ dictionaries/ru_RU/dictionary.lst	2008-10-28 20:45:43.000000000 +0500
@@ -1,2 +1,3 @@
+DICT ru RU ru_RU
 HYPH ru RU hyph_ru_RU
 THES ru RU th_ru_RU_v2
--- dictionaries/ru_RU/description.xml.orig	2008-10-28 20:45:05.000000000 +0500
+++ dictionaries/ru_RU/description.xml	2008-10-28 20:47:10.000000000 +0500
@@ -1,13 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <description xmlns="http://openoffice.org/extensions/description/2006" xmlns:d="http://openoffice.org/extensions/description/2006"  xmlns:xlink="http://www.w3.org/1999/xlink">
-    <version value="2008.07.01" />
     <identifier value="org.openoffice.ru.hunspell.dictionaries" />
-    <display-name>
-        <name lang="en">Russian hyphenation dictionary and thesaurus</name>
-    </display-name>
-    <platform value="all" />
+    <version value="0.4" />
     <dependencies>
-        <OpenOffice.org-minimal-version value="3.0" d:name="OpenOffice.org 3.0" />
+        <OpenOffice.org-minimal-version value="3.0" d:name="Russian Dictionaries for OpenOffice.org 3.0"/>
     </dependencies>
+    <registration>
+    </registration>
+    <display-name>
+      <name lang="ru">Словари для русского языка</name>
+      <name lang="en">Russian Dictionaries</name>
+    </display-name>
 </description>
-
--- dictionaries/ru_RU/dictionaries.xcu.orig	2008-10-28 20:45:30.000000000 +0500
+++ dictionaries/ru_RU/dictionaries.xcu	2008-10-28 20:47:40.000000000 +0500
@@ -2,6 +2,17 @@
 <oor:component-data xmlns:oor="http://openoffice.org/2001/registry" xmlns:xs="http://www.w3.org/2001/XMLSchema" oor:name="Linguistic" oor:package="org.openoffice.Office">
  <node oor:name="ServiceManager">
     <node oor:name="Dictionaries">
+        <node oor:name="HunSpellDic_ru-RU" oor:op="fuse">
+            <prop oor:name="Locations" oor:type="oor:string-list">
+                <value>%origin%/ru_RU.aff %origin%/ru_RU.dic</value>
+            </prop>
+            <prop oor:name="Format" oor:type="xs:string">
+                <value>DICT_SPELL</value>
+            </prop>
+            <prop oor:name="Locales" oor:type="oor:string-list">
+                <value>ru-RU</value>
+            </prop>
+        </node>
         <node oor:name="HyphDic_ru-RU" oor:op="fuse">
             <prop oor:name="Locations" oor:type="oor:string-list">
                 <value>%origin%/hyph_ru_RU.dic</value>
--- dictionaries/ru_RU/makefile.mk.orig	2008-08-15 16:41:32.000000000 +0600
+++ dictionaries/ru_RU/makefile.mk	2008-10-28 20:40:59.000000000 +0500
@@ -53,6 +53,9 @@
 
 # just copy:
 COMPONENT_FILES= \
+    $(EXTENSIONDIR)$/ru_RU.aff \
+    $(EXTENSIONDIR)$/ru_RU.dic \
+    $(EXTENSIONDIR)$/README_thes_ru_RU.txt \
     $(EXTENSIONDIR)$/hyph_ru_RU.dic
 
 COMPONENT_CONFIGDEST=.
@@ -60,7 +63,7 @@
     $(EXTENSIONDIR)$/dictionaries.xcu
 
 # disable fetching default OOo license text
-CUSTOM_LICENSE=README_thes_ru_RU.txt
+CUSTOM_LICENSE=README_ru_RU.txt
 # override default license destination
 PACKLICS= $(EXTENSIONDIR)$/$(CUSTOM_LICENSE)
 
