diff -ur a/makefile b/makefile
--- a/makefile	2008-06-07 18:04:48.000000000 -0400
+++ b/makefile	2009-12-31 19:04:15.462990041 -0500
@@ -31,10 +31,10 @@
 # cure the problem, change this file from using PC-style end-of-line (carriage 
 # return and line feed) to unix style end-of-line (line feed).
 
-CC=gcc
+CC?=gcc
 #CFLAGS=-DANSILIBS -O2 
-CFLAGS=-DANSILIBS -g 
-LNK=gcc
+CFLAGS+=-DANSILIBS 
+LNK?=gcc
 INSTALL=install
 
 prefix=/usr/local
@@ -48,26 +48,26 @@
 
 abc2midi : parseabc.o store.o genmidi.o midifile.o queues.o parser2.o
 	$(LNK) -o abc2midi parseabc.o store.o genmidi.o queues.o \
-	parser2.o midifile.o
+	parser2.o midifile.o $(LDFLAGS)
 
 abc2abc : parseabc.o toabc.o
-	$(LNK) -o abc2abc parseabc.o toabc.o
+	$(LNK) -o abc2abc parseabc.o toabc.o $(LDFLAGS)
 
 midi2abc : midifile.o midi2abc.o 
-	$(LNK) midifile.o midi2abc.o -o midi2abc
+	$(LNK) midifile.o midi2abc.o -o midi2abc $(LDFLAGS)
 
 mftext : midifile.o mftext.o crack.o
-	$(LNK) midifile.o mftext.o crack.o -o mftext
+	$(LNK) midifile.o mftext.o crack.o -o mftext $(LDFLAGS)
 
 yaps : parseabc.o yapstree.o drawtune.o debug.o pslib.o position.o parser2.o
 	$(LNK) -o yaps parseabc.o yapstree.o drawtune.o debug.o \
-	position.o pslib.o parser2.o -o yaps
+	position.o pslib.o parser2.o -o yaps $(LDFLAGS)
 
 midicopy : midicopy.o
-	$(LNK) -o midicopy midicopy.o
+	$(LNK) -o midicopy midicopy.o $(LDFLAGS)
 
 abcmatch : abcmatch.o matchsup.o parseabc.o
-	$(LNK) abcmatch.o matchsup.o parseabc.o -o abcmatch
+	$(LNK) abcmatch.o matchsup.o parseabc.o -o abcmatch $(LDFLAGS)
 
 parseabc.o : parseabc.c abc.h parseabc.h
 
diff -ur a/makefiles/unix.mak b/makefiles/unix.mak
--- a/makefiles/unix.mak	2008-05-30 19:22:22.000000000 -0400
+++ b/makefiles/unix.mak	2009-12-31 19:04:28.373020771 -0500
@@ -31,9 +31,9 @@
 # cure the problem, change this file from using PC-style end-of-line (carriage 
 # return and line feed) to unix style end-of-line (line feed).
 
-CC=gcc
-CFLAGS=-DANSILIBS -O2 
-LNK=gcc
+CC?=gcc
+CFLAGS+=-DANSILIBS 
+LNK?=gcc
 INSTALL=install
 
 prefix=/usr/local
@@ -47,26 +47,26 @@
 
 abc2midi : parseabc.o store.o genmidi.o midifile.o queues.o parser2.o
 	$(LNK) -o abc2midi parseabc.o store.o genmidi.o queues.o \
-	parser2.o midifile.o
+	parser2.o midifile.o $(LDFLAGS)
 
 abc2abc : parseabc.o toabc.o
-	$(LNK) -o abc2abc parseabc.o toabc.o
+	$(LNK) -o abc2abc parseabc.o toabc.o $(LDFLAGS)
 
 midi2abc : midifile.o midi2abc.o 
-	$(LNK) midifile.o midi2abc.o -o midi2abc
+	$(LNK) midifile.o midi2abc.o -o midi2abc $(LDFLAGS)
 
 mftext : midifile.o mftext.o crack.o
-	$(LNK) midifile.o mftext.o crack.o -o mftext
+	$(LNK) midifile.o mftext.o crack.o -o mftext $(LDFLAGS)
 
 yaps : parseabc.o yapstree.o drawtune.o debug.o pslib.o position.o parser2.o
 	$(LNK) -o yaps parseabc.o yapstree.o drawtune.o debug.o \
-	position.o pslib.o parser2.o -o yaps
+	position.o pslib.o parser2.o -o yaps $(LDFLAGS)
 
 midicopy : midicopy.o
-	$(LNK) -o midicopy midicopy.o
+	$(LNK) -o midicopy midicopy.o $(LDFLAGS)
 
 abcmatch : abcmatch.o matchsup.o parseabc.o
-	$(LNK) abcmatch.o matchsup.o parseabc.o -o abcmatch
+	$(LNK) abcmatch.o matchsup.o parseabc.o -o abcmatch $(LDFLAGS)
 
 parseabc.o : parseabc.c abc.h parseabc.h
 
