diff -ru mugshot-1.1.22-orig/Makefile-mugshot.am mugshot-1.1.22/Makefile-mugshot.am
--- mugshot-1.1.22-orig/Makefile-mugshot.am	2006-10-25 17:04:17.486420616 -0400
+++ mugshot-1.1.22/Makefile-mugshot.am	2006-10-25 18:37:07.683621928 -0400
@@ -6,8 +6,8 @@
 	$(MUGSHOT_CFLAGS)		\
 	-I$(top_srcdir)/canvas
 
-mugshot_LDFLAGS=$(MUGSHOT_LIBS)
-mugshot_LDADD= libhippoipc.la libhippo.la
+mugshot_LDFLAGS=
+mugshot_LDADD= libhippoipc.la libhippo.la $(MUGSHOT_LIBS)
 
 if !HAVE_GTK210
 mugshot_CPPFLAGS += -I$(top_srcdir)/gtkcopy
diff -ru mugshot-1.1.22-orig/Makefile-mugshot-uri-handler.am mugshot-1.1.22/Makefile-mugshot-uri-handler.am
--- mugshot-1.1.22-orig/Makefile-mugshot-uri-handler.am	2006-10-25 17:04:17.492419704 -0400
+++ mugshot-1.1.22/Makefile-mugshot-uri-handler.am	2006-10-25 18:37:24.554057232 -0400
@@ -4,8 +4,8 @@
 	$(AM_CPPFLAGS)			\
 	$(MUGSHOT_URI_HANDLER_CFLAGS)
 
-mugshot_uri_handler_LDFLAGS=$(MUGSHOT_URI_HANDLER_LIBS)
-mugshot_uri_handler_LDADD= libhippo.la
+mugshot_uri_handler_LDFLAGS=
+mugshot_uri_handler_LDADD= libhippo.la $(MUGSHOT_URI_HANDLER_LIBS)
 
 mugshot_uri_handler_SOURCES=			\
 	src/hippo-dbus-client.c			\
diff -ru mugshot-1.1.22-orig/Makefile-test-hippo-ipc.am mugshot-1.1.22/Makefile-test-hippo-ipc.am
--- mugshot-1.1.22-orig/Makefile-test-hippo-ipc.am	2006-10-25 17:04:17.509417120 -0400
+++ mugshot-1.1.22/Makefile-test-hippo-ipc.am	2006-10-25 18:37:37.675062536 -0400
@@ -4,8 +4,8 @@
 	$(AM_CPPFLAGS)			\
 	$(TEST_HIPPO_IPC_CFLAGS)
 
-test_hippo_ipc_LDFLAGS=$(TEST_HIPPO_IPC_LIBS)
-test_hippo_ipc_LDADD= libhippoipc.la
+test_hippo_ipc_LDFLAGS=
+test_hippo_ipc_LDADD= libhippoipc.la $(TEST_HIPPO_IPC_LIBS)
 
 test_hippo_ipc_SOURCES=				\
 	hippoipc/test-hippo-ipc.cpp
