diff -ru mugshot-1.1.46_orig/Makefile-mugshot.am mugshot-1.1.46_before_libxpcom/Makefile-mugshot.am
--- mugshot-1.1.46_orig/Makefile-mugshot.am	2007-06-26 12:26:53.000000000 -0400
+++ mugshot-1.1.46_before_libxpcom/Makefile-mugshot.am	2007-09-01 18:36:22.000000000 -0400
@@ -7,8 +7,8 @@
 	$(CANVAS_CFLAGS)		\
        -DHOST_CPU=\"$(host_cpu)\"
 
-mugshot_LDFLAGS=$(MUGSHOT_LIBS)
-mugshot_LDADD= libhippoipc.la libhippo.la $(CANVAS_LIBADD)
+mugshot_LDFLAGS=
+mugshot_LDADD= libhippoipc.la libhippo.la $(MUGSHOT_LIBS) $(CANVAS_LIBADD)
 
 if !HAVE_GTK210
 mugshot_CPPFLAGS += -I$(top_srcdir)/gtkcopy
diff -ru mugshot-1.1.46_orig/Makefile-mugshot-uri-handler.am mugshot-1.1.46_before_libxpcom/Makefile-mugshot-uri-handler.am
--- mugshot-1.1.46_orig/Makefile-mugshot-uri-handler.am	2006-08-16 13:06:12.000000000 -0400
+++ mugshot-1.1.46_before_libxpcom/Makefile-mugshot-uri-handler.am	2007-09-01 18:36:53.000000000 -0400
@@ -4,8 +4,8 @@
 	$(AM_CPPFLAGS)			\
 	$(MUGSHOT_URI_HANDLER_CFLAGS)
 
-mugshot_uri_handler_LDFLAGS=$(MUGSHOT_URI_HANDLER_LIBS)
-mugshot_uri_handler_LDADD= libhippo.la
+mugshot_uri_handler_LDFLAGS=
+mugshot_uri_handler_LDADD= libhippo.la $(MUGSHOT_URI_HANDLER_LIBS)
 
 mugshot_uri_handler_SOURCES=			\
 	src/hippo-dbus-client.c			\
diff -ru mugshot-1.1.46_orig/Makefile-test-hippo-ipc.am mugshot-1.1.46_before_libxpcom/Makefile-test-hippo-ipc.am
--- mugshot-1.1.46_orig/Makefile-test-hippo-ipc.am	2007-01-11 15:48:14.000000000 -0500
+++ mugshot-1.1.46_before_libxpcom/Makefile-test-hippo-ipc.am	2007-09-01 18:37:34.000000000 -0400
@@ -5,8 +5,8 @@
 	-I $(COMMONSRCDIR)/hippoipc	\
 	$(TEST_HIPPO_IPC_CFLAGS)
 
-test_hippo_ipc_LDFLAGS=$(TEST_HIPPO_IPC_LIBS)
-test_hippo_ipc_LDADD= libhippoipc.la
+test_hippo_ipc_LDFLAGS=
+test_hippo_ipc_LDADD= libhippoipc.la $(TEST_HIPPO_IPC_LIBS)
 
 test_hippo_ipc_SOURCES=				\
 	hippoipc/test-hippo-ipc.cpp
