diff -ru mugshot-1.1.93.libxpcom/configure.ac mugshot-1.1.93/configure.ac
--- mugshot-1.1.93.libxpcom/configure.ac	2008-04-04 15:22:27.603493000 -0400
+++ mugshot-1.1.93/configure.ac	2008-04-04 15:31:37.696493211 -0400
@@ -410,6 +410,16 @@
 ##
 ## Checks for Firefox build environment
 ##
+AC_ARG_ENABLE(firefox,
+       AC_HELP_STRING([--enable-firefox],
+                      [Build a Firefox extension [[default=yes]]]),
+       [case "${enableval}" in
+         yes) firefox=true ;;
+         no)  firefox=false ;;
+         *)   AC_MSG_ERROR(bad value ${enableval} for --enable-firefox) ;;
+        esac],[firefox=true])
+AM_CONDITIONAL(USE_FIREFOX, test "x$firefox" = xtrue)
+
 AC_ARG_WITH(gecko_sdk, 
 	AC_HELP_STRING([--with-gecko-sdk=DIR],
 		       [Path to Gecko SDK]))
@@ -429,6 +439,8 @@
 sdk_includes_xpcom_core=false
 sdk_includes_xpcom=false
 
+if test "x$firefox" = xtrue ; then
+
 if test "x$with_gecko_sdk" != x -a "x$with_gecko_sdk" != xyes -a "x$with_gecko_sdk" != xno ; then
    GECKO_IDLDIR="$with_gecko_sdk/idl"
    GECKO_INCLUDEDIR="$with_gecko_sdk/include"
@@ -646,6 +658,12 @@
   LIBHIPPOFIREFOX_LIBS="${LIBHIPPOFIREFOX_LIBS} `$PKG_CONFIG --libs $xulrunner_package`"
 fi
 
+else
+    AM_CONDITIONAL(HAVE_FULL_GECKO, false)
+    AM_CONDITIONAL(USE_XULRUNNER, false)
+fi ### test "x$firefox" = xtrue
+
+
 #### define absolute path to srcdir for debugging-only code
 ABSOLUTE_TOP_SRCDIR=`cd ${srcdir} && pwd`
 AC_DEFINE_UNQUOTED(ABSOLUTE_TOP_SRCDIR, "$ABSOLUTE_TOP_SRCDIR", [full path to srcdir])
@@ -669,6 +687,7 @@
 
         prefix:                   ${prefix}
         compiler:                 ${CC}
+	build Firefox extension:  ${firefox}
 	xpidl:			  ${XPIDL}
 	Gecko includes:		  ${GECKO_INCLUDEDIR}
 	Gecko idl files:	  ${GECKO_IDLDIR}
diff -ru mugshot-1.1.93.libxpcom/Makefile.am mugshot-1.1.93/Makefile.am
--- mugshot-1.1.93.libxpcom/Makefile.am	2008-03-26 15:02:48.000000000 -0400
+++ mugshot-1.1.93/Makefile.am	2008-04-04 15:24:22.154493070 -0400
@@ -34,8 +34,13 @@
 	  rmdir $(BUILT_SRC_DIR) ;		\
 	fi
 
+if USE_FIREFOX
 dist-hook: libhippo-dist-hook libhippoipc-dist-hook libhippofirefox-dist-hook mugshot-dist-hook libstacker-dist-hook
 .PHONY: libhippo-dist-hook libhippoipc-dist-hook libhippofirefox-dist-hook mugshot-dist-hook libstacker-dist-hook
+else
+dist-hook: libhippo-dist-hook libhippoipc-dist-hook mugshot-dist-hook libstacker-dist-hook
+.PHONY: libhippo-dist-hook libhippoipc-dist-hook  mugshot-dist-hook libstacker-dist-hook
+endif
 
 # These are added to using += in the included files
 bin_PROGRAMS = 
diff -ru mugshot-1.1.93.libxpcom/Makefile-libhippofirefox.am mugshot-1.1.93/Makefile-libhippofirefox.am
--- mugshot-1.1.93.libxpcom/Makefile-libhippofirefox.am	2008-04-01 10:56:52.000000000 -0400
+++ mugshot-1.1.93/Makefile-libhippofirefox.am	2008-04-04 15:24:22.154493070 -0400
@@ -1,3 +1,4 @@
+if USE_FIREFOX
 FIREFOX_SRCDIR = $(COMMONSRCDIR)/firefox
 GECKO_PRIVATE_INCLUDEDIR=$(FIREFOX_SRCDIR)/gecko-private/include
 EXTENSION_ID = firefox@mugshot.org
@@ -198,3 +199,4 @@
 	cp $(GECKO_PRIVATE_IDL_FILES) $(distdir)/common-dist/firefox/gecko-private/idl
 	mkdir $(distdir)/common-dist/firefox/gecko-private/include
 	cp $(GECKO_PRIVATE_INCLUDE_FILES) $(distdir)/common-dist/firefox/gecko-private/include
+endif
