--- udept-0.5.96.2/Makefile.orig	2006-08-12 08:26:37.000000000 +0200
+++ udept-0.5.96.2/Makefile	2006-08-12 08:34:19.000000000 +0200
@@ -20,8 +20,7 @@
 	$(INSTALL) dep $(DESTDIR)/usr/bin/dep
 	$(INSTALL) -d $(DESTDIR)/usr/share/man/man1
 	$(INSTALL) dep.1 $(DESTDIR)/usr/share/man/man1/dep.1
-	$(INSTALL) -d $(DESTDIR)/usr/share/bash-completion
-	$(INSTALL) dep.completion $(DESTDIR)/usr/share/bash-completion/dep
+	if [[ " $(USE) " == "  " ]] || [[ " $(USE) " == *" bash-completion "* ]] ; then $(INSTALL) -d $(DESTDIR)/usr/share/bash-completion; $(INSTALL) dep.completion $(DESTDIR)/usr/share/bash-completion/dep; fi
 
 clean:
 	$(RM) -f $(CLEANFILES)
