diff -Naur xml-security-c-1.5.0_pre20090310.orig/include/Makefile.am xml-security-c-1.5.0_pre20090310/include/Makefile.am
--- xml-security-c-1.5.0_pre20090310.orig/include/Makefile.am	2009-03-10 08:23:06.552988680 +0100
+++ xml-security-c-1.5.0_pre20090310/include/Makefile.am	2009-03-10 08:24:06.360010865 +0100
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-BUILT_SOURCES = xsec ${lib_includes}
+BUILT_SOURCES = ${lib_includes}
 nobase_include_HEADERS = ${lib_includes}
 
 lib_includes = 
@@ -213,7 +213,7 @@
 
 SUFFIXES = .hpp
 
-xsec/%.hpp: ../src/%.hpp
+xsec/%.hpp: xsec ../src/%.hpp
 	cp ../src/$*.hpp $@
 
 xsec:
