--- winetools-0.9jo-III/wt0.9jo	2005-12-12 11:47:00.000000000 +0100
+++ winetools-0.9jo-III.ebuild/wt0.9jo	2006-08-18 21:38:42.000000000 +0200
@@ -11,9 +11,9 @@
 VERSION="winetools-0.9"
 WINESERVER="wineserver"
 RELEASE="`echo $VERSION | sed -e "s/winetools-\(.*\)/\1/"`"
-BASEDIR="/usr/local/winetools"
+BASEDIR="/usr/winetools"
 BINDIR="$HOME/bin"
-DIALOG="$BASEDIR/Xdialog"
+DIALOG="/usr/bin/Xdialog"
 WINEDIR="`echo ~/.wine`"
 CONFIG="$WINEDIR/winetools.log"
 SYSDIR="$HOME/winetools"
