--- winetools-0.9jo-III/wt0.9jo	2005-12-12 11:47:00.000000000 +0100
+++ winetools-0.9jo-III.ebuild/wt0.9jo	2006-08-22 21:22:17.000000000 +0200
@@ -11,7 +11,8 @@
 VERSION="winetools-0.9"
 WINESERVER="wineserver"
 RELEASE="`echo $VERSION | sed -e "s/winetools-\(.*\)/\1/"`"
-BASEDIR="/usr/local/winetools"
+BASEDIR="/usr/winetools"
+DOCDIR="/usr/share/doc/winetools-0.9"
 BINDIR="$HOME/bin"
 DIALOG="$BASEDIR/Xdialog"
 WINEDIR="`echo ~/.wine`"
@@ -24,11 +25,6 @@ CDROM="`ls -l "$WINEDIR/dosdevices/" 2>/
 #[ "$SFMIRROR" = "" ] && SFMIRROR="http://mesh.dl.sourceforge.net/sourceforge"
 [ "$SFMIRROR" = "" ] && SFMIRROR="http://puzzle.dl.sourceforge.net/sourceforge"
 
-#for Debian:
-#BASEDIR="/usr/share/winetools"
-#DIALOG="/usr/bin/Xdialog"
-#WINESERVER="/usr/lib/wine/wineserver"
-
 # to generate the pot template use "xgettext -L Shell -o po/wt0.9.pot  wt0.9jo findwine"
 # to generate a language template for your language call "msginit" in the directory po
 
@@ -3234,10 +3230,10 @@ Note:	This software comes with absolutel
 function readme()
 {
     TMP=`mktemp ${TMPDIR:-/tmp}/README.XXXXXX` || mktemp_error "$?"
-    if [ -e "${BASEDIR}/doc/README.${WINEVER}" ]; then
-      cat ${BASEDIR}/doc/README.first ${BASEDIR}/doc/README.${WINEVER} > $TMP
+    if [ -e "${DOCDIR}/README.${WINEVER}" ]; then
+      zcat ${DOCDIR}/README.first ${DOCDIR}/README.${WINEVER} > $TMP
     else
-      cat ${BASEDIR}/doc/README.first ${BASEDIR}/doc/README.nover > $TMP
+      zcat ${DOCDIR}/README.first ${DOCDIR}/README.nover > $TMP
     fi
     $DIALOG --title "`eval_gettext "Intro"`" --fixed-font "" \
 	--no-cancel \
