--- Makefile	2007-07-29 14:09:34.000000000 +0200
+++ Makefile.new	2009-04-10 21:25:34.000000000 +0200
@@ -8,7 +8,7 @@
 OBJ = ${SRC:.c=.o}
 OBJ_test = ${SRC_test:.c=.o}
 
-all: options libacpi.a libacpi.so test-libacpi
+all: libacpi.a libacpi.so test-libacpi
 
 options:
 	@echo libacpi build options:
@@ -18,24 +18,19 @@
 	@echo "LD       = ${LD}"
 
 .c.o:
-	@echo CC $<
-	@${CC} -c ${CFLAGS} $<
+	${CC} -c ${CFLAGS} $<
 
 ${OBJ}: config.mk libacpi.h
 
 libacpi.a: ${OBJ}
-	@echo AR $@
-	@${AR} $@ ${OBJ}
-	@${RANLIB} $@
+	${AR} $@ ${OBJ}
+	${RANLIB} $@
 
 libacpi.so: ${OBJ}
-	@echo LD $@
-	@${LD} ${SOFLAGS} -o $@.${SOVERSION} ${OBJ}
+	${LD} ${SOFLAGS} -o $@.${SOVERSION} ${OBJ}
 
 test-libacpi: ${OBJ_test}
-	@echo LD $@
-	@${LD} -o $@ ${OBJ_test} ${LDFLAGS}
-	@strip $@
+	${LD} ${LDFLAGS} -o $@ ${OBJ_test}
 
 install: all
 	@echo installing header to ${DESTDIR}${PREFIX}/include
@@ -56,9 +51,6 @@
 	@echo installing manual page to ${DESTDIR}${MANPREFIX}/man3
 	@mkdir -p ${DESTDIR}${MANPREFIX}/man3
 	@cp libacpi.3 ${DESTDIR}${MANPREFIX}/man3
-	@echo installing documentation and misc files to ${DESTDIR}${PREFIX}/share/doc/libacpi
-	@mkdir -p ${DESTDIR}${PREFIX}/share/doc/libacpi
-	@cp -r AUTHORS CHANGES README LICENSE doc ${DESTDIR}${PREFIX}/share/doc/libacpi
 	@echo finished installation
 
 uninstall:
--- Makefile	2009-04-10 21:26:18.000000000 +0200
+++ Makefile.new	2009-04-10 21:27:45.000000000 +0200
@@ -32,7 +32,7 @@
 test-libacpi: ${OBJ_test}
 	${LD} -o $@ ${OBJ_test} ${LDFLAGS}
 
-install: all
+install:
 	@echo installing header to ${DESTDIR}${PREFIX}/include
 	@mkdir -p ${DESTDIR}${PREFIX}/include
 	@cp -f libacpi.h ${DESTDIR}${PREFIX}/include
