--- setup.py.orig	2007-02-08 00:42:41.000000000 +0100
+++ setup.py	2007-02-08 00:43:13.000000000 +0100
@@ -31,6 +31,8 @@
 ]
 
 from distutils.core import setup
+from distutils.extension import Extension
+from Pyrex.Distutils import build_ext
 
 if __name__ == '__main__':
 
@@ -49,5 +51,10 @@
 
         package_dir={'': 'lib'},
         packages=['yaml'],
+        ext_modules=[
+            Extension("_yaml", ["ext/_yaml.pyx"], libraries=['yaml']),
+        ],
+
+        cmdclass = {'build_ext': build_ext}
     )
 
