--- acinclude.m4.orig	2008-07-05 20:14:40.000000000 +0200
+++ acinclude.m4	2008-07-05 20:15:20.000000000 +0200
@@ -450,7 +450,7 @@
 # because AC_CHECK_PROG will do nothing if the variable is already set!
 #
 use_imagemagick=no
-AC_PATH_PROG(MAGIC_BIN, [Magick-config], no, $BINPATH)
+AC_PATH_PROG(MAGIC_BIN, [MagickCore-config], no, $BINPATH)
 if test "$MAGIC_BIN" != "no"; then
   use_imagemagick=yes
 #else
@@ -489,7 +489,7 @@
   if test "${use_imagemagick}" = "no"; then
     AC_MSG_WARN(*** Cannot find ImageMagick include files:  Building w/o ImageMagick support. ***)
   else
-    AC_CHECK_LIB([Magick], [WriteImage], AC_DEFINE(HAVE_IMAGEMAGICK, 1, [Imagemagick image library]), use_imagemagick="no")
+    AC_CHECK_LIB([MagickCore], [WriteImage], AC_DEFINE(HAVE_IMAGEMAGICK, 1, [Imagemagick image library]), use_imagemagick="no")
     if test "${use_imagemagick}" = "no"; then
       AC_MSG_WARN(*** Cannot find ImageMagick library files:  Building w/o ImageMagick support. ***)
     fi
