--- ParMetis-3.1.orig/Programs/Makefile
+++ ParMetis-3.1/Programs/Makefile
@@ -2,8 +2,8 @@
 
 BINDIR = ../Graphs
 
-INCLUDES = -I./ -I../ParMETISLib $(INCDIR) 
-CFLAGS = $(COPTIONS) $(OPTFLAGS) $(INCLUDES)
+#INCLUDES = -I./ -I../ParMETISLib $(INCDIR) 
+#CFLAGS = $(COPTIONS) $(OPTFLAGS) $(INCLUDES)
 
 
 LIBSDIR = -L.. $(LIBDIR) 
@@ -17,7 +17,7 @@
 
           
 .c.o:
-	$(CC) $(CFLAGS) -c $*.c
+	$(CC) -I./ -I../ParMETISLib $(CPPFLAGS) $(CFLAGS) -c $*.c
 
 
 default: $(BINDIR)/ptest$(VERNUM) $(BINDIR)/mtest$(VERNUM) 
--- ParMetis-3.1.orig/METISLib/Makefile
+++ ParMetis-3.1/METISLib/Makefile
@@ -1,5 +1,8 @@
 include ../Makefile.in
 
+all: ../libmetis.a ../libmetis.so
+
+.SUFFIXES: .c .lo .o
 
 CFLAGS = $(COPTIONS) $(OPTFLAGS) -I. $(INCDIR)
 
@@ -16,18 +19,28 @@
        kvmetis.o kwayvolrefine.o kwayvolfm.o subdomains.o \
        mfm.o memory.o mrefine.o checkgraph.o
 
+SHLOBJS = $(OBJS:.o=.lo)
+
 .c.o:
-	$(CC) $(CFLAGS) -c $*.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $*.c
+
+.c.lo:
+	$(CC) $(CPPFLAGS) $(CFLAGS) -fPIC -DPIC -c $< -o $@
 
 ../libmetis.a: $(OBJS)
 	$(AR) $@ $(OBJS)
 	$(RANLIB) $@
 
+../libmetis.so: $(SHLOBJS)
+	$(CC) -shared $^ -Wl,-soname,libmetis.so.3.1 -o $@.3.1 \
+	  $(LIBDIR) -lmpi
+	ln -s libmetis.so.3.1 $@
+
 clean:
-	rm -f *.o 
+	rm -f *.o *.lo
 
 realclean:
-	rm -f *.o ; rm -f ../libmetis.a
+	rm -f *.o *.lo ../libmetis.*
 
 
 checkin:
--- ParMetis-3.1.orig/ParMETISLib/Makefile
+++ ParMetis-3.1/ParMETISLib/Makefile
@@ -1,5 +1,8 @@
 include ../Makefile.in
 
+all: ../libparmetis.a ../libparmetis.so
+
+.SUFFIXES: .c .lo .o
 
 CFLAGS = $(COPTIONS) $(OPTFLAGS) -I. $(INCDIR)
 
@@ -18,19 +21,28 @@
        selectq.o akwayfm.o serial.o move.o \
        mmetis.o mesh.o memory.o weird.o backcompat.o
 
+SHLOBJS = $(OBJS:.o=.lo)
+
 .c.o:
-	$(CC) $(CFLAGS) -c $*.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $*.c
 
+.c.lo:
+	$(CC) $(CPPFLAGS) $(CFLAGS) -fPIC -DPIC -c $< -o $@
 
 ../libparmetis.a: $(OBJS)
 	$(AR) $@ $(OBJS)
 	$(RANLIB) $@
 
+../libparmetis.so: $(SHLOBJS)
+	$(CC) -shared $^ -Wl,-soname,libparmetis.so.3.1 -o $@.3.1 \
+	  $(LIBDIR) -lmpi
+	ln -s libparmetis.so.3.1 $@
+
 clean:
-	rm -f *.o 
+	rm -f *.o *.lo
 
 realclean:
-	rm -f *.o ; rm -f ../libparmetis.a
+	rm -f *.o *.lo ../libparmetis.*
 
 
 checkin:
