Index: INSTALL
===================================================================
--- INSTALL	(revision 512)
+++ INSTALL	(revision 514)
@@ -6,7 +6,7 @@
  - SDL_mixer >= 1.2
  - SDL_image >= 1.2
  - SDL_ttf >= 2.0
- - CMake >= 2.4.0
+ - CMake >= 2.4.4
  
 1. Configure:
 $ cmake . 
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 512)
+++ CMakeLists.txt	(revision 514)
@@ -1,6 +1,6 @@
 project(hedgewars)
 
-cmake_minimum_required(VERSION 2.4.0)
+cmake_minimum_required(VERSION 2.4.4 FATAL_ERROR)
 
 if(DEFINED DATA_INSTALL_DIR)
 	set(SHAREPATH ${DATA_INSTALL_DIR}/hedgewars/)
