--- eventlet/greenio.py	2009-07-24 04:25:27.000000000 +0200
+++ eventlet/greenio.py	2009-07-27 16:39:55.000000000 +0200
@@ -534,7 +534,7 @@
     """
     def __init__(self, fd):
         super(GreenSSL, self).__init__(fd)
-        assert(isinstance(fd, (SSL.ConnectionType)),
+        assert isinstance(fd, (SSL.ConnectionType), \
                "GreenSSL can only be constructed with an "\
                "OpenSSL Connection object")
         self.sock = self

