--- ./writerfilter/source/dmapper/DomainMapper_Impl.hxx.gcc43~  2007-10-29 20:26:21.000000000 +0500
+++ ./writerfilter/source/dmapper/DomainMapper_Impl.hxx 2008-08-15 22:57:23.000000000 +0600
@@ -46,6 +46,7 @@
 #endif
 #include <vector>
 #include <stack>
+#include <cstring>
 
 #ifndef INCLUDED_RESOURCESIDS
 #include <doctok/resourceids.hxx>
--- ./writerfilter/source/dmapper/GraphicImport.cxx.gcc43~      2008-08-15 23:00:24.000000000 +0600
+++ ./writerfilter/source/dmapper/GraphicImport.cxx     2008-08-15 23:01:02.000000000 +0600
@@ -98,7 +98,7 @@
 #endif
 #include <rtl/ustrbuf.hxx>
 
-
+#include <cstring>
 #include <iostream>
 
 namespace dmapper
--- ./writerfilter/source/odiapi/qname/QName.cxx.gcc43~ 2008-08-15 23:02:44.000000000 +0600
+++ ./writerfilter/source/odiapi/qname/QName.cxx        2008-08-15 23:03:07.000000000 +0600
@@ -43,6 +43,7 @@
 #include <assert.h>
 #include <stdexcept>
 #include <string>
+#include <cstring>
 #include <stdio.h>
 
 namespace writerfilter
--- ./writerfilter/unocomponent/debugservices/rtfsl/RTFSLParser.cxx.gcc43~      2008-08-15 23:05:02.000000000 +0600
+++ ./writerfilter/unocomponent/debugservices/rtfsl/RTFSLParser.cxx     2008-08-15 23:05:21.000000000 +0600
@@ -66,6 +66,7 @@
 #include <odiapi/qname/QName.hxx>
 #include <odiapi/props/Properties.hxx>
 #include <stack>
+#include <cstring>
 
 #include <ctype.h>
 
---  ./writerfilter/unocomponent/debugservices/rtftok/ScannerTestService.cxx.gcc43~     2008-08-15 23:07:02.000000000 +0600
+++  ./writerfilter/unocomponent/debugservices/rtftok/ScannerTestService.cxx    2008-08-15 23:07:51.000000000 +0600
@@ -63,6 +63,7 @@
 #include <com/sun/star/beans/PropertyValue.hpp>
 #include <com/sun/star/beans/XPropertySet.hpp>
 #include <comphelper/seqstream.hxx>
+#include <cstring>
 
 #include <ctype.h>
 
--- ./writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx.gcc43~      2008-08-15 23:08:39.000000000 +0600
+++ ./writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx     2008-08-15 23:08:57.000000000 +0600
@@ -55,6 +55,7 @@
 #include <hash_set>
 #include <assert.h>
 #include <string>
+#include <cstring>
 #include <cppuhelper/implbase2.hxx>
 #include <com/sun/star/embed/XTransactedObject.hpp>
 #include <com/sun/star/embed/XStorage.hpp>
--- ./writerfilter/source/ooxml/gperfattributes.hxx.gcc43~      2008-08-16 10:37:04.000000000 +0600
+++  ./writerfilter/source/ooxml/gperfattributes.hxx    2008-08-16 10:37:22.000000000 +0600
@@ -31,6 +31,7 @@
 
 
 #include "OOXMLtokens.hxx"
+#include <cstring>
 
 namespace ooxml { namespace tokenmap { namespace attributes {
 struct token { const char * name; TokenEnum_t nToken; };
--- ./writerfilter/source/ooxml/gperfelements.hxx.gcc43~        2008-08-16 10:36:49.000000000 +0600
+++ ./writerfilter/source/ooxml/gperfelements.hxx       2008-08-16 10:37:27.000000000 +0600
@@ -31,6 +31,7 @@
 
 
 #include "OOXMLtokens.hxx"
+#include <cstring>
 
 namespace ooxml { namespace tokenmap { namespace elements {
 struct token { const char * name; TokenEnum_t nToken; };
--- crashrep/source/unx/main.cxx.orig	2009-02-21 19:15:51.000000000 +0500
+++ crashrep/source/unx/main.cxx	2009-02-21 19:16:13.000000000 +0500
@@ -45,6 +45,7 @@
 #include <unistd.h>
 #include <pwd.h>
 #include <pthread.h>
+#include <limits.h>
 
 #include <hash_map>
 #include <vector>
