--- patches/src680/apply.orig	2009-02-20 22:00:46.000000000 +0500
+++ patches/src680/apply	2009-02-20 22:01:35.000000000 +0500
@@ -89,7 +89,7 @@
 # openmamba
 openmamba: LinuxCommon, GlobalDesktopFiles
 #Gentoo
-GentooBase : LinuxCommon, GentooOnly, NotDebian, Lockdown, CalcSolver, CairoFonts, BerkeleyDB43, Gcc43
+GentooBase : LinuxCommon, GentooOnly, NotDebian, Lockdown, CairoFonts, BerkeleyDB43, Gcc43
 GentooUnstable: GentooBase, GentooExperimental, TransparentPrinting, PptExport
 Gentoo : GentooBase
 Gentoo64: GentooBase
@@ -131,7 +131,10 @@
 
 #InfraUnix:  LinuxCommon, EMFPlus, CustomUserConfig, PostgreSQL, InfraPatches, InfraNotUbuntu,  InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, NotDebian, CustomKeyBinding
 InfraUnix:  LinuxCommon, EMFPlus, CJK, OOXML, CustomUserConfig, PostgreSQL, InfraPatches, InfraNotUbuntu, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, CustomKeyBinding
-UbuntuInfra:  LinuxCommon, EMFPlus, SVGImport, UpdaterRemoval, UbuntuBased, Layout, LayoutDialogs, CJK, OOXML, Split, Fpickers, QuickStarter, DebianBaseOnly, UbuntuOnly, CustomUserConfig, PostgreSQL, InfraPatches, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, CustomKeyBinding, InfraUbuntu
+UbuntuInfra:  LinuxCommon, EMFPlus, SVGImport, UpdaterRemoval, UbuntuBased, Layout, LayoutDialogs, CJK, OOXML, Fpickers, QuickStarter, DebianBaseOnly, UbuntuOnly, CustomUserConfig, PostgreSQL, InfraPatches, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, CustomKeyBinding, InfraUbuntu
+InfraGentooBase: Gentoo, EMFPlus, SVGImport, UpdaterRemoval, Layout, LayoutDialogs, CJK, OOXML, Fpickers, QuickStarter, InfraNotUbuntu, CustomUserConfig, InfraPatches, InfraUnixPatches, Infra1252toLocale, InfraNotSolaris, CustomKeyBinding
+InfraGentoo: InfraGentooBase
+InfraGentooPG: InfraGentooBase, InfraUnixPatchesPG, PostgreSQL
 InfraSolaris:  LinuxCommon, EMFPlus, CJK, OOXML, CustomUserConfig, InfraPatches, InfraNotUbuntu, Infra1252toLocale, InfraSolaris, NotDebian, CustomKeyBinding
 InfraWin32: Win32Common, EMFPlus, CJK, OOXML, CustomUserConfig, InfraPatches, InfraNotUbuntu, InfraWin32Patches, Infra1252toLocale, InfraNotSolaris, NotDebian, CustomKeyBinding
 InfraFake: QuickStarter, WMF, SVGImport, UnitTesting, Layout, CJK, EMFPlus, Fpickers, LayoutDialogs, UnitBootstrap, AutoCorrectCapsLock
@@ -1207,7 +1210,7 @@
 # strictly required on SL10.1/SLED10 where the xulrunner-nss.pc is broken, n#195272
 libxmlsec-system-nss.diff, i#69368, pmladek
 
-# xulrunner-1.9.diff, i#86670
+xulrunner-1.9.diff, i#86670
 
 # allow separate helpcontent build
 
@@ -1849,7 +1852,7 @@
 sdbc-postgresql.diff
 sdbc-postgresql-build-lst.diff
 sdbc-postgresql-config_office.diff
-# connectivity-workben-postgresql.diff
+connectivity-workben-postgresql.diff
 gcc-4.3-postgresql.diff
 
 
@@ -2304,6 +2307,7 @@
 [ Gcc43 ]
 # Make it compile with gcc 4.3
 cws-gcc430two.diff, i#83276
+build-gcc43-missingincludes.diff
 
 [ OxygenOfficePalettes ]
 palette-enhanced-arrow.diff
@@ -2598,8 +2602,10 @@
 [InfraUbuntu]
 disable-scsolver-menu-entry.diff
 
-[InfraUnixPatches]
+[InfraUnixPatchesPG]
 connectivity-freebsd-libs.diff
+
+[InfraUnixPatches]
 freebsd-store-alloca.diff
 freebsd-writerfilter.diff
 
