--- canvas/source/tools/image.cxx.orig	2007-09-20 20:50:23.000000000 +0600
+++ canvas/source/tools/image.cxx	2008-07-27 14:33:26.000000000 +0600
@@ -2302,8 +2302,8 @@
             // TODO(E1): Return value
             // TODO(F1): FillRule
             fillGradient( rValues,
-                          rValues.maColor1,
-                          rValues.maColor2,
+                          rValues.maColors [0],
+                          rValues.maColors [rValues.maColors.getLength () - 1],
                           aPoly,
                           aTextureTransform,
                           textures[0] );
