--- soltools/mkdepend/makefile.mk.orig	2007-11-02 17:42:04.000000000 +0500
+++ soltools/mkdepend/makefile.mk	2008-08-06 20:28:10.000000000 +0600
@@ -56,7 +56,7 @@
 LIBSALCPPRT=$(0)
 UWINAPILIB=$(0)
 
-CDEFS+=-DNO_X11 -DXP_PC -DHW_THREADS -DINCLUDEDIR=$(EMQ)".$(EMQ)"
+CDEFS+=-Wno-implicit-function-declaration -DNO_X11 -DXP_PC -DHW_THREADS -DINCLUDEDIR=$(EMQ)".$(EMQ)"
 
 OBJFILES=  \
 		$(OBJ)$/cppsetup.obj \
