--- officecfg/util/makefile.mk.orig	2009-01-11 01:08:01.000000000 +0500
+++ officecfg/util/makefile.mk	2009-01-11 01:08:45.000000000 +0500
@@ -41,7 +41,6 @@
 
 $(MISC)$/$(TARGET)_delzip :
 	-$(RM) $(BIN)$/registry_{$(alllangiso)}.zip	
-	-$(RM) $(COMMON_OUTPUT)$/bin$/registry_{$(alllangiso)}.zip
 
 $(BIN)$/registry_{$(alllangiso)}.zip : $(MISC)$/$(TARGET)_delzip
 	cd $(MISC)$/registry$/res$/$(@:b:s/registry_//) && zip -ru ..$/..$/..$/..$/bin$/registry_$(@:b:s/registry_//).zip org/*
