diff -u bin/apply.pl bin/apply.pl
--- bin/apply.pl
+++ bin/apply.pl
@@ -753,7 +753,7 @@
     my $ver_line;
     my $is_old = 1;
 
-    open ($Patch, "/usr/local/bin/gpatch --version|") || die "Can't run patch: $!";
+    open ($Patch, "/usr/bin/patch --version|") || die "Can't run patch: $!";
     $ver_line = <$Patch>;
     $ver_line =~ m/\s+(\d+)\.(\d+)\.(\d+)/ || die "Can't get patch version\n";
     if ($1 >= 2 && $2 >= 5 && $3 >= 9) {
@@ -1040,7 +1040,7 @@
     $applied_patches = $dest_dir.'/applied_patches';
 
     $opts = join ' ', @arguments;
-    $base_cmd = "/usr/local/bin/gpatch -l -p0 $opts -d $dest_dir";
+    $base_cmd = "/usr/bin/patch -l -p0 $opts -d $dest_dir";
     if (is_old_patch_version()) {
 $base_cmd = 'sed \'s/^\(@.*\)\r$/\1/\' | ' . $base_cmd;
     }
diff -u bin/enable-dict bin/enable-dict
--- bin/enable-dict
+++ bin/enable-dict
@@ -6,7 +6,7 @@
 
 test -f $ARC || exit 1
 cd $WRKDIR/dictionaries || exit 2
-gtar xjf $ARC
+tar xjf $ARC
 grep -q $LNG prj/build.lst && exit 0
 line="di      dictionaries\\$LNG      nmake   -       all     di_$LNG di_diclst NULL"
 head -n `wc -l prj/build.lst |awk '{print $1 - 1}'` prj/build.lst > prj/build.lst.tmp
