diff -Naur homebank-3.3_unstable.orig/src/import.c homebank-3.3_unstable/src/import.c
--- homebank-3.3_unstable.orig/src/import.c	2007-03-30 17:49:34.000000000 +0200
+++ homebank-3.3_unstable/src/import.c	2007-03-30 17:32:10.000000000 +0200
@@ -21,7 +21,10 @@
 #include "import.h"
 #include "list_operation.h"
 
+
+#ifndef NOOFX
 #include <libofx/libofx.h>
+#endif
 
 /****************************************************************************/
 /* Debug macros                                                             */
@@ -247,6 +250,8 @@
 	return list;
 }
 
+
+#ifndef NOOFX
 /*
 ****
 ****
@@ -599,6 +604,8 @@
 	return ofx_ope_list;
 }
 
+#endif
+
 /*
 ***
 *** qif part
@@ -625,16 +632,17 @@
 	gtk_list_store_clear (GTK_LIST_STORE(gtk_tree_view_get_model(GTK_TREE_VIEW(data->imported_ope))));
 
 
-
+#ifndef NOOFX
 	ofx_acc_list = NULL;
 	da_account_destroy(ofx_acc_list);
+#endif
 
 	da_operation_destroy(data->ope_imp_list);
 	data->ope_imp_list = NULL;
+
+#ifndef NOOFX
 	ofx_ope_list = NULL;
-	
-	
-	
+#endif	
 	
 }
 
@@ -670,10 +678,12 @@
 			case FILETYPE_UNKNOW:
 				gtk_label_set_text(GTK_LABEL(data->user_info), _("Unknow/Invalid file..."));
 				break;
+#ifndef NOOFX
 			case FILETYPE_OFX:
 				gtk_label_set_text(GTK_LABEL(data->user_info), _("OFX file recognized !"));
 				data->valid = TRUE;
 				break;
+#endif
 			case FILETYPE_CSV_HB:
 				gtk_label_set_text(GTK_LABEL(data->user_info), _("CSV operation file recognized !"));
 				data->valid = TRUE;
@@ -805,6 +815,7 @@
 
 	DB( g_print("(import) apply\n") );
 
+#ifndef NOOFX
 	GList *tmplist = g_list_first(ofx_acc_list);
 
 	while (tmplist != NULL)
@@ -815,6 +826,7 @@
 		
 		tmplist = g_list_next(tmplist);
 	}
+#endif
 
 	// then import operations
 	model = gtk_tree_view_get_model(GTK_TREE_VIEW(data->imported_ope));
@@ -1066,13 +1078,18 @@
 			import_clearall(data);
 			switch(data->filetype)
 			{
+
+#ifndef NOOFX
 				case FILETYPE_OFX:
 					/* ofx_acc_list & ofx_ope_list are filled here */
 					data->ope_imp_list = homebank_ofx_import(data->filename);
 					break;
+#endif
+
 				case FILETYPE_CSV_HB:
 					accnum = gtk_combo_box_get_active(GTK_COMBO_BOX(data->PO_acc));
 					data->ope_imp_list = homebank_csv_import(data->filename, accnum);
+				
 					break;
 			}
 								
@@ -1089,7 +1106,8 @@
 		gchar *txt;
 		
 			import_count_changes(data);
-		
+
+		#ifndef NOOFX
 			txt = g_strdup_printf(
 				_(
 				"%d account(s) will be created.\n\n" \
@@ -1100,6 +1118,18 @@
 				data->imported,
 				data->total-data->imported
 				);
+
+		#else
+			txt = g_strdup_printf(
+				_(
+				"%d operation(s) will be imported.\n" \
+				"%d operation(s) will be rejected."
+				),
+				data->imported,
+				data->total-data->imported
+				);
+		#endif
+
 			gtk_label_set_text(GTK_LABEL(data->last_info), txt);
 			g_free(txt);		
 
@@ -1149,7 +1179,7 @@
 	
 		"HomeBank can import operations from several file format:\n" \
 		"\n" \
-		"- OFX/QFX\n" \
+		"- OFX/QFX (optional)\n" \
 		"- CSV (Homebank operation csv export format only)\n" \
 		"\n" \
 		"The import process has several steps. Your HomeBank accounts\n" \
diff -Naur homebank-3.3_unstable.orig/src/list_operation.c homebank-3.3_unstable/src/list_operation.c
--- homebank-3.3_unstable.orig/src/list_operation.c	2007-03-30 17:49:34.000000000 +0200
+++ homebank-3.3_unstable/src/list_operation.c	2007-03-30 17:32:10.000000000 +0200
@@ -317,7 +317,7 @@
 	gtk_tree_view_column_pack_start(column, renderer, TRUE);
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_date_cell_data_function, NULL, NULL);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_DATE);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
 	/* column 3: Infos */
@@ -332,7 +332,7 @@
 	gtk_tree_view_column_pack_start(column, renderer, TRUE);
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_info_cell_data_function, (gpointer)2, NULL);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_INFO);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
 	/* column 4: Payee */
@@ -342,7 +342,7 @@
 	gtk_tree_view_column_pack_start(column, renderer, TRUE);
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_payee_cell_data_function, NULL, NULL);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_PAYEE);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
 	/* column 5: Wording */
@@ -352,7 +352,7 @@
 	gtk_tree_view_column_pack_start(column, renderer, TRUE);
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_wording_cell_data_function, NULL, NULL);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_WORDING);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
 	/* column 6: Expense */
@@ -364,7 +364,7 @@
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_amount_cell_data_function, (gpointer)-1, NULL);
 	gtk_tree_view_column_set_alignment (column, 1.0);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_EXPENSE);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
 	/* column 7: Income */
@@ -376,7 +376,7 @@
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_amount_cell_data_function, (gpointer)1, NULL);
 	gtk_tree_view_column_set_alignment (column, 1.0);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_INCOME);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
 	/* column 8: Category */
@@ -386,7 +386,7 @@
 	gtk_tree_view_column_pack_start(column, renderer, TRUE);
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_category_cell_data_function, NULL, NULL);
 	gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_CATEGORY);
-	//gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
   /* column 9: empty */
@@ -416,6 +416,7 @@
 
 /* ---------------------------------------------- */
 
+#ifndef NOOFX
 /*
 ** account cell function
 */
@@ -443,6 +444,7 @@
 
 	g_object_set(renderer, "text", acc->name, NULL);
 }
+#endif
 
 /*
 ** amount cell function
@@ -567,6 +569,7 @@
 	gtk_tree_view_column_set_cell_data_func(column, renderer, ope_importstate_cell_data_function, NULL, NULL);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
 
+#ifndef NOOFX
 	/* column: Account */
 	column = gtk_tree_view_column_new();
 	gtk_tree_view_column_set_title(column, _("Account"));
@@ -576,6 +579,7 @@
 	//gtk_tree_view_column_set_sort_column_id (column, LST_DSPOPE_DATE);
 	//gtk_tree_view_column_set_resizable(column, TRUE);
 	gtk_tree_view_append_column (GTK_TREE_VIEW(view), column);
+#endif	
 	
 	/* column: Date */
 	column = gtk_tree_view_column_new();
