--- monitorix.conf.orig	2009-08-11 01:04:55.000000000 +0200
+++ monitorix.conf	2009-08-11 01:05:56.000000000 +0200
@@ -31,7 +31,7 @@
 # General
 our $TITLE = "FIBRANET NSP, SL";	# your company name
 our $HOSTNAME = "";			# hostname of this server
-our $OSTYPE = "Linux-RHFC";		# choose your OS type (see below)
+our $OSTYPE = "Linux-Gentoo";		# choose your OS type (see below)
 our $IDATE = "01 Jan 2000";		# initial statistics date
 our $SAMBAVER = "3";			# Samba version (2 or 3)
 our $MULTIHOST = "N";			# Enable multihost feature
@@ -223,7 +223,7 @@
 
 our %HTTP_LOG = ("Linux-RHFC"	=> "/var/log/httpd/access_log",
 		"Linux-Debian"	=> "/var/log/apache2/access.log",
-		"Linux-Gentoo"	=> "/var/log/apache2/access.log",
+		"Linux-Gentoo"	=> "/var/log/apache2/access_log",
 		"Linux-Slack"	=> "/var/log/httpd/access.log",
 		"Linux-SuSE"	=> "/var/log/apache2/access_log",
 		"Linux-Generic"	=> "/var/log/httpd/access_log",
