diff -Naur soci-2.2.0.orig/src/backends/empty/test/test-empty.cpp soci-2.2.0/src/backends/empty/test/test-empty.cpp
--- soci-2.2.0.orig/src/backends/empty/test/test-empty.cpp	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/backends/empty/test/test-empty.cpp	2008-06-18 18:42:08.000000000 +0200
@@ -9,6 +9,7 @@
 #include "soci-empty.h"
 #include <iostream>
 #include <string>
+#include <cstdlib>
 #include <cassert>
 #include <ctime>
 
diff -Naur soci-2.2.0.orig/src/backends/mysql/common.h soci-2.2.0/src/backends/mysql/common.h
--- soci-2.2.0.orig/src/backends/mysql/common.h	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/backends/mysql/common.h	2008-06-18 18:47:10.000000000 +0200
@@ -9,6 +9,7 @@
 #define MYSQL_COMMON_H_INCLUDED
 
 #include "soci-mysql.h"
+#include <ctime>
 
 namespace SOCI {
 
diff -Naur soci-2.2.0.orig/src/backends/mysql/session.cpp soci-2.2.0/src/backends/mysql/session.cpp
--- soci-2.2.0.orig/src/backends/mysql/session.cpp	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/backends/mysql/session.cpp	2008-06-18 18:50:01.000000000 +0200
@@ -12,7 +12,8 @@
 #include <cctype>
 #include <cerrno>
 #include <ciso646>
-
+#include <cstdlib>
+#include <climits>
 
 #ifdef _MSC_VER
 #pragma warning(disable:4355)
diff -Naur soci-2.2.0.orig/src/backends/mysql/standard-into-type.cpp soci-2.2.0/src/backends/mysql/standard-into-type.cpp
--- soci-2.2.0.orig/src/backends/mysql/standard-into-type.cpp	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/backends/mysql/standard-into-type.cpp	2008-06-18 18:51:40.000000000 +0200
@@ -11,7 +11,8 @@
 #include <soci.h>
 #include <soci-platform.h>
 #include "common.h"
-
+#include <cstdlib>
+#include <cstring>
 #include <ciso646>
 
 #ifdef _MSC_VER
diff -Naur soci-2.2.0.orig/src/backends/mysql/standard-use-type.cpp soci-2.2.0/src/backends/mysql/standard-use-type.cpp
--- soci-2.2.0.orig/src/backends/mysql/standard-use-type.cpp	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/backends/mysql/standard-use-type.cpp	2008-06-18 18:54:01.000000000 +0200
@@ -13,6 +13,7 @@
 #include <soci-platform.h>
 #include <ciso646>
 #include <limits>
+#include <cstring>
 
 #ifdef _MSC_VER
 #pragma warning(disable:4355)
diff -Naur soci-2.2.0.orig/src/backends/mysql/vector-into-type.cpp soci-2.2.0/src/backends/mysql/vector-into-type.cpp
--- soci-2.2.0.orig/src/backends/mysql/vector-into-type.cpp	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/backends/mysql/vector-into-type.cpp	2008-06-18 18:59:01.000000000 +0200
@@ -12,6 +12,7 @@
 #include <soci.h>
 #include <soci-platform.h>
 #include <ciso646>
+#include <cstdlib>
 
 #ifdef _MSC_VER
 #pragma warning(disable:4355)
diff -Naur soci-2.2.0.orig/src/core/soci.h soci-2.2.0/src/core/soci.h
--- soci-2.2.0.orig/src/core/soci.h	2008-06-18 18:37:17.000000000 +0200
+++ soci-2.2.0/src/core/soci.h	2008-06-18 18:40:27.000000000 +0200
@@ -13,6 +13,8 @@
 #include <vector>
 #include <ostream>
 #include <sstream>
+#include <typeinfo>
+#include <memory>
 #include <ctime>
 #include <cassert>
 
