--- Makefile.old	2010-05-27 19:22:34.000000000 +0200
+++ Makefile	2010-05-29 19:24:55.000000000 +0200
@@ -7,6 +7,7 @@
 
 build ?= debug
 prefix ?= /usr/local
+libprefix ?= $(prefix)/lib
 
 OBJDIR := build/$(build)
 GENDIR := build/generated
@@ -301,7 +302,8 @@
 	rm -rf build
 
 install: $(DIRS) $(APPS) $(MUPDF_LIB)
-	install $(APPS) $(prefix)/bin
-	install $(MUPDF_LIB) $(prefix)/lib
-	install $(MUPDF_HDR) $(prefix)/include
+	install -d $(prefix)/bin $(libprefix) $(prefix)/include
+	install $(APPS) -t $(prefix)/bin
+	install $(MUPDF_LIB) -t $(libprefix)
+	install $(MUPDF_HDR) -t $(prefix)/include
 
--- Makerules.old	2010-05-20 18:37:16.000000000 +0200
+++ Makerules	2010-05-29 19:29:49.000000000 +0200
@@ -4,8 +4,8 @@
 OS := $(OS:MINGW%=MINGW)
 
 LIBS := -ljbig2dec -lopenjpeg -lfreetype -ljpeg -lz -lm
-CFLAGS := -Wall --std=gnu99 -Ifitz -Imupdf
-LDFLAGS =
+CFLAGS += -Wall --std=gnu99 -Ifitz -Imupdf
+LDFLAGS +=
 CC = cc
 LD = $(CC)
 AR = ar
@@ -21,7 +21,7 @@
 ifeq "$(OS)" "Linux"
 CFLAGS += `pkg-config --cflags freetype2`
 LDFLAGS += `pkg-config --libs freetype2`
-X11LIBS = -lX11 -lXext
+X11LIBS += `pkg-config --libs x11 xext`
 PDFVIEW_EXE = $(X11VIEW_EXE)
 ifeq  "$(build)" "release"
 CFLAGS += -mmmx -ftree-vectorize -msse -msse2 -march=k8 -DARCH_X86
