--- Makefile.am.orig	2008-06-29 13:55:18.000000000 +0200
+++ Makefile.am	2008-06-29 13:56:03.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS=data doc share src 
+SUBDIRS=data share src 
 
 MAINTAINERCLEANFILES = aclocal.m4 config.h.in configure Makefile.in stamp-h.in
 
@@ -4,10 +4,6 @@
 
 EXTRA_DIST = COPYING INSTALL rebuild debian/* intl/* win32/* pkg/*
 
-docsdir = $(datadir)/tucnak2/doc
-	
-docs_DATA = AUTHORS ChangeLog TODO   
-
 stamp = `date +%Y%m%d_%H%M%S`
 
 distd: dist
