--- src/Makefile.am.orig	2008-09-03 13:35:33.000000000 +0200
+++ src/Makefile.am	2008-09-11 10:57:05.000000000 +0200
@@ -1,7 +1,7 @@
 #bin_PROGRAMS = tucnak soundwrapper stucnak
-bin_PROGRAMS = tucnak soundwrapper 
+bin_PROGRAMS = tucnak2 soundwrapper 
 
-tucnak_SOURCES = winkey.c excdb.c adif.c alsa.c bfu.c charsets.c codepage.h codepage.inc \
+tucnak2_SOURCES = winkey.c excdb.c adif.c alsa.c bfu.c charsets.c codepage.h codepage.inc \
 	control.c cor.c cor.h cor.inc \
 	cwdaemon.c cwdb.c cwwindow.c davac4.c dsp.c dwdb.c dxc.c ebw.c edi.c \
 	error.c fhs.c fifo.c font.c font.inc gfx_png.c ghash.c \
@@ -14,7 +14,7 @@
 	ssbd.c stats.c subwin.c terminal.c threadpipe.c titlpage.c trig.c \
 	ttys.c txts.c txts.h txts.inc uni_7b.inc wizz.c zstring.c
 
-tucnak_LDADD = @TUCNAK_LDADD@
+tucnak2_LDADD = @TUCNAK_LDADD@
 
 soundwrapper_SOURCES = soundwrapper.c	
 
