diff -ur jailkit.orig/Makefile.in jailkit/Makefile.in
--- jailkit.orig/Makefile.in	2006-11-12 16:41:47.000000000 +0100
+++ jailkit/Makefile.in	2006-11-12 16:42:32.000000000 +0100
@@ -70,10 +70,10 @@
 	@cd man/ && $(MAKE) install
 	# test if the jk_chrootsh is already in /etc/shells
 	# this previously had @echo but that fails on FreeBSD
-	if ! grep @prefix@/sbin/jk_chrootsh $(DESTDIR)/etc/shells ; then \
-		echo "appending @prefix@/sbin/jk_chrootsh to $(DESTDIR)/etc/shells";\
-		echo @prefix@/sbin/jk_chrootsh >> $(DESTDIR)/etc/shells ;\
-	fi
+	# if ! grep @prefix@/sbin/jk_chrootsh $(DESTDIR)/etc/shells ; then \
+	# 	echo "appending @prefix@/sbin/jk_chrootsh to $(DESTDIR)/etc/shells";\
+	# 	echo @prefix@/sbin/jk_chrootsh >> $(DESTDIR)/etc/shells ;\
+	# fi
 
 
 uninstall:
diff -ur jailkit.orig/py/Makefile.in jailkit/py/Makefile.in
--- jailkit.orig/py/Makefile.in	2006-11-12 16:41:47.000000000 +0100
+++ jailkit/py/Makefile.in	2006-11-12 16:42:06.000000000 +0100
@@ -39,22 +39,22 @@
 	python -c "import py_compile;py_compile.compile('jk_lib.py')"
 
 jk_init: jk_init.in
-	sed -e "s!LIBDIR='[a-z/]*'!LIBDIR='${PYLIBDIR}'!" -e "s:#!/usr/bin/python:#!$(PYTHONPATH):" < jk_init.in > jk_init
+	cp jk_init.in jk_init
 
 jk_check: jk_check.in
-	sed -e "s!LIBDIR='[a-z/]*'!LIBDIR='${PYLIBDIR}'!" -e "s:#!/usr/bin/python:#!$(PYTHONPATH):" < jk_check.in > jk_check
+	cp jk_check.in jk_check
 
 jk_cp: jk_cp.in
-	sed -e "s!LIBDIR='[a-z/]*'!LIBDIR='${PYLIBDIR}'!" -e "s:#!/usr/bin/python:#!$(PYTHONPATH):" < jk_cp.in > jk_cp
+	cp jk_cp.in jk_cp
 
 jk_list: jk_list.in
-	sed -e "s!LIBDIR='[a-z/]*'!LIBDIR='${PYLIBDIR}'!" -e "s:#!/usr/bin/python:#!$(PYTHONPATH):" < jk_list.in > jk_list
+	cp jk_list.in jk_list
 
 jk_addjailuser: jk_addjailuser.in
-	sed -e "s!LIBDIR='[a-z/]*'!LIBDIR='${PYLIBDIR}'!" -e "s:#!/usr/bin/python:#!$(PYTHONPATH):" < jk_addjailuser.in > jk_addjailuser
+	cp jk_addjailuser.in jk_addjailuser
 
 jk_jailuser: jk_jailuser.in
-	sed -e "s!LIBDIR='[a-z/]*'!LIBDIR='${PYLIBDIR}'!" -e "s:#!/usr/bin/python:#!$(PYTHONPATH):" < jk_jailuser.in > jk_jailuser
+	cp jk_jailuser.in jk_jailuser
 
 jailkit: jk_lib.pyc jk_cp jk_init jk_check jk_addjailuser jk_jailuser jk_list 
 
