--- /var/tmp/portage/media-libs/glfw-2.6/work/glfw/compile.sh	2007-07-01 11:46:46.000000000 +0200
+++ glfw/compile.sh	2007-10-09 15:37:11.000000000 +0200
@@ -68,7 +68,7 @@
 LDFLAGS=
 INCS=
 LIBS="-lGL -lX11"
-
+SONAME=libglfw.so.2.6
 
 ##########################################################################
 # Compilation commands
@@ -85,7 +85,7 @@
 case "x`uname 2> /dev/null`" in
 xLinux)
   CFLAGS="$CFLAGS -Dlinux"
-  LDFLAGS="-shared"
+  LDFLAGS="-shared -Wall -lc -Wl,-soname -Wl,$SONAME"
   echo " Linux" 1>&6
   ;;
 xDarwin)
@@ -534,9 +534,9 @@
 # Post fixups
 ##########################################################################
 if [ "x$use_gcc" = xyes ]; then
-  CFLAGS_SPEED="-c -I. -I.. $CFLAGS -O3 -ffast-math -Wall"
-  CFLAGS="-c -I. -I.. $CFLAGS -Os -Wall"
-  CFLAGS_LINK="$INCS -O3 -ffast-math -Wall"
+  CFLAGS_SPEED="-c -I. -I.. $CFLAGS -O3 -ffast-math -Wall -fPIC"
+  CFLAGS="-c -I. -I.. $CFLAGS -Os -Wall -fPIC"
+  CFLAGS_LINK="$INCS -O3 -ffast-math -Wall -fPIC"
 else
   CFLAGS_SPEED="-c -I. -I.. $CFLAGS -O"
   CFLAGS="-c -I. -I.. $CFLAGS -O"
@@ -564,6 +564,7 @@
 echo "LDFLAGS      = $LDFLAGS" >>$MKNAME
 echo "LFLAGS       = $LFLAGS" >>$MKNAME
 echo "LIBS         = $LIBS" >>$MKNAME
+echo "SONAME       = $SONAME" >>$MKNAME
 echo " " >>$MKNAME
 cat './lib/x11/Makefile.x11.in' >>$MKNAME
 
--- /var/tmp/portage/media-libs/glfw-2.6/work/glfw/lib/x11/Makefile.x11.in	2007-07-01 11:46:46.000000000 +0200
+++ glfw/lib/x11/Makefile.x11.in	2007-10-09 15:22:20.000000000 +0200
@@ -105,7 +105,7 @@
 # Rule for building shared library
 ##########################################################################
 libglfw.so: $(SHARED_OBJS)
-	$(CC) $(LDFLAGS) -o $@ $(SHARED_OBJS) $(LFLAGS) $(LIBS)
+	$(CC) $(LDFLAGS) -o $(SONAME) $(SHARED_OBJS) $(LFLAGS) $(LIBS)
 
 
 ##########################################################################

