--- Configure	2012-04-28 16:47:50.589162001 +0200
+++ Configure.NEW	2012-06-04 19:44:28.757397001 +0200
@@ -43,6 +43,7 @@
    -upnp        Compiler avec le support UPNP avec miniupnpc (pour les Routeurs NAT)
    -ruby        Compiler avec le support du script RUBY
    -no-blowfish Compiler sans encodage blowfish
+   -no-chroot   Compiler sans support chroot
    -no-openssl  Compiler sans OpenSSL lib
    -no-http     Compiler sans HTTP server
    -no-admin    Compiler sans HTTP Admin
@@ -71,6 +72,7 @@
    -upnp        Benutze miniupnpc Bibliothek (für UPNP-NAT Routers)
    -ruby        Benutze RUBY Skript Erweiterung
    -no-blowfish Baue ohne Blowfish Verschlusselung
+   -no-chroot   Baue ohne chroot
    -no-openssl  Baue ohne OpenSSL Bibliothek
    -no-http     Baue ohne HTTP Server
    -no-admin    Baue ohne HTTP Admin
@@ -99,6 +101,7 @@
    -upnp        Compile with UPNP support with miniupnpc (for NAT Routers)
    -ruby        Compile with RUBY scripting support
    -no-blowfish Compile without blowfish encryption
+   -no-chroot   Compile without chroot support
    -no-openssl  Compile without OpenSSL lib
    -no-http     Compile without HTTP server
    -no-admin    Compile without HTTP Admin
@@ -455,6 +458,7 @@
 OPT_RUBY=false
 OPT_NOADMIN=false
 OPT_NOBLOWFISH=false
+OPT_NOCHROOT=false
 OPT_NOHTTP=false
 OPT_NOTELNET=false
 OPT_NOMEMSAVE=false
@@ -490,6 +494,7 @@
  -r*) OPT_RUBY=true;;
  -no-a*) OPT_NOADMIN=true ;;
  -no-b*) OPT_NOBLOWFISH=true ;;
+ -no-c*) OPT_NOCHROOT=true ;;
  -no-h*) OPT_NOHTTP=true ;;
  -no-t*) OPT_NOTELNET=true ;;
  -no-o*) OPT_SSL=false ;;
@@ -634,8 +639,6 @@
 "
  ;;
 esac
-else
-DEBUG="-O2"
 fi
 
 if $OPT_WARN; then
@@ -747,31 +750,36 @@
  ${msg} "${irt_start} UPNP ${irt_with} lminiupnpc."
 fi
 
-if $OPT_NOHTTP; then
- ${msg} "${irt_configure} ${irt_without} HTTP Server."
- echo "#define WITHOUT_HTTP" >> src/iroffer_config.h
-fi
-
 if $OPT_NOADMIN; then
  ${msg} "${irt_configure} ${irt_without} HTTP Admin."
  echo "#define WITHOUT_HTTP_ADMIN" >> src/iroffer_config.h
 fi
 
-if $OPT_NOTELNET; then
- ${msg} "${irt_configure} ${irt_without} Telnet Server."
- echo "#define WITHOUT_TELNET" >> src/iroffer_config.h
-fi
-
 if $OPT_NOBLOWFISH; then
  ${msg} "${irt_configure} ${irt_without} ${irt_blowfish}."
  echo "#define WITHOUT_BLOWFISH" >> src/iroffer_config.h
 fi
 
+if $OPT_NOCHROOT; then
+ ${msg} "${irt_configure} ${irt_without} ${irt_nochroot}."
+ echo "#define NO_CHROOT" >> src/iroffer_config.h
+fi
+
+if $OPT_NOHTTP; then
+ ${msg} "${irt_configure} ${irt_without} HTTP Server."
+ echo "#define WITHOUT_HTTP" >> src/iroffer_config.h
+fi
+
 if $OPT_NOMEMSAVE; then
  ${msg} "${irt_configure} ${irt_without} ${irt_memory}."
  echo "#define WITHOUT_MEMSAVE" >> src/iroffer_config.h
 fi
 
+if $OPT_NOTELNET; then
+ ${msg} "${irt_configure} ${irt_without} Telnet Server."
+ echo "#define WITHOUT_TELNET" >> src/iroffer_config.h
+fi
+
 ${msg} -n "${irt_checkfor} make... "
 makebin="`type make | awk '{print $NF}'`"
 gmakebin="`type gmake | awk '{print $NF}'`"
@@ -1296,18 +1304,6 @@
  echo "#define NO_CRYPT" >> src/iroffer_config.h
 fi
 
-
-${msg} -n "${irt_checkfor} chroot()... "
-echo "${irmain1}
- chroot(\".\");
-${main2}" > config.temp.c
-if $cctype -o config.temp $WARNS $WERROR $CFLAGS config.temp.c $LDFLAGS $libs $LIBS; then
-${msg} "${irt_found}"
-else
-echo "#define NO_CHROOT" >> src/iroffer_config.h
-${msg} "${irt_msssing1}, ${irt_nochroot}."
-fi
-
 ${msg} -n "${irt_seeingif_lib} NSS ${irt_lib_exists} (${irt_for} chroot)... "
 echo "${irmain0}" > config.temp.c
 if $cctype -o config.temp $WARNS $WERROR $CFLAGS config.temp.c $LDFLAGS -lnss_files -lnss_dns $libs $LIBS; then
@@ -1881,7 +1877,7 @@
 	groff -t -e -mandoc -Tps iroffer.1 > doc/iroffer.1.ps
 
 install: all
-	install -o root -g $(INSTALL_GROUP) -m 0755 $(IRNAME)$(CONFIG_EXE) $(DESTDIR)$(INSDIR)/$(IRNAME)$(CONFIG_EXE)
+	install -D -o root -g $(INSTALL_GROUP) -m 0755 $(IRNAME)$(CONFIG_EXE) $(DESTDIR)$(INSDIR)/$(IRNAME)$(CONFIG_EXE)
 
 installuser: all
 	install -m 0755 $(IRNAME)$(CONFIG_EXE) $(DESTDIR)$(INSDIR)/$(IRNAME)$(CONFIG_EXE)
