--- src/iroffer_misc.c	2011-09-04 08:50:18.000000000 +0200
+++ src/iroffer_misc.c.NEW	2012-05-03 18:05:27.163446001 +0200
@@ -726,7 +726,7 @@
       outerror(OUTERROR_TYPE_CRASH, "Unable to Fork: %s", strerror(errno));
    else if (s > 0) {
       /* parent exits */
-      exit(69);
+      exit(0);
       }
 
 /*   struct rlimit r = { 0 }; */
@@ -748,7 +748,7 @@
       outerror(OUTERROR_TYPE_CRASH, "Unable to Fork: %s", strerror(errno));
    else if (s > 0)
       /* parent exits */
-      exit(69);
+      exit(0);
    
    
    /* background continues... */
