From: hasufell <julian.ospald@googlemail.com>
Date: Sat Apr  7 13:44:53 UTC 2012

make build system gentoo-compatible

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -18,8 +18,6 @@
 INCLUDE(CMakeMacroEnsureOutOfSourceBuild)
 
 ################################################################################
-
-INCLUDE(cmake/crosscompile.cmake)
 
 PROJECT(s25client)
 
@@ -106,38 +104,6 @@
 			   build_paths.h)
 CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/build_version.h.cmake"
 			   build_version.h)
-CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/build/preinstall.sh.cmake"
-			   preinstall.sh @ONLY)
-CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/build/postinstall.sh.cmake"
-			   postinstall.sh @ONLY)
-
-################################################################################
-
-ADD_SUBDIRECTORY(version)
-
-ADD_CUSTOM_TARGET(updateversion ALL
-				  COMMAND "${CMAKE_BINARY_DIR}/version/src/version" "${CMAKE_SOURCE_DIR}"
-				  DEPENDS version
-				  WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
-)
-
-ADD_CUSTOM_TARGET(mrproper
-				  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/build/cleanup.sh"
-				  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-)
-
-ADD_CUSTOM_TARGET(create_nightly
-				  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/release/create_release.sh" "nightly" "${CMAKE_SOURCE_DIR}"
-				  WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
-)
-
-#ADD_CUSTOM_TARGET(create_stable
-#				  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/release/create_release.sh" "stable" "${CMAKE_SOURCE_DIR}"
-#				  WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
-#)
-
-# preinstall script
-INSTALL(SCRIPT "${CMAKE_CURRENT_SOURCE_DIR}/cmake/preinstall.cmake")
 
 ################################################################################
 
@@ -148,7 +114,6 @@
 ADD_SUBDIRECTORY(libutil)
 ADD_SUBDIRECTORY(liblobby)
 ADD_SUBDIRECTORY(s-c)
-ADD_SUBDIRECTORY(s25update)
 ADD_SUBDIRECTORY(src)
 
 ################################################################################
@@ -157,7 +122,6 @@
 INSTALL(CODE "INCLUDE(driver/cmake_install.cmake)")
 INSTALL(CODE "INCLUDE(libsiedler2/cmake_install.cmake)")
 INSTALL(CODE "INCLUDE(s-c/cmake_install.cmake)")
-INSTALL(CODE "INCLUDE(s25update/cmake_install.cmake)")
 INSTALL(CODE "INCLUDE(src/cmake_install.cmake)")
 
 # and then dont let him install anything
--- driver/video/SDL/src/CMakeLists.txt
+++ driver/video/SDL/src/CMakeLists.txt
@@ -30,9 +30,9 @@
 	ADD_CUSTOM_COMMAND(TARGET videoSDL POST_BUILD COMMAND ln -fvs SDL/src/${SL_PREFIX}videoSDL${CMAKE_SHARED_LIBRARY_SUFFIX} ../../videoSDL${CMAKE_SHARED_LIBRARY_SUFFIX})
 	
 	INSTALL(TARGETS videoSDL 
-		RUNTIME DESTINATION ${LIBDIR}/driver/video 
-		LIBRARY DESTINATION ${LIBDIR}/driver/video
-		ARCHIVE DESTINATION ${LIBDIR}/driver/video
+		RUNTIME DESTINATION ${LIBDIR} 
+		LIBRARY DESTINATION ${LIBDIR}
+		ARCHIVE DESTINATION ${LIBDIR}
 	)
 ELSE (SDL_FOUND)
 	MESSAGE(WARNING ": SDL library not found: Not building SDL videodriver")
--- driver/video/GLFW/src/CMakeLists.txt
+++ driver/video/GLFW/src/CMakeLists.txt
@@ -17,7 +17,7 @@
 	
 		ADD_CUSTOM_COMMAND(TARGET videoGLFW POST_BUILD COMMAND ln -fvs GLFW/src/${CMAKE_SHARED_LIBRARY_PREFIX}videoGLFW${CMAKE_SHARED_LIBRARY_SUFFIX} ../../videoGLFW${CMAKE_SHARED_LIBRARY_SUFFIX})
 	
-		INSTALL(TARGETS videoGLFW DESTINATION ${LIBDIR}/driver/video)
+		INSTALL(TARGETS videoGLFW DESTINATION ${LIBDIR})
 	ELSE (GLFW_LIBRARY)
 		MESSAGE(WARNING ": GLFW library not found: Not building GLFW videodriver")
 	ENDIF (GLFW_LIBRARY)
--- driver/audio/SDL/src/CMakeLists.txt
+++ driver/audio/SDL/src/CMakeLists.txt
@@ -34,9 +34,9 @@
         ADD_CUSTOM_COMMAND(TARGET audioSDL POST_BUILD COMMAND ln -fvs SDL/src/${SL_PREFIX}audioSDL${CMAKE_SHARED_LIBRARY_SUFFIX} ../../audioSDL${CMAKE_SHARED_LIBRARY_SUFFIX})
 
         INSTALL(TARGETS audioSDL 
-            RUNTIME DESTINATION ${LIBDIR}/driver/audio 
-            LIBRARY DESTINATION ${LIBDIR}/driver/audio
-            ARCHIVE DESTINATION ${LIBDIR}/driver/audio
+            RUNTIME DESTINATION ${LIBDIR} 
+            LIBRARY DESTINATION ${LIBDIR}
+            ARCHIVE DESTINATION ${LIBDIR}
         )
     ELSE (SDLMIXER_FOUND)
         MESSAGE(WARNING ": SDL_mixer library not found: Not building SDL audiodriver")
--- build_paths.h.cmake
+++ build_paths.h.cmake
@@ -41,6 +41,7 @@
 
 //	set game directory (normal $(datadir)/s25rttr/S2)
 //#define GAMEDIR DATADIR"/S2"
+#cmakedefine GAMEDIR "${GAMEDIR}"
 
 //	set driver directory (normal $(libdir)/driver)
 //#define DRIVERDIR LIBDIR"/driver"
--- build_version.h.cmake
+++ build_version.h.cmake
@@ -2,7 +2,7 @@
 #define VERSION_H_INCLUDED
 
 #define WINDOW_TITLE "Return To The Roots"
-#define WINDOW_VERSION "20000211"
-#define WINDOW_REVISION "0"
+#define WINDOW_VERSION "${WINDOW_VERSION}"
+#define WINDOW_REVISION "${WINDOW_REVISION}"
 
 #endif // VERSION_H_INCLUDED
