--- connectivity/source/drivers/postgresql/makefile.mk.orig	2008-11-22 13:45:15.000000000 +0500
+++ connectivity/source/drivers/postgresql/makefile.mk	2008-11-22 13:46:58.000000000 +0500
@@ -94,6 +94,10 @@
 POSTGRESQL_INCLUDES=-I`pg_config --includedir`
 .ENDIF
 
+.IF "$(SYSTEM_POSTGRESQL)" = "YES"
+POSTGRESQL_LIBS=-L`pg_config --libdir`
+.ENDIF
+
 CFLAGS+=$(POSTGRESQL_INCLUDES) \
 	-DPOSTGRESQL_MAJOR=$(POSTGRESQL_MAJOR) \
 	-DPOSTGRESQL_MINOR=$(POSTGRESQL_MINOR) \
@@ -124,6 +128,8 @@
 PQLIB=-lpq -lcrypt
 .IF "$(OS)"=="FREEBSD"
 PQLIB+=-L/usr/local/lib
+.ELSE
+PQLIB+=$(POSTGRESQL_LIBS)
 .ENDIF
 .ENDIF
 SHL2TARGET=postgresql-sdbc-impl.uno
