--- slim-1.2.5/cfg.cpp.orig	2006-08-18 14:18:12.000000000 +0900
+++ slim-1.2.5/cfg.cpp	2006-08-18 14:20:34.000000000 +0900
@@ -154,7 +154,7 @@
     }
     int pos = 0;
     string line = s;
-    string::size_type len = line.length();
+    int len = line.length();
     while ( pos < len && isspace( line[pos] ) ) {
         ++pos;
     }
@@ -240,7 +240,7 @@
     if (sessions.size() <= 1)
         return current;
 
-    for (int i=0; i<sessions.size()-1; i++) {
+    for (int i=0; i<(int)sessions.size()-1; i++) {
         if (current == sessions[i]) {
             return sessions[i+1];
         }
--- slim-1.2.5/image.cpp.orig	2006-08-18 13:39:03.000000000 +0900
+++ slim-1.2.5/image.cpp	2006-08-18 13:39:29.000000000 +0900
@@ -391,7 +391,7 @@
 void Image::Center(const int w, const int h, const char *hex) {
 
     unsigned long packed_rgb;
-    sscanf(hex, "%x", &packed_rgb);  
+    sscanf(hex, "%lx", &packed_rgb);  
 
     unsigned long r = packed_rgb>>16;
     unsigned long g = packed_rgb>>8 & 0xff;
@@ -471,7 +471,7 @@
 void Image::Plain(const int w, const int h, const char *hex) {
 
     unsigned long packed_rgb;
-    sscanf(hex, "%x", &packed_rgb);  
+    sscanf(hex, "%lx", &packed_rgb);  
 
     unsigned long r = packed_rgb>>16;
     unsigned long g = packed_rgb>>8 & 0xff;
--- slim-1.2.5/panel.cpp.orig	2006-08-18 13:49:18.000000000 +0900
+++ slim-1.2.5/panel.cpp	2006-08-18 13:55:55.000000000 +0900
@@ -239,7 +239,7 @@
 
 void Panel::Cursor(int visible) {
     char* text;
-    int xx, yy, x2,y2, cheight;
+    int xx, yy, y2, cheight;
     char* txth = "Wj"; // used to get cursor height
 
     switch(In->GetField()) {
@@ -254,6 +254,12 @@
             xx = input_name_x;
             yy = input_name_y;
             break;
+
+		default: /* Origin & NULL string as default values. */
+			text = (char *)NULL;
+			xx = (int)0;
+			yy = (int)0;
+			break;
     }
 
 
@@ -347,7 +353,7 @@
     del = In->Key(buffer, keysym, singleInputMode);
     Action = In->GetAction();
 
-    XGlyphInfo extents, delextents;
+    XGlyphInfo extents;
     XftDraw *draw = XftDrawCreate(Dpy, Win,
                                   DefaultVisual(Dpy, Scr), DefaultColormap(Dpy, Scr));
 
@@ -397,6 +403,12 @@
             xx = input_name_x;
             yy = input_name_y;
             break;
+
+		default: /* Origin & NULL string as default values. */
+			text = (char *)NULL;
+			xx = (int)0;
+			yy = (int)0;
+			break;
     }
 
     char* txth = "Wj"; // get proper maximum height ?
@@ -431,7 +443,6 @@
 // Draw welcome and "enter username" message
 void Panel::ShowText(){
     string cfgX, cfgY;
-    int n=-1;
     XGlyphInfo extents;
 
     bool singleInputMode =
--- slim-1.2.5/switchuser.cpp.orig	2006-08-18 20:24:00.000000000 +0900
+++ slim-1.2.5/switchuser.cpp	2006-08-18 20:25:12.000000000 +0900
@@ -14,8 +14,8 @@
 using namespace std;
 
 SwitchUser::SwitchUser(struct passwd *pw, Cfg *c, const string& display)
-    : Pw(pw),
-      cfg(c),
+    : cfg(c),
+      Pw(pw),
       displayName(display)
 {
 }

