--- Makefile.orig	2006-07-24 18:40:12.000000000 +0200
+++ Makefile	2006-08-07 14:30:36.000000000 +0200
@@ -5,13 +5,12 @@
 #######################################################
 CXX=/usr/bin/g++
 CC=/usr/bin/gcc
-CFLAGS=-I. -I/usr/X11R6/include -I/usr/include/freetype2 -I/usr/include/freetype2/config -I/usr/include/libpng12 -I/usr/include
-CXXLAGS=$(CFLAGS)
-LDFLAGS=-L/usr/X11R6/lib -lXft -lX11 -lfreetype -lXrender -lfontconfig -lpng12 -lz -lm -lcrypt -lXpm -lXmu -lpng -ljpeg
+INCLUDES=-I. -I/usr/include/freetype2 -I/usr/include/freetype2/config -I/usr/include/libpng12
+MYLDFLAGS=-lXft -lX11 -lfreetype -lXrender -lfontconfig -lpng12 -lz -lm -lcrypt -lXpm -lXmu -lpng -ljpeg
 CUSTOM=-DHAVE_SHADOW
 PREFIX=/usr
 CFGDIR=/etc
-MANDIR=/usr/man
+MANDIR=/usr/share/man
 DESTDIR=
 #######################################################
 
@@ -26,13 +25,13 @@
 all: slim
 
 slim: $(OBJECTS)
-	$(CXX) $(LDFLAGS) $(OBJECTS) -o $(NAME)
+	$(CXX) $(LDFLAGS) $(MYLDFLAGS) $(OBJECTS) -o $(NAME)
 
 .cpp.o:
-	$(CXX) $(CFLAGS) $(DEFINES) $(CUSTOM) -c $< -o $@
+	$(CXX) $(CFLAGS) $(INCLUDES) $(DEFINES) $(CUSTOM) -c $< -o $@
 
 .c.o:
-	$(CC) $(CFLAGS) $(DEFINES) $(CUSTOM) -c $< -o $@
+	$(CC) $(CFLAGS) $(INCLUDES) $(DEFINES) $(CUSTOM) -c $< -o $@
 
 install: slim install-theme
 	install -D -m 755 slim $(DESTDIR)$(PREFIX)/bin/slim
