--- Makefile	2007-10-30 17:39:43.000000000 +0100
+++ Makefile.new	2009-04-10 21:45:42.000000000 +0200
@@ -4,11 +4,10 @@
 #  ---------------------------------------------
 
 BIN     = yacpi
-prefix  = /usr/local
+prefix  = /usr/
 INSPATH = ${prefix}/bin/
-CFLAGS  = -O2 -Wall -g -DVERSION=\"${VERSION}\"
-CC      = cc
-DOCPATH = ${prefix}/share/doc/yacpi
+CFLAGS  += -DVERSION=\"${VERSION}\"
+DOCPATH = ${prefix}/share/doc/yacpi-${VERSION}
 MANPATH = ${prefix}/share/man/man1
 VERSION = 3.0.1
 SRC_yacpi       = yacpi.c
@@ -17,13 +16,10 @@
 OBJ_get_cpu     = ${SRC_get_cpu:.c=.o}
 
 .c.o :
-	@echo CC $<
-	@${CC} -c ${CFLAGS} $<
+	${CC} -c ${CFLAGS} $<
 
-all : options ${OBJ_yacpi} ${OBJ_get_cpu}
-	@${CC} -Wall get_cpu.o yacpi.o -o ${BIN} -lncurses -lacpi
-	@strip ${BIN}
-	@echo built yacpi
+all : ${OBJ_yacpi} ${OBJ_get_cpu}
+	${CC} $(LDFLAGS) get_cpu.o yacpi.o -o ${BIN} -lncurses -lacpi
 
 dist : clean
 	@mkdir -p yacpi-${VERSION}
@@ -36,12 +36,12 @@
 	@echo "BIN    = ${BIN}"
 
 install :
-	@mkdir -p ${DOCPATH}
-	@mkdir -p ${INSPATH}
-	@mkdir -p ${MANPATH}
-	install -m644 CHANGELOG README THANKS COPYING ${DOCPATH}
-	install -m644 yacpi.1 ${MANPATH}
-	install ${BIN} ${INSPATH}
+	@mkdir -p $(DESTDIR)/${DOCPATH}
+	@mkdir -p $(DESTDIR)/${INSPATH}
+	@mkdir -p $(DESTDIR)/${MANPATH}
+	install -m644 CHANGELOG README THANKS $(DESTDIR)/${DOCPATH}
+	install -m644 yacpi.1 $(DESTDIR)/${MANPATH}
+	install ${BIN} $(DESTDIR)/${INSPATH}
 
 uninstall :
 	rm -f ${INSPATH}${BIN}
