--- tkdnd-1.0a2/unix/Makefile.in	2001-02-23 10:00:28.000000000 +0100
+++ tkdnd-1.0a2/unix/Makefile.in	2006-06-21 22:38:57.000000000 +0200
@@ -296,7 +296,7 @@
 	    $(RANLIB) $(DESTDIR)$(bindir)/$$p; \
 	  else :; fi; \
 	done
-	$(TCLSH_PROG) mkIndex.tcl
+	$(TCLSH_PROG) mkIndex.tcl $(DESTDIR)$(pkglibdir)
 
 #========================================================================
 # Install binary executables (e.g. .exe files)
--- tkdnd-1.0a2/unix/mkIndex.tcl.in	2000-09-09 03:16:12.000000000 +0200
+++ tkdnd-1.0a2/unix/mkIndex.tcl.in	2006-06-21 22:35:37.000000000 +0200
@@ -88,9 +88,9 @@
 set package_dir [Nativepath [file dirname [pwd]]]
 cd $libdir

-set dest_dir [file join $libdir $package]
+set dest_dir [lindex $::argv 0]
 ## Everything is ready, just copy to proper location...
 foreach file {libtkdnd.dll pkgIndex.tcl tkDND_Utils.tcl} {
-  puts "Copying $file $dest_dir"
+  puts "Moving [file join $package_dir lib $package $file] to $dest_dir"
-  file copy -force [file join $package_dir lib $package $file] $dest_dir
+  file rename -force [file join $package_dir lib $package $file] $dest_dir
 }
