diff -urN gizmod-3.5/cmake/Modules/FindLibvisual.cmake gizmod-3.5-fixed/cmake/Modules/FindLibvisual.cmake
--- gizmod-3.5/cmake/Modules/FindLibvisual.cmake	2008-08-24 12:45:54.000000000 -0400
+++ gizmod-3.5-fixed/cmake/Modules/FindLibvisual.cmake	2009-11-26 16:08:22.000000000 -0500
@@ -12,43 +12,48 @@
 
 set(_visual "visual-0.4")
 
-if(NOT WIN32)
-    # use pkg-config to get the directories and then use these values
-    # in the FIND_PATH() and FIND_LIBRARY() calls
-    INCLUDE(UsePkgConfig)
-    INCLUDE(UsePkgConfigVariable)
-  
-    # get info from pkgconfig
-    PKGCONFIG(lib${_visual} _LIBVISUALIncDir _LIBVISUALLinkDir _LIBVISUALLinkFlags _LIBVISUALCflags)
-    set(LIBVISUAL_DEFINITIONS ${_LIBVISUALCflags})
-    
-    # get plugins dir
-    PKGCONFIG_VAR(lib${_visual} "pluginsbasedir" _LIBVISUALPluginDir)
-    set(LIBVISUAL_PLUGINS_DIR ${_LIBVISUALPluginDir})
-endif(NOT WIN32)
-FIND_PATH(LIBVISUAL_INCLUDE_DIR libvisual/libvisual.h
-    ${_LIBVISUALIncDir}
-    /usr/include
-    /usr/local/include
-)
+if(BUILD_VIS_PLUGIN)
 
-FIND_LIBRARY(LIBVISUAL_LIBRARIES NAMES ${_visual}
-    PATHS
-    ${_LIBVISUALLinkDir}
-    /usr/lib
-    /usr/local/lib
-)
-  
-if (LIBVISUAL_INCLUDE_DIR AND LIBVISUAL_LIBRARIES)
-     set(LIBVISUAL_FOUND TRUE)
-endif (LIBVISUAL_INCLUDE_DIR AND LIBVISUAL_LIBRARIES)
-  
-if (NOT LIBVISUAL_FOUND)
-    if (Libvisual_FIND_REQUIRED)
-      message(FATAL_ERROR "Could NOT find libvisual")
-    else (Libvisual_FIND_REQUIRED)
-      message(STATUS "Could NOT find libvisual")
-    endif (Libvisual_FIND_REQUIRED)
-endif (NOT LIBVISUAL_FOUND)
-  
-MARK_AS_ADVANCED(LIBVISUAL_INCLUDE_DIR LIBVISUAL_LIBRARIES)
+	if(NOT WIN32)
+	    # use pkg-config to get the directories and then use these values
+	    # in the FIND_PATH() and FIND_LIBRARY() calls
+	    INCLUDE(UsePkgConfig)
+	    INCLUDE(UsePkgConfigVariable)
+	  
+	    # get info from pkgconfig
+	    PKGCONFIG(lib${_visual} _LIBVISUALIncDir _LIBVISUALLinkDir _LIBVISUALLinkFlags _LIBVISUALCflags)
+	    set(LIBVISUAL_DEFINITIONS ${_LIBVISUALCflags})
+	    
+	    # get plugins dir
+	    PKGCONFIG_VAR(lib${_visual} "pluginsbasedir" _LIBVISUALPluginDir)
+	    set(LIBVISUAL_PLUGINS_DIR ${_LIBVISUALPluginDir})
+	endif(NOT WIN32)
+	FIND_PATH(LIBVISUAL_INCLUDE_DIR libvisual/libvisual.h
+	    ${_LIBVISUALIncDir}
+	    /usr/include
+	    /usr/local/include
+	)
+
+	FIND_LIBRARY(LIBVISUAL_LIBRARIES NAMES ${_visual}
+	    PATHS
+	    ${_LIBVISUALLinkDir}
+	    /usr/lib
+	    /usr/local/lib
+	)
+	  
+	if (LIBVISUAL_INCLUDE_DIR AND LIBVISUAL_LIBRARIES)
+	     set(LIBVISUAL_FOUND TRUE)
+	endif (LIBVISUAL_INCLUDE_DIR AND LIBVISUAL_LIBRARIES)
+	  
+	if (NOT LIBVISUAL_FOUND)
+	    if (Libvisual_FIND_REQUIRED)
+	      message(FATAL_ERROR "Could NOT find libvisual")
+	    else (Libvisual_FIND_REQUIRED)
+	      message(STATUS "Could NOT find libvisual")
+	    endif (Libvisual_FIND_REQUIRED)
+	else (NOT LIBVISUAL_FOUND)
+	    message(STATUS "Found libvisual: ${LIBVISUAL_LIBRARIES}")
+	endif (NOT LIBVISUAL_FOUND)
+	  
+	MARK_AS_ADVANCED(LIBVISUAL_INCLUDE_DIR LIBVISUAL_LIBRARIES)
+endif(BUILD_VIS_PLUGIN)
diff -urN gizmod-3.5/ConfigureChecks.cmake gizmod-3.5-fixed/ConfigureChecks.cmake
--- gizmod-3.5/ConfigureChecks.cmake	2008-11-03 07:45:04.000000000 -0500
+++ gizmod-3.5-fixed/ConfigureChecks.cmake	2009-11-26 15:59:44.000000000 -0500
@@ -46,7 +46,7 @@
 
 # set default build type to release
 if (NOT BUILD_VIS_PLUGIN)
-	set(BUILD_VIS_PLUGIN True CACHE BOOLEAN TRUE FORCE)
+	set(BUILD_VIS_PLUGIN False CACHE BOOLEAN FALSE FORCE)
 endif (NOT BUILD_VIS_PLUGIN)
 
 #################################
