=== modified file 'mixxx/build/mixxx.py'
--- mixxx/build/mixxx.py	2011-06-13 01:29:39 +0000
+++ mixxx/build/mixxx.py	2011-06-17 17:37:59 +0000
@@ -188,7 +188,7 @@
         # Should cover {Net,Open,Free,DragonFly}BSD, but only tested on OpenBSD
         if 'bsd' in sys.platform:
             return 'bsd'
-        if 'linux2' == sys.platform:
+        if sys.platform in ['linux2', 'linux3']:
             return 'linux'
         if sys.platform == 'darwin':
             return 'osx'
--- mixxx/build/qt4.py	2011-09-15 01:05:21.872433812 +0100
+++ mixxx/build/qt4.py	2011-09-15 01:06:34.628431259 +0100
@@ -409,7 +409,7 @@
 		self.AppendUnique(CPPFLAGS=['-DQT_GUI_LIB'])
 
 	debugSuffix = ''
-	if sys.platform == "linux2" :
+	if sys.platform in ["linux2", "linux3"]:
 		if debug : debugSuffix = '_debug'
 		for module in modules :
 			if module not in pclessModules : continue
