--- Makefile.orig	2007-12-12 10:11:42.377646285 -0500
+++ Makefile	2007-12-12 10:25:46.425745675 -0500
@@ -72,7 +72,9 @@
 
 # Install
 install: all
-	install -t ${BINDIR} pngnq pngcomp
-	install -t ${MANDIR} pngnq.1 
-	install -d ${DOCDIR} 
-	install -t ${DOCDIR} README LICENSE README.pngcomp 
+	install -d ${DESTDIR}/${BINDIR}
+	install -t ${DESTDIR}/${BINDIR} pngnq pngcomp
+	install -d ${DESTDIR}/${MANDIR}
+	install -t ${DESTDIR}/${MANDIR} pngnq.1 
+	install -d ${DESTDIR}/${DOCDIR} 
+	install -t ${DESTDIR}/${DOCDIR} README LICENSE README.pngcomp 
