--- nurbs/Makefile.am	2009-07-07 22:32:45.000000000 +1000
+++ nurbs/Makefile.am.new	2009-07-07 22:33:16.000000000 +1000
@@ -44,7 +44,7 @@
 
 lib_LTLIBRARIES = libnurbsf.la libnurbsd.la
 libnurbsf_la_SOURCES = $(float_sources)
-libnurbsf_la_LDFLAGS = $(ldflags)
+libnurbsf_la_LDFLAGS = $(ldflags) -lmatrix -lmatrixN -lmatrixI
 libnurbsd_la_SOURCES = $(double_sources)
-libnurbsd_la_LDFLAGS = $(ldflags)
+libnurbsd_la_LDFLAGS = $(ldflags) -lmatrix -lmatrixN -lmatrixI
 
--- image/Makefile.am	2002-05-14 06:04:34.000000000 +1000
+++ image/Makefile.am.new	2009-07-09 00:21:55.000000000 +1000
@@ -9,4 +9,4 @@
 
 lib_LTLIBRARIES = libmatrixI.la
 libmatrixI_la_SOURCES = color.cpp image_.cpp  rec_filter_.cpp filter_.cpp
-libmatrixI_la_LDFLAGS = -version-info 1:0:0
+libmatrixI_la_LDFLAGS = -version-info 1:0:0 -lmatrix

--- numerical/Makefile.am	2002-05-14 06:04:38.000000000 +1000
+++ numerical/Makefile.am.new	2009-07-09 00:22:24.000000000 +1000
@@ -6,5 +6,5 @@
 
 lib_LTLIBRARIES = libmatrixN.la
 libmatrixN_la_SOURCES = matrixMat_.cpp fft_.cpp chebexp_.cpp intccq_.cpp statistic_.cpp
-libmatrixN_la_LDFLAGS = -version-info 1:0:0
+libmatrixN_la_LDFLAGS = -version-info 1:0:0 -lmatrix
 
--- Doxyfile.in    2009-11-11 09:20:32.000000000 +0100
+++ Doxyfile.in.new       2009-11-11 09:20:14.000000000 +0100
@@ -123,7 +123,7 @@
 # and *.h) to filter out the source-files in the directories. If left
 # blank all files are included.

-FILE_PATTERNS    = *.hh *.cc
+FILE_PATTERNS    =

 # The EXAMPLE_PATH tag can be used to specify one or more files or
 # directories that contain example code fragments that are included (see
