diff -ru tapestrea-0.1.0.3~/scripting/chuck-1.2.0.6/src/makefile.alsa tapestrea-0.1.0.3/scripting/chuck-1.2.0.6/src/makefile.alsa
--- tapestrea-0.1.0.3~/scripting/chuck-1.2.0.6/src/makefile.alsa	2006-08-15 19:39:11.000000000 +0200
+++ tapestrea-0.1.0.3/scripting/chuck-1.2.0.6/src/makefile.alsa	2006-08-15 19:42:16.000000000 +0200
@@ -1,9 +1,11 @@
 
-CXX=gcc
+CXX?=g++
+CC?=gcc
 LEX=flex
 YACC=bison
 INCLUDES=
-FLAGS=-D__LINUX_ALSA__ -O3 -c
+CFLAGS?=-O3
+FLAGS=-D__LINUX_ALSA__ -c $(CFLAGS)
 LIBS=-lasound -lstdc++ -ldl -lm
 SF_OBJ=util_sndfile.o
 
@@ -46,7 +48,7 @@
 	$(CXX) -o chuck $(OBJS) $(LIBS)
 
 chuck.tab.o: chuck.tab.c
-	$(CXX) $(FLAGS) chuck.tab.c
+	$(CC) $(FLAGS) chuck.tab.c
 
 chuck.tab.c: chuck.y
 	$(YACC) -dv -b chuck chuck.y
@@ -55,7 +57,7 @@
 	echo "chuck.tab.h was created at the same time as chuck.tab.c"
 
 chuck.yy.o: chuck.yy.c chuck.tab.h chuck_errmsg.h chuck_utils.h
-	$(CXX) $(FLAGS) chuck.yy.c
+	$(CC) $(FLAGS) chuck.yy.c
 
 chuck.yy.c: chuck.lex
 	$(LEX) -ochuck.yy.c chuck.lex
@@ -178,19 +180,19 @@
 	$(CXX) $(FLAGS) util_console.cpp
 
 util_hid.o: util_hid.h util_hid.c
-	$(CXX) $(FLAGS) util_hid.c
+	$(CC) $(FLAGS) util_hid.c
 
 util_math.o: util_math.h util_math.c
-	$(CXX) $(FLAGS) util_math.c
+	$(CC) $(FLAGS) util_math.c
 
 util_network.o: util_network.h util_network.c
-	$(CXX) $(FLAGS) util_network.c
+	$(CC) $(FLAGS) util_network.c
 
 util_opsc.o: util_opsc.h util_opsc.cpp
 	$(CXX) $(FLAGS) util_opsc.cpp
 
 util_raw.o: util_raw.h util_raw.c
-	$(CXX) $(FLAGS) util_raw.c
+	$(CC) $(FLAGS) util_raw.c
 
 util_string.o: util_string.h util_string.cpp
 	$(CXX) $(FLAGS) util_string.cpp
@@ -199,10 +201,10 @@
 	$(CXX) $(FLAGS) util_thread.cpp
 
 util_xforms.o: util_xforms.h util_xforms.c
-	$(CXX) $(FLAGS) util_xforms.c
+	$(CC) $(FLAGS) util_xforms.c
 
 util_sndfile.o: util_sndfile.h util_sndfile.c
-	$(CXX) $(FLAGS) util_sndfile.c
+	$(CC) $(FLAGS) util_sndfile.c
 
 clean: 
 	rm -f chuck *~ *.o chuck.output chuck.tab.h chuck.tab.c chuck.yy.c
diff -ru tapestrea-0.1.0.3~/scripting/chuck-1.2.0.6/src/makefile.jack tapestrea-0.1.0.3/scripting/chuck-1.2.0.6/src/makefile.jack
--- tapestrea-0.1.0.3~/scripting/chuck-1.2.0.6/src/makefile.jack	2006-08-15 19:39:11.000000000 +0200
+++ tapestrea-0.1.0.3/scripting/chuck-1.2.0.6/src/makefile.jack	2006-08-15 19:43:22.000000000 +0200
@@ -1,9 +1,11 @@
 
-CXX=gcc
+CXX?=g++
+CC?=gcc
 LEX=flex
 YACC=bison
 INCLUDES=
-FLAGS=-D__LINUX_JACK__ -O3 -c
+CFLAGS?=-O3
+FLAGS=-D__LINUX_JACK__ -c $(CFLAGS)
 LIBS=-lasound -ljack -lstdc++ -ldl -lm
 SF_OBJ=util_sndfile.o
 
@@ -46,7 +48,7 @@
 	$(CXX) -o chuck $(OBJS) $(LIBS)
 
 chuck.tab.o: chuck.tab.c
-	$(CXX) $(FLAGS) chuck.tab.c
+	$(CC) $(FLAGS) chuck.tab.c
 
 chuck.tab.c: chuck.y
 	$(YACC) -dv -b chuck chuck.y
@@ -55,7 +57,7 @@
 	echo "chuck.tab.h was created at the same time as chuck.tab.c"
 
 chuck.yy.o: chuck.yy.c chuck.tab.h chuck_errmsg.h chuck_utils.h
-	$(CXX) $(FLAGS) chuck.yy.c
+	$(CC) $(FLAGS) chuck.yy.c
 
 chuck.yy.c: chuck.lex
 	$(LEX) -ochuck.yy.c chuck.lex
@@ -178,19 +180,19 @@
 	$(CXX) $(FLAGS) util_console.cpp
 
 util_hid.o: util_hid.h util_hid.c
-	$(CXX) $(FLAGS) util_hid.c
+	$(CC) $(FLAGS) util_hid.c
 
 util_math.o: util_math.h util_math.c
-	$(CXX) $(FLAGS) util_math.c
+	$(CC) $(FLAGS) util_math.c
 
 util_network.o: util_network.h util_network.c
-	$(CXX) $(FLAGS) util_network.c
+	$(CC) $(FLAGS) util_network.c
 
 util_opsc.o: util_opsc.h util_opsc.cpp
 	$(CXX) $(FLAGS) util_opsc.cpp
 
 util_raw.o: util_raw.h util_raw.c
-	$(CXX) $(FLAGS) util_raw.c
+	$(CC) $(FLAGS) util_raw.c
 
 util_string.o: util_string.h util_string.cpp
 	$(CXX) $(FLAGS) util_string.cpp
@@ -199,10 +201,10 @@
 	$(CXX) $(FLAGS) util_thread.cpp
 
 util_xforms.o: util_xforms.h util_xforms.c
-	$(CXX) $(FLAGS) util_xforms.c
+	$(CC) $(FLAGS) util_xforms.c
 
 util_sndfile.o: util_sndfile.h util_sndfile.c
-	$(CXX) $(FLAGS) util_sndfile.c
+	$(CC) $(FLAGS) util_sndfile.c
 
 clean: 
 	rm -f chuck *~ *.o chuck.output chuck.tab.h chuck.tab.c chuck.yy.c
diff -ru tapestrea-0.1.0.3~/scripting/chuck-1.2.0.6/src/makefile.oss tapestrea-0.1.0.3/scripting/chuck-1.2.0.6/src/makefile.oss
--- tapestrea-0.1.0.3~/scripting/chuck-1.2.0.6/src/makefile.oss	2006-08-15 19:39:11.000000000 +0200
+++ tapestrea-0.1.0.3/scripting/chuck-1.2.0.6/src/makefile.oss	2006-08-15 19:44:27.000000000 +0200
@@ -1,9 +1,11 @@
 
-CXX=gcc
+CXX?=g++
+CC?=gcc
 LEX=flex
 YACC=bison
 INCLUDES=
-FLAGS=-D__LINUX_OSS__ -O3 -c
+CFLAGS?=-O3
+FLAGS=-D__LINUX_OSS__ -c $(CFLAGS)
 LIBS=-lpthread -lstdc++ -ldl -lm
 SF_OBJ=util_sndfile.o
 
@@ -46,7 +48,7 @@
 	$(CXX) -o chuck $(OBJS) $(LIBS)
 
 chuck.tab.o: chuck.tab.c
-	$(CXX) $(FLAGS) chuck.tab.c
+	$(CC) $(FLAGS) chuck.tab.c
 
 chuck.tab.c: chuck.y
 	$(YACC) -dv -b chuck chuck.y
@@ -55,7 +57,7 @@
 	echo "chuck.tab.h was created at the same time as chuck.tab.c"
 
 chuck.yy.o: chuck.yy.c chuck.tab.h chuck_errmsg.h chuck_utils.h
-	$(CXX) $(FLAGS) chuck.yy.c
+	$(CC) $(FLAGS) chuck.yy.c
 
 chuck.yy.c: chuck.lex
 	$(LEX) -ochuck.yy.c chuck.lex
@@ -178,19 +180,19 @@
 	$(CXX) $(FLAGS) util_console.cpp
 
 util_hid.o: util_hid.h util_hid.c
-	$(CXX) $(FLAGS) util_hid.c
+	$(CC) $(FLAGS) util_hid.c
 
 util_math.o: util_math.h util_math.c
-	$(CXX) $(FLAGS) util_math.c
+	$(CC) $(FLAGS) util_math.c
 
 util_network.o: util_network.h util_network.c
-	$(CXX) $(FLAGS) util_network.c
+	$(CC) $(FLAGS) util_network.c
 
 util_opsc.o: util_opsc.h util_opsc.cpp
 	$(CXX) $(FLAGS) util_opsc.cpp
 
 util_raw.o: util_raw.h util_raw.c
-	$(CXX) $(FLAGS) util_raw.c
+	$(CC) $(FLAGS) util_raw.c
 
 util_string.o: util_string.h util_string.cpp
 	$(CXX) $(FLAGS) util_string.cpp
@@ -199,10 +201,10 @@
 	$(CXX) $(FLAGS) util_thread.cpp
 
 util_xforms.o: util_xforms.h util_xforms.c
-	$(CXX) $(FLAGS) util_xforms.c
+	$(CC) $(FLAGS) util_xforms.c
 
 util_sndfile.o: util_sndfile.h util_sndfile.c
-	$(CXX) $(FLAGS) util_sndfile.c
+	$(CC) $(FLAGS) util_sndfile.c
 
 clean: 
 	rm -f chuck *~ *.o chuck.output chuck.tab.h chuck.tab.c chuck.yy.c
diff -ru tapestrea-0.1.0.3~/src/makefile.alsa tapestrea-0.1.0.3/src/makefile.alsa
--- tapestrea-0.1.0.3~/src/makefile.alsa	2006-08-15 19:39:11.000000000 +0200
+++ tapestrea-0.1.0.3/src/makefile.alsa	2006-08-15 19:45:48.000000000 +0200
@@ -1,8 +1,10 @@
 
-CC=gcc
+CC?=gcc
+CXX?=g++
 CHUCK_DIR=../scripting/chuck-1.2.0.6/src
 INCLUDES=
-FLAGS=-D__LINUX_ALSA__ `pkg-config --cflags gtk+-2.0` -O3 -c
+CFLAGS?=-O3
+FLAGS=-D__LINUX_ALSA__ `pkg-config --cflags gtk+-2.0` -c $(CFLAGS)
 LIBS=-lasound -lstdc++ -lm -lsndfile -lGL -lGLU -lglut `pkg-config --libs gtk+-2.0`
 
 TAPS_OBJS= taps_analysis.o taps_birdbrain.o taps_driver.o \
@@ -55,7 +57,7 @@
 
 
 taps: $(OBJS) chuck
-	gcc -o taps $(OBJS) $(LIBS)
+	$(CXX) -o taps $(OBJS) $(LIBS)
 
 chuck:
 	make -C $(CHUCK_DIR) linux-alsa
@@ -66,4 +68,4 @@
 	$(CC) $(FLAGS) $(INCLUDES) $*.c
         
 .cpp.o: $*.h $*.cpp
-	$(CC) $(FLAGS) $(INCLUDES) $*.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) $*.cpp
diff -ru tapestrea-0.1.0.3~/src/makefile.jack tapestrea-0.1.0.3/src/makefile.jack
--- tapestrea-0.1.0.3~/src/makefile.jack	2006-08-15 19:39:11.000000000 +0200
+++ tapestrea-0.1.0.3/src/makefile.jack	2006-08-15 19:46:32.000000000 +0200
@@ -1,8 +1,10 @@
 
-CC=gcc
+CC?=gcc
+CXX?=g++
 CHUCK_DIR=../scripting/chuck-1.2.0.6/src
 INCLUDES=
-FLAGS=-D__LINUX_JACK__ `pkg-config --cflags gtk+-2.0` -O3 -c
+CFLAGS?=-O3
+FLAGS=-D__LINUX_JACK__ `pkg-config --cflags gtk+-2.0` -c $(CFLAGS)
 LIBS=-lasound -lstdc++ -lm -lsndfile -lGL -lGLU -lglut -lpthread -ljack `pkg-config --libs gtk+-2.0`
 
 TAPS_OBJS= taps_analysis.o taps_birdbrain.o taps_driver.o \
@@ -55,7 +57,7 @@
 
 
 taps: $(OBJS) chuck
-	gcc -o taps $(OBJS) $(LIBS)
+	$(CC) -o taps $(OBJS) $(LIBS)
 
 chuck:
 	make -C $(CHUCK_DIR) linux-jack
@@ -66,4 +68,4 @@
 	$(CC) $(FLAGS) $(INCLUDES) $*.c
         
 .cpp.o: $*.h $*.cpp
-	$(CC) $(FLAGS) $(INCLUDES) $*.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) $*.cpp
diff -ru tapestrea-0.1.0.3~/src/makefile.oss tapestrea-0.1.0.3/src/makefile.oss
--- tapestrea-0.1.0.3~/src/makefile.oss	2006-08-15 19:39:11.000000000 +0200
+++ tapestrea-0.1.0.3/src/makefile.oss	2006-08-15 19:47:19.000000000 +0200
@@ -1,8 +1,10 @@
 
-CC=gcc
+CC?=gcc
+CXX?=g++
 CHUCK_DIR=../scripting/chuck-1.2.0.6/src
 INCLUDES=
-FLAGS=-D__LINUX_OSS__ `pkg-config --cflags gtk+-2.0` -O3 -c
+CFLAGS?=-O3
+FLAGS=-D__LINUX_OSS__ `pkg-config --cflags gtk+-2.0` -c $(CFLAGS)
 LIBS=-lstdc++ -lm -lsndfile -lGL -lGLU -lglut -lpthread `pkg-config --libs gtk+-2.0`
 
 TAPS_OBJS= taps_analysis.o taps_birdbrain.o taps_driver.o \
@@ -55,7 +57,7 @@
 
 
 taps: $(OBJS) chuck
-	gcc -o taps $(OBJS) $(LIBS)
+	$(CC) -o taps $(OBJS) $(LIBS)
 
 chuck:
 	make -C $(CHUCK_DIR) linux-oss
@@ -66,4 +68,4 @@
 	$(CC) $(FLAGS) $(INCLUDES) $*.c
         
 .cpp.o: $*.h $*.cpp
-	$(CC) $(FLAGS) $(INCLUDES) $*.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) $*.cpp
