--- GNUmakefile.org	2008-05-12 13:59:01.000000000 +0200
+++ GNUmakefile	2008-05-12 15:04:17.000000000 +0200
@@ -1,6 +1,6 @@
 #DEBUG=1
 ZLIB=1
-prefix=/opt/diet
+prefix=/usr
 BINDIR=${prefix}/bin
 MANDIR=${prefix}/man
 man1dir=$(MANDIR)/man1
@@ -14,8 +14,7 @@
 CROSS=
 #CROSS=i686-mingw32-
 CC=$(CROSS)gcc
-CFLAGS=-pipe -Wall
-LDFLAGS=
+CFLAGS+=-pipe -Wall -I/usr/include/libowfat
 
 path = $(subst :, ,$(PATH))
 diet_path = $(foreach dir,$(path),$(wildcard $(dir)/diet))
