diff -ru olsrd-0.4.10~/gui/linux-gtk/Makefile olsrd-0.4.10/gui/linux-gtk/Makefile
--- olsrd-0.4.10~/gui/linux-gtk/Makefile	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/gui/linux-gtk/Makefile	2006-11-01 17:51:27.000000000 +0100
@@ -1,7 +1,7 @@
-CC=	gcc
+CC?=	gcc
 #CC=     arm-linux-gcc
 
-CFLAGS= -Wall `pkg-config --cflags gtk+-2.0` #-g #to use GDB
+CFLAGS+= -Wall `pkg-config --cflags gtk+-2.0` #-g #to use GDB
 LFLAGS= `pkg-config --libs gtk+-2.0`
 
 SRCS=	src/main.c src/interface.c src/callbacks.c src/ipc.c \
@@ -23,4 +23,5 @@
 	rm -f $(OBJS)
 
 install:
-	cp ./olsrd-gui /usr/bin/
+	mkdir -p ${DESTDIR}/usr/bin
+	cp ./olsrd-gui ${DESTDIR}/usr/bin/
diff -ru olsrd-0.4.10~/make/Makefile.fbsd olsrd-0.4.10/make/Makefile.fbsd
--- olsrd-0.4.10~/make/Makefile.fbsd	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.fbsd	2006-11-01 17:51:05.000000000 +0100
@@ -2,12 +2,12 @@
 # FREEBSD SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?= /usr/local
-SBINDIR = $(INSTALL_PREFIX)/sbin
-ETCDIR = $(INSTALL_PREFIX)/etc
-LIBDIR = $(INSTALL_PREFIX)/lib
-DOCDIR = $(INSTALL_PREFIX)/share/doc
-MANDIR = $(INSTALL_PREFIX)/man
+DESTDIR ?= /usr/local
+SBINDIR = $(DESTDIR)/sbin
+ETCDIR = $(DESTDIR)/etc
+LIBDIR = $(DESTDIR)/lib
+DOCDIR = $(DESTDIR)/share/doc
+MANDIR = $(DESTDIR)/man
 
 SRCS +=		$(wildcard src/bsd/*.c) $(wildcard src/unix/*.c)
 HDRS +=		$(wildcard src/bsd/*.h) $(wildcard src/unix/*.h)
@@ -22,8 +22,8 @@
 
 ifdef LIBNET
 
-CFLAGS +=	-DSPOOF -I$(INSTALL_PREFIX)/include
-LIBS +=		-L$(INSTALL_PREFIX)/lib -lnet
+CFLAGS +=	-DSPOOF -I$(DESTDIR)/include
+LIBS +=		-L$(DESTDIR)/lib -lnet
 
 endif
 
diff -ru olsrd-0.4.10~/make/Makefile.linux olsrd-0.4.10/make/Makefile.linux
--- olsrd-0.4.10~/make/Makefile.linux	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.linux	2006-11-01 17:51:05.000000000 +0100
@@ -2,8 +2,8 @@
 # LINUX SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?=
-LIBDIR =	$(INSTALL_PREFIX)/usr/lib
+DESTDIR ?=
+LIBDIR =	$(DESTDIR)/usr/lib
 
 SRCS += 	$(wildcard src/linux/*.c) $(wildcard src/unix/*.c)
 HDRS +=		$(wildcard src/linux/*.h) $(wildcard src/unix/*.h)
diff -ru olsrd-0.4.10~/make/Makefile.nbsd olsrd-0.4.10/make/Makefile.nbsd
--- olsrd-0.4.10~/make/Makefile.nbsd	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.nbsd	2006-11-01 17:51:05.000000000 +0100
@@ -2,8 +2,8 @@
 # NETBSD SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?=
-LIBDIR =	$(INSTALL_PREFIX)/usr/lib
+DESTDIR ?=
+LIBDIR =	$(DESTDIR)/usr/lib
 
 SRCS +=		$(wildcard src/bsd/*.c) $(wildcard src/unix/*.c)
 HDRS +=		$(wildcard src/bsd/*.h) $(wildcard src/unix/*.h)
diff -ru olsrd-0.4.10~/make/Makefile.obsd olsrd-0.4.10/make/Makefile.obsd
--- olsrd-0.4.10~/make/Makefile.obsd	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.obsd	2006-11-01 17:51:05.000000000 +0100
@@ -2,8 +2,8 @@
 # OPENBSD SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?=
-LIBDIR =	$(INSTALL_PREFIX)/usr/lib
+DESTDIR ?=
+LIBDIR =	$(DESTDIR)/usr/lib
 
 SRCS +=		$(wildcard src/bsd/*.c) $(wildcard src/unix/*.c)
 HDRS +=		$(wildcard src/bsd/*.h) $(wildcard src/unix/*.h)
diff -ru olsrd-0.4.10~/make/Makefile.osx olsrd-0.4.10/make/Makefile.osx
--- olsrd-0.4.10~/make/Makefile.osx	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.osx	2006-11-01 17:51:05.000000000 +0100
@@ -2,8 +2,8 @@
 # MAC OSX SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?=
-LIBDIR =	$(INSTALL_PREFIX)/usr/lib
+DESTDIR ?=
+LIBDIR =	$(DESTDIR)/usr/lib
 
 SRCS +=		$(wildcard src/bsd/*.c) $(wildcard src/unix/*.c)
 HDRS +=		$(wildcard src/bsd/*.h) $(wildcard src/unix/*.h)
diff -ru olsrd-0.4.10~/make/Makefile.win32 olsrd-0.4.10/make/Makefile.win32
--- olsrd-0.4.10~/make/Makefile.win32	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.win32	2006-11-01 17:51:05.000000000 +0100
@@ -2,7 +2,7 @@
 # WINDOWS32 SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?=
+DESTDIR ?=
 EXENAME = olsrd.exe
 
 SRCS +=		$(wildcard src/win32/*.c)
diff -ru olsrd-0.4.10~/make/Makefile.wince olsrd-0.4.10/make/Makefile.wince
--- olsrd-0.4.10~/make/Makefile.wince	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/make/Makefile.wince	2006-11-01 17:51:05.000000000 +0100
@@ -2,8 +2,8 @@
 # WINDOWS CE SPECIFIC CONFIGURATION
 #
 
-INSTALL_PREFIX ?=
-LIBDIR =	$(INSTALL_PREFIX)/usr/lib
+DESTDIR ?=
+LIBDIR =	$(DESTDIR)/usr/lib
 
 SRCS +=		$(wildcard src/win32/*.c)
 HDRS +=		$(wildcard src/win32/*.h)
diff -ru olsrd-0.4.10~/Makefile.inc olsrd-0.4.10/Makefile.inc
--- olsrd-0.4.10~/Makefile.inc	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/Makefile.inc	2006-11-01 18:02:31.000000000 +0100
@@ -1,17 +1,17 @@
 # programs
 CC ?= 		gcc
-STRIP ?=	strip
+STRIP ?=	true	
 BISON ?=	bison
 FLEX ?=		flex
 TAGCMD ?=	etags
 
 # target directories and names
-#INSTALL_PREFIX = 
-SBINDIR = $(INSTALL_PREFIX)/usr/sbin
-ETCDIR = $(INSTALL_PREFIX)/etc
-LIBDIR = $(INSTALL_PREFIX)/usr/lib
-DOCDIR = $(INSTALL_PREFIX)/usr/share/doc
-MANDIR = $(INSTALL_PREFIX)/usr/share/man
+# DESTDIR =
+SBINDIR = $(DESTDIR)/usr/sbin
+ETCDIR = $(DESTDIR)/etc
+LIBDIR = $(DESTDIR)/usr/lib
+DOCDIR = $(DESTDIR)/usr/share/doc
+MANDIR = $(DESTDIR)/usr/share/man
 CFGFILE = $(ETCDIR)/olsrd.conf
 EXENAME = olsrd
 
diff -ru olsrd-0.4.10~/src/cfgparser/Makefile olsrd-0.4.10/src/cfgparser/Makefile
--- olsrd-0.4.10~/src/cfgparser/Makefile	2006-11-01 17:50:34.000000000 +0100
+++ olsrd-0.4.10/src/cfgparser/Makefile	2006-11-01 18:17:04.000000000 +0100
@@ -74,7 +74,7 @@
 LDFLAGS +=	-shared -Wl,-soname,$(LIBNAME)
 endif
 
-CFLAGS +=	-DMAKELIB
+CFLAGS +=	-DMAKELIB -fPIC
 NAME =		$(LIBNAME)
 
 endif
