--- tinyerp-client-3.5.0~/setup.py	2006-11-03 22:53:27.000000000 +0100
+++ tinyerp-client-3.5.0/setup.py	2006-11-03 22:56:30.000000000 +0100
@@ -44,9 +44,9 @@
              (opj('share','doc', 'tinyerp-client-%s' % version), 
               [f for f in glob.glob('doc/*') if os.path.isfile(f)]),
              (opj('share', 'pixmaps', 'tinyerp-client'),
-              glob.glob('pixmaps/*.png') + glob.glob('bin/*.png')),
+              glob.glob('pixmaps/*.png')),
              (opj('share', 'tinyerp-client'),
-              ['bin/terp.glade', 'bin/tipoftheday.txt'])]
+              ['bin/terp.glade', 'bin/tipoftheday.txt'] + glob.glob('bin/*.png'))]
     return files
 
 included_plugins = ['workflow_print']
@@ -81,7 +81,7 @@
 Programming Language :: Python
 """
 
-check_modules()
+#check_modules()
 
 # create startup script
 start_script = \
