--- tinyerp-client-3.4.2~/bin/options.py	2006-10-17 20:35:15.000000000 +0200
+++ tinyerp-client-3.4.2/bin/options.py	2006-10-17 20:35:59.000000000 +0200
@@ -65,8 +65,8 @@
 			'client.modepda': False,
 			'client.toolbar': 'both',
 			'client.theme': 'none',
-			'path.share': os.path.dirname(__file__),
-			'path.pixmaps': os.path.dirname(__file__),
+			'path.share': '/usr/share/tinyerp-client',
+			'path.pixmaps': '/usr/share/pixmaps/tinyerp-client',
 			'tip.autostart': True,
 			'tip.position': 0,
 			'survey.position': 0,
