--- a/doc/CMakeLists.txt	2011-04-11 20:08:30.871957979 -0400
+++ b/doc/CMakeLists.txt	2011-04-11 20:16:24.765291311 -0400
@@ -25,7 +25,7 @@
     TARGET ManPages 
     SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/e4rat.conf.pod 
     COMMAND pod2man ARGS ${CMAKE_CURRENT_SOURCE_DIR}/e4rat.conf.pod
-                         ${CMAKE_CURRENT_SOURCE_DIR}/e4rat.conf.5
+                         ${CMAKE_CURRENT_BINARY_DIR}/e4rat.conf.5
   	  	         --section 5
                          --center "User Manuals"
                          --release "e4rat ${E4RAT_VERSION}"
