
--- a/libs/owutil/thread/RecursiveMutex.h	Mon Jan 28 20:19:41 2008 +0100
+++ b/libs/owutil/thread/RecursiveMutex.h	Tue Sep 09 11:32:52 2008 +0200
@@ -21,6 +21,7 @@
 #define OWRECURSIVEMUTEX_H
 
 #include <boost/thread/recursive_mutex.hpp>
+#include <boost/version.hpp>
 
 /**
  * Recursive Mutex helper for Boost.
@@ -33,6 +34,7 @@
 
 	typedef scoped_lock ScopedLock;
 
+#if BOOST_VERSION < 103500
 	void lock() {
 		boost::detail::thread::lock_ops<boost::recursive_mutex>::lock(*this);
 	}
@@ -40,6 +42,7 @@
 	void unlock() {
 		boost::detail::thread::lock_ops<boost::recursive_mutex>::unlock(*this);
 	}
+#endif
 
 };
 




