diff -r -u MyPasswordSafe-20060326/src/mypasswordsafe.ui MyPasswordSafe-20060326-patch/src/mypasswordsafe.ui
--- MyPasswordSafe-20060326/src/mypasswordsafe.ui	2005-12-17 12:33:21.000000000 +0100
+++ MyPasswordSafe-20060326-patch/src/mypasswordsafe.ui	2006-06-17 18:52:43.000000000 +0200
@@ -1024,7 +1024,4 @@
 </functions>
 <pixmapinproject/>
 <layoutdefaults spacing="6" margin="11"/>
-<includehints>
-    <includehint>safelistview.h</includehint>
-</includehints>
 </UI>
