diff -Naur rbutilqt.orig/rbutilqt-lang.qrc rbutilqt/rbutilqt-lang.qrc
--- rbutilqt.orig/rbutilqt-lang.qrc	2010-11-03 12:37:27.231077390 +0100
+++ rbutilqt/rbutilqt-lang.qrc	2010-11-03 12:37:53.904077392 +0100
@@ -1,19 +1,19 @@
 <RCC>
     <qresource prefix="/" >
-        <file>lang/rbutil_cs.qm</file>
-        <file>lang/rbutil_de.qm</file>
-        <file>lang/rbutil_fi.qm</file>
-        <file>lang/rbutil_fr.qm</file>
-        <file>lang/rbutil_gr.qm</file>
-        <file>lang/rbutil_he.qm</file>
-        <file>lang/rbutil_it.qm</file>
-        <file>lang/rbutil_ja.qm</file>
-        <file>lang/rbutil_nl.qm</file>
-        <file>lang/rbutil_pl.qm</file>
-        <file>lang/rbutil_pt.qm</file>
-        <file>lang/rbutil_pt_BR.qm</file>
-        <file>lang/rbutil_ru.qm</file>
-        <file>lang/rbutil_zh_CN.qm</file>
-        <file>lang/rbutil_zh_TW.qm</file>
+        <file>lang/rbutil_cs.ts</file>
+        <file>lang/rbutil_de.ts</file>
+        <file>lang/rbutil_fi.ts</file>
+        <file>lang/rbutil_fr.ts</file>
+        <file>lang/rbutil_gr.ts</file>
+        <file>lang/rbutil_he.ts</file>
+        <file>lang/rbutil_it.ts</file>
+        <file>lang/rbutil_ja.ts</file>
+        <file>lang/rbutil_nl.ts</file>
+        <file>lang/rbutil_pl.ts</file>
+        <file>lang/rbutil_pt.ts</file>
+        <file>lang/rbutil_pt_BR.ts</file>
+        <file>lang/rbutil_ru.ts</file>
+        <file>lang/rbutil_zh_CN.ts</file>
+        <file>lang/rbutil_zh_TW.ts</file>
         </qresource>
 </RCC>
