--- sic-0.2~/config.mk	2006-07-21 22:16:00.000000000 +0200
+++ sic-0.2/config.mk	2006-07-21 22:17:14.000000000 +0200
@@ -1,8 +1,8 @@
 # Customize to fit your system
 
 # paths
-PREFIX = /usr/local
-MANPREFIX = ${PREFIX}/share/man
+PREFIX ?= /usr/local
+MANPREFIX ?= ${PREFIX}/share/man
 
 VERSION = 0.2
 
@@ -10,7 +10,7 @@
 LIBS = -L${PREFIX}/lib -L/usr/lib -lc
 
 # compiler
-CFLAGS      = -O3 -I${PREFIX}/include -I/usr/include \
+CFLAGS      += -I${PREFIX}/include -I/usr/include \
 			-DVERSION=\"${VERSION}\"
 LDFLAGS     = ${LIBS}
 #CFLAGS      = -g -Wall -O2 -I${PREFIX}/include -I/usr/include \
