--- ../configure	2008-08-27 23:42:25.000000000 +0300
+++ ./configure	2008-09-05 15:31:30.403065399 +0300
@@ -2372,7 +2372,10 @@
 echo $ECHO_N "checking whether to enable maintainer-specific portions of Makefiles... $ECHO_C" >&6; }
     # Check whether --enable-maintainer-mode was given.
 if test "${enable_maintainer_mode+set}" = set; then
-  enableval=$enable_maintainer_mode; USE_MAINTAINER_MODE=$enableval
+  enableval=$enable_maintainer_mode; 
+    if test "$enableval" != "no"; then   
+  USE_MAINTAINER_MODE=$enableval
+    fi
 else
   USE_MAINTAINER_MODE=no
 fi
@@ -11109,8 +11112,10 @@
 
 # Check whether --enable-static-daemon was given.
 if test "${enable_static_daemon+set}" = set; then
-  enableval=$enable_static_daemon; LDFLAGS_DAEMON=-static
-
+  enableval=$enable_static_daemon; 
+    if test "x$enableval" != "xno"; then   
+  LDFLAGS_DAEMON=-static
+    fi
 fi
 
 
@@ -11131,10 +11136,11 @@
 # Check whether --enable-icmp was given.
 if test "${enable_icmp+set}" = set; then
   enableval=$enable_icmp;
+    if test "x$enableval" != "xno"; then   
 cat >>confdefs.h <<_ACEOF
 #define ENABLE_ICMP 1
 _ACEOF
-
+    fi
 
 fi
 
@@ -11142,9 +11148,11 @@
 # Check whether --enable-tcp was given.
 if test "${enable_tcp+set}" = set; then
   enableval=$enable_tcp;
+    if test "x$enableval" != "xno"; then 
 cat >>confdefs.h <<_ACEOF
 #define ENABLE_TCP 1
 _ACEOF
+    fi
 
 
 fi
@@ -11153,6 +11161,7 @@
 # Check whether --enable-dns was given.
 if test "${enable_dns+set}" = set; then
   enableval=$enable_dns;
+  if test "x$enableval" != "xno"; then 
     if test "${ac_cv_header_gmp_h+set}" = set; then
   { echo "$as_me:$LINENO: checking for gmp.h" >&5
 echo $ECHO_N "checking for gmp.h... $ECHO_C" >&6; }
@@ -11362,7 +11372,7 @@
 #define ENABLE_DNS 1
 _ACEOF
 
-
+fi
 
 fi
 
