--- config.mak.orig	2007-07-27 03:51:04.000000000 -0400
+++ config.mak	2007-07-27 06:43:43.000000000 -0400
@@ -12,26 +12,20 @@
 includedir=/usr/include
 endif
 
-libdir=/usr/local/lib
-
 ifndef libdir
 libdir=/usr/lib
 endif
 
-DESTDIR=
-
-BINDIR=/usr/local/bin
-
 ifndef BINDIR
 BINDIR=/usr/bin
 endif
 
-INSTALL=/usr/bin/install -c
+INSTALL=/usr/bin/install -D
 INSTALL_PROGRAM=${INSTALL}
 INSTALL_DATA=${INSTALL} -m 644
 
 ifndef INSTALL
-INSTALL=install -c
+INSTALL=install -D
 INSTALL_PROGRAM=${INSTALL}
 INSTALL_DATA=${INSTALL} -m 644
 endif
