--- setup.py	2006-05-02 00:38:42.000000000 -0400
+++ setup.py.new	2006-05-02 00:39:28.000000000 -0400
@@ -157,8 +157,8 @@
 
 #### MozillaBrowser Extension ####
 mozilla_browser_options = parsePkgConfig("pkg-config" , 
-        "gtk+-2.0 glib-2.0 pygtk-2.0 mozilla-gtkmozembed mozilla-xpcom")
-mozilla_lib_path = parsePkgConfig('pkg-config', 'mozilla-gtkmozembed')['library_dirs']
+        "gtk+-2.0 glib-2.0 pygtk-2.0 gecko-sdk-gtkmozembed gecko-sdk-xpcom")
+mozilla_lib_path = parsePkgConfig('pkg-config', 'gecko-sdk-gtkmozembed')['library_dirs']
 mozilla_browser_ext = Extension("democracy.MozillaBrowser",
         [ os.path.join(frontend_implementation_dir,'MozillaBrowser.pyx'),
           os.path.join(frontend_implementation_dir,'MozillaBrowserXPCOM.cc'),
