--- gogglesmm-0.12.2/Makefile	2011-06-08 21:00:13.000000000 +0100
+++ gogglesmm-0.12.2-patched/Makefile	2011-08-04 11:34:14.563894694 +0100
@@ -126,6 +126,8 @@
 
 endif
 
+$(OBJECTS): src/icons.h src/icons.cpp
+
 src/icons.h: $(ICONS)
 	@echo "    Creating Icon Resource Header"
 	@$(RESWRAP_H) -o $@ $(ICONS)
@@ -139,7 +141,6 @@
 src/GMMediaPlayerService.cpp: src/mpris_xml.h
 endif
 
-src/GMAbout.cpp: src/icons.h src/icons.cpp
 
 TRANSLATIONS:=$(basename $(notdir $(wildcard po/*.mo)))
 LINGUAS?=$(TRANSLATIONS)
