diff -dupr gestikk-old/gestikk/tools.py gestikk-0.6/gestikk/tools.py
--- gestikk-old/gestikk/tools.py	2008-11-22 22:17:34.905524960 +0100
+++ gestikk-0.6/gestikk/tools.py	2008-11-22 22:18:01.084529360 +0100
@@ -114,7 +114,7 @@ def get_path(kind):
         :param kind: one of 'pixmap', 'icon'
     """
     return {'pixmap':'/usr/share/pixmaps',
-            'icon':'/usr/share/icons',
+            'icon':'/usr/share/pixmaps',
             'gladefile':'/usr/share/gestikk',
             'executable':'/usr/bin'}.get(kind, '')
 
