--- python-libtorrent.cpp	2006-11-28 15:21:59.000000000 +0100
+++ python-libtorrent/python-libtorrent.cpp	2007-01-11 19:28:22.000000000 +0100
@@ -139,19 +139,19 @@
 
 void print_uniqueIDs()
 {
-//#ifdef AMD64
-//	for (unsigned long i = 0; i < uniqueIDs->size(); i++)
-//		printf("--uniqueIDs[%ld] = %ld\r\n", i, (*uniqueIDs)[i]);
-//#endif
+#ifdef AMD64
+	for (unsigned long i = 0; i < uniqueIDs->size(); i++)
+		printf("--uniqueIDs[%ld] = %ld\r\n", i, (*uniqueIDs)[i]);
+#endif
 }
 
 long get_index_from_unique(long uniqueID)
 {
 	assert(handles->size() == uniqueIDs->size());
 
-//#ifdef AMD64
-//	printf("Request for uniqueID: %ld\r\n", uniqueID);
-//#endif
+#ifdef AMD64
+	printf("Request for uniqueID: %ld\r\n", uniqueID);
+#endif
 	print_uniqueIDs();
 
 	for (unsigned long i = 0; i < uniqueIDs->size(); i++)
