diff -ur k.orig/admin/acinclude.m4.in k/admin/acinclude.m4.in
--- k.orig/admin/acinclude.m4.in	2005-11-26 22:46:32.000000000 +0100
+++ k/admin/acinclude.m4.in	2005-11-26 22:47:46.000000000 +0100
@@ -513,13 +513,6 @@
 ])
 ])
 
-AC_DEFUN(KDE_MISSING_ARTS_ERROR,
-[
-    AC_MSG_ERROR([The important program $1 was not found!
-Please check whether you installed aRts correctly.
-])
-])
-
 AC_DEFUN(KDE_SUBST_PROGRAMS,
 [
 
@@ -536,8 +529,6 @@
         kde_default_bindirs="$exec_prefix/bin $prefix/bin $kde_default_bindirs"
         KDE_FIND_PATH(dcopidl, DCOPIDL, [$kde_default_bindirs], [KDE_MISSING_PROG_ERROR(dcopidl)])
         KDE_FIND_PATH(dcopidl2cpp, DCOPIDL2CPP, [$kde_default_bindirs], [KDE_MISSING_PROG_ERROR(dcopidl2cpp)])
-        KDE_FIND_PATH(mcopidl, MCOPIDL, [$kde_default_bindirs], [KDE_MISSING_ARTS_ERROR(mcopidl)])
-        KDE_FIND_PATH(artsc-config, ARTSCCONFIG, [$kde_default_bindirs], [KDE_MISSING_ARTS_ERROR(artsc-config)])
         KDE_FIND_PATH(kde-config, KDECONFIG, [$kde_default_bindirs])
         KDE_FIND_PATH(meinproc, MEINPROC, [$kde_default_bindirs])
       
@@ -557,7 +548,6 @@
         AC_SUBST(DCOPIDL2CPP)
         AC_SUBST(DCOP_DEPENDENCIES)
         AC_SUBST(MCOPIDL)
-        AC_SUBST(ARTSCCONFIG)
         AC_SUBST(KDECONFIG)
 	AC_SUBST(MEINPROC)
  	AC_SUBST(KDE_XSL_STYLESHEET)
