--- Makefile
+++ Makefile
@@ -1,27 +1,46 @@
-ALL : markdown
-
-PROGRAM=markdown
+PROGRAM=peg-markdown
+LNAME=peg-markdown
+VER=0.4.12
+LIBRARY=lib$(LNAME).so
+REALNAME=$(LIBRARY).$(VER)
+CC ?= gcc
 CFLAGS ?= -Wall -O3 -ansi
 OBJS=markdown_parser.o markdown_output.o markdown_lib.o
 PEGDIR=peg-0.1.4
-LEG=$(PEGDIR)/leg
+LEG=leg
+
+PREFIX	?= /usr/local
+BINDIR	?= $(PREFIX)/bin
+LIBDIR ?= $(PREFIX)/lib
+INCDIR ?= $(PREFIX)/inc
+
+INSTALL ?= install
+INSTALL_PROG ?= $(INSTALL) -m755
+INSTALL_H ?= $(INSTALL) -m644
 
-$(LEG):
-	CC=gcc make -C $(PEGDIR)
+ALL : $(PROGRAM) build-doc
 
 %.o : %.c markdown_peg.h
-	$(CC) -c `pkg-config --cflags glib-2.0` $(CFLAGS) -o $@ $<
+	$(CC) -c `pkg-config --cflags glib-2.0` $(CFLAGS) -fPIC -o $@ $<
+
+$(PROGRAM) : markdown.c $(LIBRARY)
+	$(CC) `pkg-config --cflags glib-2.0` `pkg-config --libs glib-2.0` $(CFLAGS) $(LDFLAGS) -o $@ $< \
+	-L. -l $(LNAME)
+
+$(LIBRARY) : $(REALNAME)
+	ln -s $< $@
 
-$(PROGRAM) : markdown.c $(OBJS)
-	$(CC) `pkg-config --cflags glib-2.0` `pkg-config --libs glib-2.0` $(CFLAGS) -o $@ $(OBJS) $<
+$(REALNAME) : $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o $@ $? \
+	`pkg-config --cflags glib-2.0` `pkg-config --libs glib-2.0`
 
-markdown_parser.c : markdown_parser.leg $(LEG) markdown_peg.h parsing_functions.c utility_functions.c
+markdown_parser.c : markdown_parser.leg markdown_peg.h parsing_functions.c utility_functions.c
 	$(LEG) -o $@ $<
 
 .PHONY: clean test
 
 clean:
-	rm -f markdown_parser.c $(PROGRAM) $(OBJS); \
+	rm -f markdown_parser.c $(PROGRAM) $(OBJS) $(LIBRARY) $(REALNAME); \
 	make -C $(PEGDIR) clean
 
 distclean: clean
@@ -29,8 +48,13 @@
 
 test: $(PROGRAM)
 	cd MarkdownTest_1.0.3; \
+	export LD_LIBRARY_PATH=..; \
 	./MarkdownTest.pl --script=../$(PROGRAM) --tidy
 
 leak-check: $(PROGRAM)
-	valgrind --leak-check=full ./markdown README
+	export LD_LIBRARY_PATH=.; \
+	valgrind --leak-check=full ./$(PROGRAM) README
 
+build-doc: $(PROGRAM)
+	export LD_LIBRARY_PATH=.; \
+	./$(PROGRAM) README > README.html
