--- src/ant/build.gant	2010-01-27 17:39:20.612426014 +0100
+++ src/ant/build.gant.new	2010-01-27 17:37:33.816426014 +0100
@@ -210,26 +210,6 @@
 
   mkdir(dir: '${eclim.plugins}')
 
-  // if the eclim version has changed, stop eclimd before deploying
-  pluginsDir = new File(binding.getVariable('eclim.plugins'))
-  existing = null
-  pluginsDir.eachDirMatch(~/org\.eclim_.*/){
-    file -> existing = file
-  }
-
-  if (existing != null && existing.getName() != "org.eclim_${version}"){
-    echo('###\n# New eclim version, shutting down old version...\n###')
-    execute("${existing.path}/bin/eclim", '-command shutdown', failonerror=false)
-    execute("${existing.path}/bin/eclim",
-        '-f ${basedir}/src/test/resources/eclimrc.test -command shutdown',
-        failonerror=false)
-  }
-
-  delete(includeemptydirs: true){
-    fileset(dir: '${eclim.plugins}', includes: 'org.eclim*/**/*')
-    dirset(dir: '${eclim.plugins}', includes: 'org.eclim*')
-  }
-
   copy(todir: '${eclim.plugins}'){
     fileset(dir: '${build.plugins}', includes: '**/*')
   }
@@ -242,14 +222,6 @@
   }else{
     chmod(dir: '${eclim.plugins}/org.eclim_${eclim.version}/bin',
         includes: '**/*', excludes: '*.sed', perm: 'ugo+x')
-    // java symlink support sucks, especially if the link is broken
-    exec(executable: 'rm'){
-      arg(line: '${eclipse.home}/eclim ${eclipse.home}/eclimd')
-    }
-    symlink(link: '${eclipse.home}/eclim',
-        resource: '${eclim.plugins}/org.eclim_${eclim.version}/bin/eclim')
-    symlink(link: '${eclipse.home}/eclimd',
-        resource: '${eclim.plugins}/org.eclim_${eclim.version}/bin/eclimd')
   }
 
   replace(file: '${eclim.plugins}/org.eclim_${eclim.version}/plugin.properties',
