diff -Naur dpsrc-orig/gl_src.c dpsrc/gl_src.c
--- dpsrc-orig/gl_src.c	2001-02-28 16:45:56.000000000 -0500
+++ dpsrc/gl_src.c	2008-04-10 17:03:03.000000000 -0400
@@ -154,7 +154,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include <strings.h>
+#include <string.h>
 
 #if UNIX == 0
 #include <X11/Xlib.h>
@@ -257,7 +257,7 @@
 /* flags for current attribute settings */
 static int    OPEN_FLAG = 0;      /* 0 - X11 closed, 1 - X11 open */
 static int    OPEN_FLAG_2 = 0;    /* 0 - repeat graph window closed, 1 - open */
-int           GRAPH_FLAG = 0;     /* 0 - gleras has not been called, 1 - gleras has been called  */
+static int           GRAPH_FLAG = 0;     /* 0 - gleras has not been called, 1 - gleras has been called  */
 int           WIDTH_CURRENT;      /* current line width */
 int           LINE_STYLE_CURRENT; /* current line style */
 char          FONT_NAME_CURRENT[80];  /* name of current font */
@@ -922,7 +922,7 @@
    glNewList(list_id, GL_COMPILE_AND_EXECUTE);
 
    glClearColor(red_temp,green_temp,blue_temp,1.0);
-   glclear(GL_COLOR_BUFFER_BIT);
+   glClear(GL_COLOR_BUFFER_BIT);
 
    GRAPH_FLAG = 1;
 }
@@ -1486,7 +1486,7 @@
  *           Color only set if being changed.
  */
 #if APPEND_UNDERSCORE == 1 && SUBROUTINE_CASE == 1
-void glseco(jcol, red, green, blue)
+void glseco_(jcol, red, green, blue)
 #elif APPEND_UNDERSCORE == 1 && SUBROUTINE_CASE == 0
 void GLSECO_(jcol, red, green, blue)
 #elif APPEND_UNDERSCORE == 0 && SUBROUTINE_CASE == 1
@@ -1691,7 +1691,7 @@
    width = isize[0];
 #endif
    if (width < 1.0) width = 1.0;
-   glPointsize(width);
+   glPointSize(width);
    glBegin(GL_POINTS);
 #if PRECISION == 0
    glVertex2f(*ix, *iy);
diff -Naur dpsrc-orig/x11_src.c dpsrc/x11_src.c
--- dpsrc-orig/x11_src.c	2007-08-23 13:57:06.000000000 -0400
+++ dpsrc/x11_src.c	2008-04-10 17:02:18.000000000 -0400
@@ -214,7 +214,7 @@
 /* add following 2 lines MAY 1995   */
 #include <stdlib.h>
 #include <math.h>
-#include <strings.h>
+#include <string.h>
 
 /* global definitions */
 
@@ -507,7 +507,7 @@
 /* flags for current attribute settings */
 static int    OPEN_FLAG = 0;          /* 0 - X11 closed, 1 - X11 open */
 static int    OPEN_FLAG_2 = 0;        /* 0 - repeat graph window closed, 1 - open */
-int           GRAPH_FLAG = 0;         /* 0 - xerase has not been called, 1 - xerase has been called  */
+static int           GRAPH_FLAG = 0;         /* 0 - xerase has not been called, 1 - xerase has been called  */
 int           WIDTH_CURRENT;          /* current line width */
 int           LINE_STYLE_CURRENT;     /* current line style */
 int           CAP_STYLE_CURRENT;      /* current cap style */
--- dpsrc-orig/gs_src.c 2002-02-20 13:28:36.000000000 -0500
+++ dpsrc/gs_src.c  2008-04-28 08:54:28.000000000 -0400
@@ -111,9 +111,9 @@
 int    max_colors;                     /* maximum colors actually allocated */
 int    CURRENT_COLOR;                  /* Define current color */
 int    CURRENT_LINE_STYLE[12];         /* Define current line style */
-int    NPTS_STYLE = 0;                 /* Number of points in style */
+static int    NPTS_STYLE = 0;                 /* Number of points in style */
 int    color_table[MAX_COLORS + 100];  /* color table */
-int red[MAX_COLORS] = {
+static int red[MAX_COLORS] = {
     /*  0 -  7 */ 255,   0, 255,   0,   0, 255, 255,   0,
     /*  8 - 15 */ 255, 154,   0, 173, 138, 208,  47, 211,
     /* 16 - 23 */ 127, 165,  95, 255, 100,  85, 153,  72,
@@ -127,7 +127,7 @@
     /* 80 - 87 */ 238, 205, 139, 238, 205, 139, 238, 205,
     /* 88 - 88 */ 139
 };
-int green[MAX_COLORS] = {
+static int green[MAX_COLORS] = {
     /*  0 -  7 */ 255,   0,   0,   0, 255,   0, 165, 255,
     /*  8 - 15 */ 255, 205, 100, 216,  43,  32,  79, 211,
     /* 16 - 23 */ 255,  42, 158, 127, 149, 107,  50,  61,
@@ -141,7 +141,7 @@
     /* 80 - 87 */ 154, 133,  90,   0,   0,   0,   0,   0,
     /* 88 - 88 */   0
 };
-int blue[MAX_COLORS] = {
+static int blue[MAX_COLORS] = {
     /*  0 -  7 */ 255,   0,   0, 255,   0, 255,   0, 255,
     /*  8 - 15 */   0,  50,   0, 230, 226, 144,  79, 211,
     /* 16 - 23 */ 212,  42, 160,  80, 237,  47, 204, 139,
@@ -158,7 +158,7 @@

 /* flags for current attribute settings */
 static int    OPEN_FLAG = 0;          /* 0 - GS closed, 1 - GS open */
-int           DEVICE_TYPE = 0;        /* define device */
+static int           DEVICE_TYPE = 0;        /* define device */
                                       /* 1 - jpeg */
                                       /* 2 - png */
                                       /* 3 - windows bmp */

