--- libiqxmlrpc-0.7.2/doc/Makefile.in	2006-06-18 17:32:16.000000000 +0200
+++ libiqxmlrpc-0.7.2.new/doc/Makefile.in	2006-06-25 21:52:20.000000000 +0200
@@ -208,9 +208,6 @@
 	[ ${MKDOC} == Xyes ] && doxygen libiqxmlrpc.doxygen || true
 
 install-data-local: libiqxmlrpc.html index.html
-	[ ${MKDOC} == Xyes ] && mkdir -p ${DOCDIR}/libiqxmlrpc.html || true
-	[ ${MKDOC} == Xyes ] && install -m 644 index.html ${DOCDIR} || true
-	[ ${MKDOC} == Xyes ] && install -m 644 libiqxmlrpc.html/* ${DOCDIR}/libiqxmlrpc.html || true
 
 uninstall-local:
 	rm -f ${DOCDIR}/index.html
