--- m4/ax_boost_unit_test_framework.m4.orig	2008-02-17 17:25:21.000000000 +0100
+++ m4/ax_boost_unit_test_framework.m4	2008-02-17 17:26:42.000000000 +0100
@@ -62,11 +62,11 @@
 			AC_DEFINE(HAVE_BOOST_UNIT_TEST_FRAMEWORK,,[define if the Boost::Unit_Test_Framework library is available])
 			BN=boost_unit_test_framework
             if test "x$ax_boost_user_unit_test_framework_lib" = "x"; then
-         		saved_ldflags="${LDFLAGS}"
+         		saved_libs="${LIBS}"
 		    	for ax_lib in $BN $BN-$CC $BN-$CC-mt $BN-$CC-mt-s $BN-$CC-s \
                              lib$BN lib$BN-$CC lib$BN-$CC-mt lib$BN-$CC-mt-s lib$BN-$CC-s \
                              $BN-mgw $BN-mgw $BN-mgw-mt $BN-mgw-mt-s $BN-mgw-s ; do
-                   LDFLAGS="${LDFLAGS} -l$ax_lib"
+                   LIBS="${LIBS} -l$ax_lib"
     			   AC_CACHE_CHECK(Boost::UnitTestFramework library linkage,
 	      			    		   ax_cv_boost_unit_test_framework_link,
 						  [AC_LANG_PUSH([C++])
@@ -81,7 +81,7 @@
                                  link_unit_test_framework="yes",link_unit_test_framework="no")
 			      AC_LANG_POP([C++])
                   ])
-                  LDFLAGS="${saved_ldflags}"
+                  LIBS="${saved_libs}"
 
 			      if test "x$link_unit_test_framework" = "xyes"; then
                       BOOST_UNIT_TEST_FRAMEWORK_LIB="-l$ax_lib"
@@ -90,9 +90,9 @@
 				  fi
                 done
             else
-         		saved_ldflags="${LDFLAGS}"
+         		saved_libs="${LIBS}"
                for ax_lib in $ax_boost_user_unit_test_framework_lib $BN-$ax_boost_user_unit_test_framework_lib; do
-                   LDFLAGS="${LDFLAGS} -l$ax_lib"
+                   LIBS="${LIBS} -l$ax_lib"
               			   AC_CACHE_CHECK(Boost::UnitTestFramework library linkage,
 	      			    		   ax_cv_boost_unit_test_framework_link,
 						  [AC_LANG_PUSH([C++])
@@ -107,7 +107,7 @@
                                  link_unit_test_framework="yes",link_unit_test_framework="no")
 			      AC_LANG_POP([C++])
                   ])
-                  LDFLAGS="${saved_ldflags}"
+                  LIBS="${saved_libs}"
 			      if test "x$link_unit_test_framework" = "xyes"; then
                       BOOST_UNIT_TEST_FRAMEWORK_LIB="-l$ax_lib"
                       AC_SUBST(BOOST_UNIT_TEST_FRAMEWORK_LIB)
