diff -Nau Makefile.orig Makefile
--- Makefile.orig	2006-11-11 14:26:28.000000000 +0100
+++ Makefile	2006-12-08 02:47:11.000000000 +0100
@@ -4,10 +4,10 @@
 #  This project has been placed under the GPL Licence.
 #
 
-CXXFLAGS	:= -O2 `cups-config --cflags` 
-LDFLAGS		:= `cups-config --ldflags`
-CUPSFILTER	:= `cups-config --serverbin`/filter
-CUPSPPD		:= `cups-config --datadir`/model
+CXXFLAGS	+= 
+LDFLAGS		+= 
+CUPSFILTER	:= $(DESTDIR)`cups-config --serverbin`/filter
+CUPSPPD		:= $(DESTDIR)`cups-config --datadir`/model
 
 # === DON'T CHANGE ANYTHING AFTER THIS MESSAGE ====
 
--- src/Makefile.orig	2006-11-11 14:26:28.000000000 +0100
+++ src/Makefile	2006-12-08 02:56:14.000000000 +0100
@@ -20,7 +20,7 @@
 	$(CXX) $(CXXFLAGS) -c $<
 
 install: rastertospl2
-	install -m 755 -s rastertospl2 ${CUPSFILTER}
+	install -m 755 rastertospl2 ${CUPSFILTER}
 
 .PHONY: clean distclean
 clean:
