diff -Naur screenkast-0.1.1/src/screenkast/srkcategorycombobox.h screenkast-0.1.1.new/src/screenkast/srkcategorycombobox.h
--- screenkast-0.1.1/src/screenkast/srkcategorycombobox.h	2006-02-25 14:49:24.000000000 +0100
+++ screenkast-0.1.1.new/src/screenkast/srkcategorycombobox.h	2006-07-10 20:21:33.000000000 +0200
@@ -32,7 +32,7 @@
 
 #include <kcombobox.h>
 #include <vector>
-#include <isdwscategory.h>
+#include <libinstrudeo/isdwscategory.h>
 
 class SRKCategoryComboBox : public KComboBox
 {
diff -Naur screenkast-0.1.1/src/screenkast/srkcommentbox.cpp screenkast-0.1.1.new/src/screenkast/srkcommentbox.cpp
--- screenkast-0.1.1/src/screenkast/srkcommentbox.cpp	2006-03-30 21:33:00.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkcommentbox.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -28,7 +28,7 @@
 
 #include <kcursor.h>
 
-#include <isdcommentbox.h>
+#include <libinstrudeo/isdcommentbox.h>
 
 #include "srkcommentbox.h"
 
diff -Naur screenkast-0.1.1/src/screenkast/srkcommentpropertiesdialog.cpp screenkast-0.1.1.new/src/screenkast/srkcommentpropertiesdialog.cpp
--- screenkast-0.1.1/src/screenkast/srkcommentpropertiesdialog.cpp	2006-05-03 15:34:54.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkcommentpropertiesdialog.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -28,7 +28,7 @@
 
 #include <kcolorbutton.h>
 
-#include <isdcommentbox.h>
+#include <libinstrudeo/isdcommentbox.h>
 
 #include "srkcommentbox.h"
 #include "srkcommentpropertiesdialog.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkmainview.cpp screenkast-0.1.1.new/src/screenkast/srkmainview.cpp
--- screenkast-0.1.1/src/screenkast/srkmainview.cpp	2006-07-08 19:43:32.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkmainview.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -45,10 +45,10 @@
 #include <kprogress.h>
 #include <kprocio.h>
 
-#include <isdvideoproperties.h>
-#include <isdxmlfile.h>
-#include <isdrecording.h>
-#include <isdvnclogimporter.h>
+#include <libinstrudeo/isdvideoproperties.h>
+#include <libinstrudeo/isdxmlfile.h>
+#include <libinstrudeo/isdrecording.h>
+#include <libinstrudeo/isdvnclogimporter.h>
 
 #include "srktransmitrecordingdialog.h"
 #include "srkmetainfodialog.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkmainview.h screenkast-0.1.1.new/src/screenkast/srkmainview.h
--- screenkast-0.1.1/src/screenkast/srkmainview.h	2006-07-08 18:44:20.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkmainview.h	2006-07-10 20:21:33.000000000 +0200
@@ -32,9 +32,9 @@
 #include <string>
 #include <qwidget.h>
 #include <qfile.h>
-#include <isdcommentbox.h>
-#include <isdexporter.h>
-#include <isdprogresscallback.h>
+#include <libinstrudeo/isdcommentbox.h>
+#include <libinstrudeo/isdexporter.h>
+#include <libinstrudeo/isdprogresscallback.h>
 #include "srkerror.h"
 
 using namespace std;
diff -Naur screenkast-0.1.1/src/screenkast/srkmainwindow.cpp screenkast-0.1.1.new/src/screenkast/srkmainwindow.cpp
--- screenkast-0.1.1/src/screenkast/srkmainwindow.cpp	2006-06-29 21:37:09.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkmainwindow.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -45,7 +45,7 @@
 #include <klanguagebutton.h>
 #include <kseparator.h>
 
-#include <isdcommentbox.h>
+#include <libinstrudeo/isdcommentbox.h>
 
 #include "srkaddlanguagedialog.h"
 #include "srkvideocanvas.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkmetainfodialog.cpp screenkast-0.1.1.new/src/screenkast/srkmetainfodialog.cpp
--- screenkast-0.1.1/src/screenkast/srkmetainfodialog.cpp	2006-05-20 14:34:06.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkmetainfodialog.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -32,9 +32,9 @@
 #include <kglobal.h>
 #include <kcursor.h>
 
-#include <isdwscategory.h>
-#include <isdwscommunicator.h>
-#include <isdrecordingmetainfo.h>
+#include <libinstrudeo/isdwscategory.h>
+#include <libinstrudeo/isdwscommunicator.h>
+#include <libinstrudeo/isdrecordingmetainfo.h>
 
 #include "srkplatformcombobox.h"
 #include "srkcategorycombobox.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkplatformcombobox.h screenkast-0.1.1.new/src/screenkast/srkplatformcombobox.h
--- screenkast-0.1.1/src/screenkast/srkplatformcombobox.h	2006-03-06 16:04:20.000000000 +0100
+++ screenkast-0.1.1.new/src/screenkast/srkplatformcombobox.h	2006-07-10 20:21:33.000000000 +0200
@@ -32,7 +32,7 @@
 
 #include <kcombobox.h>
 #include <vector>
-#include <isdwsplatform.h>
+#include <libinstrudeo/isdwsplatform.h>
 
 class SRKPlatformComboBox : public KComboBox
 {
diff -Naur screenkast-0.1.1/src/screenkast/srktransmitrecordingdialog.cpp screenkast-0.1.1.new/src/screenkast/srktransmitrecordingdialog.cpp
--- screenkast-0.1.1/src/screenkast/srktransmitrecordingdialog.cpp	2006-06-27 15:50:02.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srktransmitrecordingdialog.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -34,10 +34,10 @@
 #include <kglobal.h>
 #include <kcursor.h>
 
-#include <isdutils.h>
-#include <isdwscommunicator.h>
-#include <isdobject.h>
-#include <isdrecording.h>
+#include <libinstrudeo/isdutils.h>
+#include <libinstrudeo/isdwscommunicator.h>
+#include <libinstrudeo/isdobject.h>
+#include <libinstrudeo/isdrecording.h>
 
 #include "srkmainview.h"
 #include "srktransmitrecordingdialog.h"
diff -Naur screenkast-0.1.1/src/screenkast/srktransmitrecordingdialog.h screenkast-0.1.1.new/src/screenkast/srktransmitrecordingdialog.h
--- screenkast-0.1.1/src/screenkast/srktransmitrecordingdialog.h	2006-03-01 17:03:16.000000000 +0100
+++ screenkast-0.1.1.new/src/screenkast/srktransmitrecordingdialog.h	2006-07-10 20:21:33.000000000 +0200
@@ -22,7 +22,7 @@
 #ifndef SRKTRANSMITRECORDINGDIALOG_H
 #define SRKTRANSMITRECORDINGDIALOG_H
 
-#include <isdprogresscallback.h>
+#include <libinstrudeo/isdprogresscallback.h>
 
 #include "srktransmitrecordingdialogbase.h"
 
diff -Naur screenkast-0.1.1/src/screenkast/srkvideocanvas.cpp screenkast-0.1.1.new/src/screenkast/srkvideocanvas.cpp
--- screenkast-0.1.1/src/screenkast/srkvideocanvas.cpp	2006-04-22 13:35:08.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkvideocanvas.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -24,9 +24,9 @@
 
 #include <qobjectlist.h>
 
-#include <isdvideoproperties.h>
-#include <isdrecording.h>
-#include <isdvideocanvas.h>
+#include <libinstrudeo/isdvideoproperties.h>
+#include <libinstrudeo/isdrecording.h>
+#include <libinstrudeo/isdvideocanvas.h>
 
 #include "srkcommentbox.h"
 #include "srkvideocanvas.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkvideocontainer.cpp screenkast-0.1.1.new/src/screenkast/srkvideocontainer.cpp
--- screenkast-0.1.1/src/screenkast/srkvideocontainer.cpp	2006-06-26 16:08:01.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkvideocontainer.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -35,10 +35,10 @@
 #include <kinputdialog.h>
 #include <kmessagebox.h>
 
-#include <isdvideocanvas.h>
-#include <isdxmlfile.h>
-#include <isdrecording.h>
-#include <isdcommentbox.h>
+#include <libinstrudeo/isdvideocanvas.h>
+#include <libinstrudeo/isdxmlfile.h>
+#include <libinstrudeo/isdrecording.h>
+#include <libinstrudeo/isdcommentbox.h>
 #include <glibmm/ustring.h>
 
 #include "srkcommentpropertiesdialog.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkvideocontainer.h screenkast-0.1.1.new/src/screenkast/srkvideocontainer.h
--- screenkast-0.1.1/src/screenkast/srkvideocontainer.h	2006-03-24 05:11:50.000000000 +0100
+++ screenkast-0.1.1.new/src/screenkast/srkvideocontainer.h	2006-07-10 20:21:33.000000000 +0200
@@ -30,7 +30,7 @@
 
 #include <string>
 #include <qscrollview.h>
-#include <isdcommentbox.h>
+#include <libinstrudeo/isdcommentbox.h>
 #include "srkerror.h"
 
 using namespace std;
diff -Naur screenkast-0.1.1/src/screenkast/srkvideopanel.cpp screenkast-0.1.1.new/src/screenkast/srkvideopanel.cpp
--- screenkast-0.1.1/src/screenkast/srkvideopanel.cpp	2006-06-26 16:06:47.000000000 +0200
+++ screenkast-0.1.1.new/src/screenkast/srkvideopanel.cpp	2006-07-10 20:21:33.000000000 +0200
@@ -32,8 +32,8 @@
 #include <kmessagebox.h>
 #include <klocale.h>
 
-#include <isdrecording.h>
-#include <isdvideoproperties.h>
+#include <libinstrudeo/isdrecording.h>
+#include <libinstrudeo/isdvideoproperties.h>
 
 #include "srkmainview.h"
 #include "srkvideocontainer.h"
diff -Naur screenkast-0.1.1/src/screenkast/srkvideopanel.h screenkast-0.1.1.new/src/screenkast/srkvideopanel.h
--- screenkast-0.1.1/src/screenkast/srkvideopanel.h	2006-03-23 16:12:18.000000000 +0100
+++ screenkast-0.1.1.new/src/screenkast/srkvideopanel.h	2006-07-10 20:21:33.000000000 +0200
@@ -23,7 +23,7 @@
 #define SRKVIDEOPANEL_H
 
 #include "srkvideopanelbase.h"
-#include <isdcommentbox.h>
+#include <libinstrudeo/isdcommentbox.h>
 #include "srkerror.h"
 
 //number of millisecs to poll for changes
