--- contrib/fec/fec_src/com/onionnetworks/fec/Native8Code.java.old	2008-02-21 19:46:43.000000000 +0100
+++ contrib/fec/fec_src/com/onionnetworks/fec/Native8Code.java	2008-02-21 19:27:10.000000000 +0100
@@ -3,6 +3,7 @@
 //import java.security.AccessController;
 //import sun.security.action.*;
 import com.onionnetworks.util.*;
+import java.io.File;
 
 /**
  * This class is the frontend for the JNI wrapper for the C implementation of
@@ -21,14 +22,9 @@
     private int code;
     
     static {
-        String path = NativeDeployer.getLibraryPath
-            (Native8Code.class.getClassLoader(),"fec8");
-        if (path != null) {
-            System.load(path);
-        } else {
-            System.out.println("Unable to find native library for fec8 for platform "+NativeDeployer.OS_ARCH);
-	    System.out.println(path);
-        }
+			File path = new File("lib/libfec8.so");
+			System.out.println("Loading FEC lib ["+path.getAbsolutePath()+']');
+			System.load(path.getAbsolutePath());
     }
     
     public Native8Code(int k, int n) {
