--- wrapper.conf	2008-03-30 16:12:53.000000000 +0200
+++ wrapper.conf.new	2008-03-30 16:16:12.000000000 +0200
@@ -1,8 +1,10 @@
 wrapper.java.command=java
-wrapper.working.dir=../
+wrapper.working.dir=/opt/freenet/
 wrapper.java.mainclass=freenet.node.NodeStarter
-wrapper.java.classpath.1=freenet.jar
-wrapper.java.classpath.2=freenet-ext.jar
+wrapper.java.classpath.1=/usr/share/freenet/lib/freenet.jar
+wrapper.java.classpath.2=/usr/share/java-service-wrapper/lib/wrapper.jar
+wrapper.java.classpath.3=/usr/share/db-je-3.2/lib/je.jar
+wrapper.java.classpath.4=/usr/share/fec/lib/fec.jar
 wrapper.java.library.path.1=lib
 wrapper.java.initmemory=60
 wrapper.java.maxmemory=128
