--- Makefile.am.orig	2007-11-06 10:41:51.000000000 -0800
+++ Makefile.am	2007-11-06 10:42:01.000000000 -0800
@@ -1,4 +1,4 @@
-SUBDIRS = goocanv8 src doc pixmaps data po
+SUBDIRS = src doc pixmaps data po
 
 install-data-local:
 	@$(NORMAL_INSTALL)
--- configure.ac.orig	2007-11-06 10:42:13.000000000 -0800
+++ configure.ac	2007-11-06 10:44:00.000000000 -0800
@@ -21,7 +21,7 @@
 fi
 
 dnl chack for glib and gtk libraries
-pkg_modules="gtk+-2.0 >= 2.12.0 glib-2.0 >= 2.14.0 gthread-2.0 >= 2.14.0"
+pkg_modules="gtk+-2.0 >= 2.12.0 glib-2.0 >= 2.14.0 gthread-2.0 >= 2.14.0 goocanvas >= 0.8"
 PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)
@@ -76,6 +76,7 @@
 GTHR_V=`pkg-config --modversion gthread-2.0`
 GDK_V=`pkg-config --modversion gdk-2.0`
 GTK_V=`pkg-config --modversion gtk+-2.0`
+GOO_V=`pkg-config --modversion goocanvas`
 dnl SQL_V=`pkg-config --modversion sqlite3`
 if test "$havecurl" = true ; then
    CURL_V=`pkg-config --modversion libcurl`
@@ -88,8 +88,6 @@
 AC_SUBST(CFLAGS)
 AC_SUBST(LDFLAGS)
 
-AC_CONFIG_SUBDIRS(goocanv8)
-
 AC_OUTPUT([
 Makefile
 doc/Makefile
@@ -112,6 +112,7 @@
 echo Gthread version.... : $GTHR_V
 echo Gdk version........ : $GDK_V
 echo Gtk+ version....... : $GTK_V
+echo Goocanvas version.. : $GOO_V
 dnl echo SQLite version..... : $SQL_V
 if test "$havecurl" = true ; then
 echo Curl version....... : $CURL_V
--- src/Makefile.am.orig	2007-11-06 10:44:26.000000000 -0800
+++ src/Makefile.am	2007-11-06 10:44:46.000000000 -0800
@@ -3,7 +3,7 @@
 SUBDIRS = sgpsdp
 
 INCLUDES = \
-	@PACKAGE_CFLAGS@ -I.. -I$(top_srcdir)/goocanv8/src \
+	@PACKAGE_CFLAGS@ -I.. \
 	-DDATADIR=\""$(datadir)"\" \
 	-DPACKAGE_DATA_DIR=\""$(datadir)/gpredict"\" \
 	-DPACKAGE_PIXMAPS_DIR=\""$(datadir)/pixmaps/gpredict"\" \
@@ -103,7 +103,7 @@
 
 
 ##gpredict_LDADD = ./sgpsdp/libsgp4sdp4.a @PACKAGE_LIBS@
-gpredict_LDADD = @PACKAGE_LIBS@ ../goocanv8/src/.libs/libgoocanvas.a
+gpredict_LDADD = @PACKAGE_LIBS@
 
 ## $(INTLLIBS)
 
