diff --git a/terminator b/terminator
index 5d449e6..3ffb1a0 100755
--- a/terminator
+++ b/terminator
@@ -174,14 +174,3 @@ See the following bug report for more details:
 
   gtk.main()
 
-  try:
-    import pynotify
-    pynotify.init(APP_NAME.capitalize())
-    icontheme = gtk.IconTheme()
-    icon = icontheme.lookup_icon(APP_NAME, 48, 0)
-    notification = pynotify.Notification('Terminator', "I'll be back", icon.get_filename())
-    notification.set_urgency(pynotify.URGENCY_LOW)
-    notification.show()
-  except:
-    pass
-
