diff -urN lloyd-yajl-9c15d72//src/CMakeLists.txt lloyd-yajl-9c15d72.new//src/CMakeLists.txt
--- lloyd-yajl-9c15d72//src/CMakeLists.txt	2010-01-08 18:05:11.000000000 +0100
+++ lloyd-yajl-9c15d72.new//src/CMakeLists.txt	2010-04-04 23:49:02.000000000 +0200
@@ -42,7 +42,7 @@
 ADD_DEFINITIONS(-DYAJL_BUILD)
 
 # set up some paths
-SET (libDir ${CMAKE_CURRENT_BINARY_DIR}/../${YAJL_DIST_NAME}/lib)
+SET (libDir ${CMAKE_CURRENT_BINARY_DIR}/../${YAJL_DIST_NAME}/lib${LIB_SUFFIX})
 SET (incDir ${CMAKE_CURRENT_BINARY_DIR}/../${YAJL_DIST_NAME}/include/yajl)
 
 # set the output path for libraries
@@ -77,8 +77,8 @@
 INCLUDE_DIRECTORIES(${incDir}/..)
 
 IF(NOT WIN32)
-  INSTALL(TARGETS yajl LIBRARY DESTINATION lib)
-  INSTALL(TARGETS yajl_s ARCHIVE DESTINATION lib)
+  INSTALL(TARGETS yajl LIBRARY DESTINATION lib${LIB_SUFFIX})
+  INSTALL(TARGETS yajl_s ARCHIVE DESTINATION lib${LIB_SUFFIX})
   INSTALL(DIRECTORY api/ DESTINATION include/yajl
           FILES_MATCHING PATTERN "*.h")
 ENDIF()
