--- orig-work/evanphx-rubinius-aee1b9b/config.rb	2010-09-28 22:30:51.000000000 -0700
+++ new-work/evanphx-rubinius-aee1b9b/config.rb	2010-09-28 23:13:18.000000000 -0700
@@ -5,7 +5,7 @@
     :llvm           => :no,
     :llvm_configure => "",
     :compiler       => "gcc",
-    :defines        => ["HAS_EXECINFO"],
+    :defines        => ["HAS_EXECINFO", "__STDC_LIMIT_MACROS", "__STDC_CONSTANT_MACROS"],
     :host           => "i686-pc-linux-gnu",
     :cpu            => "i686",
     :vendor         => "pc",
