diff -u -r work/Rakefile old-work/Rakefile
--- work/Rakefile	2010-05-14 10:21:35.000000000 -0700
+++ old-work/Rakefile	2010-06-10 11:39:56.000000000 -0700
@@ -38,7 +38,7 @@
     ENV["CFLAGS"]      = "-Ivm/capi"
   end
 
-  sh "bin/mspec ci --background"
+  sh "RUBYOPT='' bin/mspec ci --background"
 end
 
 # See vm.rake for more information
diff -u -r work/bin/rbx-build old-work/bin/rbx-build
--- work/bin/rbx-build	2010-05-14 10:21:35.000000000 -0700
+++ old-work/bin/rbx-build	2010-06-10 11:37:12.000000000 -0700
@@ -7,4 +7,6 @@
 export RBX_LIB="$BASE_DIR/lib"
 export RBX_CAPI_DIR="$BASE_DIR/vm/capi"
 
+unset RUBYOPT
+
 exec $BIN_DIR/../vm/vm "$*"
