--- dwm-0.5~/event.c	2006-07-22 15:38:12.000000000 +0200
+++ dwm-0.5-new/event.c	2006-07-23 12:25:29.000000000 +0200
@@ -64,6 +64,9 @@
 	*/
 };
 
+#define NumLockMask Mod2Mask
+unsigned int valid_mask =  255 &  ~( NumLockMask | LockMask);
+
 /* END CUSTOMIZE */
 
 /* static */
@@ -277,10 +280,11 @@
 	unsigned int i;
 	KeySym keysym;
 	XKeyEvent *ev = &e->xkey;
+	ev->state &= valid_mask;
 
 	keysym = XKeycodeToKeysym(dpy, (KeyCode)ev->keycode, 0);
 	for(i = 0; i < len; i++)
-		if((keysym == key[i].keysym) && (key[i].mod == ev->state)) {
+		if((keysym == key[i].keysym) && ((key[i].mod & valid_mask) == ev->state)) {
 			if(key[i].func)
 				key[i].func(&key[i].arg);
 			return;
@@ -383,7 +387,19 @@
 	for(i = 0; i < len; i++) {
 		code = XKeysymToKeycode(dpy, key[i].keysym);
 		XUngrabKey(dpy, code, key[i].mod, root);
+		if (NumLockMask)
+		{
+			XUngrabKey(dpy, code, key[i].mod | NumLockMask, root);
+			XUngrabKey(dpy, code, key[i].mod | NumLockMask | LockMask, root);
+		}
 		XGrabKey(dpy, code, key[i].mod, root, True,
 				GrabModeAsync, GrabModeAsync);
+		if (NumLockMask)
+		{
+			XGrabKey(dpy, code, key[i].mod | NumLockMask, root, True,
+					GrabModeAsync, GrabModeAsync);
+			XGrabKey(dpy, code, key[i].mod | NumLockMask | LockMask, root, True,
+					GrabModeAsync, GrabModeAsync);
+		}
 	}
 }
