diff -ru dwm-1.0~/client.c dwm-1.0/client.c
--- dwm-1.0~/client.c	2006-08-24 11:40:13.000000000 +0200
+++ dwm-1.0/client.c	2006-08-24 11:43:53.000000000 +0200
@@ -26,8 +26,6 @@
 static void
 resizetitle(Client *c)
 {
-	int i;
-
 	c->tw = textw(c->name);
 	if(c->tw > c->w)
 		c->tw = c->w + 2;
@@ -325,7 +323,7 @@
 settitle(Client *c)
 {
 	char **list = NULL;
-	int i, n;
+	int n;
 	XTextProperty name;
 
 	name.nitems = 0;
diff -ru dwm-1.0~/draw.c dwm-1.0/draw.c
--- dwm-1.0~/draw.c	2006-08-24 11:40:13.000000000 +0200
+++ dwm-1.0/draw.c	2006-08-24 11:43:12.000000000 +0200
@@ -151,7 +151,6 @@
 void
 drawtitle(Client *c)
 {
-	int i;
 	Bool istile = arrange == dotile;
 
 	if(c == sel && issel) {
