diff -ru dwm-0.5~/client.c dwm-0.5-fixfocus/client.c
--- dwm-0.5~/client.c	2006-07-22 15:38:12.000000000 +0200
+++ dwm-0.5-fixfocus/client.c	2006-07-23 20:22:23.000000000 +0200
@@ -244,11 +244,11 @@
 	c->next = clients;
 	clients = c;
 
-	XGrabButton(dpy, Button1, MODKEY, c->win, False, ButtonPressMask,
+	XGrabButton(dpy, Button1, MODKEY, c->win, False, ButtonMask,
 			GrabModeAsync, GrabModeSync, None, None);
-	XGrabButton(dpy, Button2, MODKEY, c->win, False, ButtonPressMask,
+	XGrabButton(dpy, Button2, MODKEY, c->win, False, ButtonMask,
 			GrabModeAsync, GrabModeSync, None, None);
-	XGrabButton(dpy, Button3, MODKEY, c->win, False, ButtonPressMask,
+	XGrabButton(dpy, Button3, MODKEY, c->win, False, ButtonMask,
 			GrabModeAsync, GrabModeSync, None, None);
 
 	if(!c->isfloat)
diff -ru dwm-0.5~/dwm.h dwm-0.5-fixfocus/dwm.h
--- dwm-0.5~/dwm.h	2006-07-22 15:38:12.000000000 +0200
+++ dwm-0.5-fixfocus/dwm.h	2006-07-23 20:22:48.000000000 +0200
@@ -24,6 +24,10 @@
 enum { Tfnord, Tdev, Tnet, Twork, Tmisc, TLast };
 
 /* END CUSTOMIZE */
+/* mask shorthands, used in event.c and client.c */
+#define ButtonMask     (ButtonPressMask | ButtonReleaseMask)
+#define MouseMask      (ButtonMask | PointerMotionMask)
+
 
 typedef union Arg Arg;
 typedef struct Client Client;
diff -ru dwm-0.5~/event.c dwm-0.5-fixfocus/event.c
--- dwm-0.5~/event.c	2006-07-22 15:38:12.000000000 +0200
+++ dwm-0.5-fixfocus/event.c	2006-07-23 20:23:47.000000000 +0200
@@ -8,9 +8,6 @@
 #include <X11/keysym.h>
 #include <X11/Xatom.h>
 
-#define ButtonMask      (ButtonPressMask | ButtonReleaseMask)
-#define MouseMask       (ButtonMask | PointerMotionMask)
-
 /* CUSTOMIZE */
 
 typedef struct {
@@ -247,7 +244,7 @@
 	Client *c;
 	XCrossingEvent *ev = &e->xcrossing;
 
-	if(ev->mode != NotifyNormal || ev->detail == NotifyInferior)
+	if(ev->detail == NotifyInferior)
 		return;
 
 	if((c = getclient(ev->window)))
diff -ru dwm-0.5~/tag.c dwm-0.5-fixfocus/tag.c
--- dwm-0.5~/tag.c	2006-07-22 15:38:12.000000000 +0200
+++ dwm-0.5-fixfocus/tag.c	2006-07-23 20:28:06.000000000 +0200
@@ -67,6 +67,8 @@
 			higher(sel);
 			focus(sel);
 		}
+		else
+			XSetInputFocus(dpy, root, RevertToPointerRoot, CurrentTime);
 	}
 	drawall();
 }
@@ -130,6 +132,8 @@
 			higher(sel);
 			focus(sel);
 		}
+		else
+			XSetInputFocus(dpy, root, RevertToPointerRoot, CurrentTime);
 	}
 	drawall();
 }
