diff -Naur uzbl-9999.orig/Makefile uzbl-9999/Makefile
--- uzbl-9999.orig/Makefile	2009-11-08 20:15:30.000000000 -0300
+++ uzbl-9999/Makefile	2009-11-08 20:16:04.000000000 -0300
@@ -1,7 +1,7 @@
 # first entries are for gnu make, 2nd for BSD make.  see http://lists.uzbl.org/pipermail/uzbl-dev-uzbl.org/2009-July/000177.html
 
-CFLAGS:=-std=c99 $(shell pkg-config --cflags gtk+-2.0 webkit-1.0 libsoup-2.4 gthread-2.0) -ggdb -Wall -W -DARCH="\"$(shell uname -m)\"" -lgthread-2.0 -DCOMMIT="\"$(shell git log | head -n1 | sed "s/.* //")\"" $(CPPFLAGS) -fPIC -W -Wall -Wextra -pedantic -ggdb3
-CFLAGS!=echo -std=c99 `pkg-config --cflags gtk+-2.0 webkit-1.0 libsoup-2.4 gthread-2.0` -ggdb -Wall -W -DARCH='"\""'`uname -m`'"\""' -lgthread-2.0 -DCOMMIT='"\""'`git log | head -n1 | sed "s/.* //"`'"\""' $(CPPFLAGS) -fPIC -W -Wall -Wextra -pedantic -ggdb3
+CFLAGS:=-std=c99 $(shell pkg-config --cflags gtk+-2.0 webkit-1.0 libsoup-2.4 gthread-2.0) -DARCH="\"gentoo\"" -lgthread-2.0 -DCOMMIT="\"$(shell git log | head -n1 | sed "s/.* //")\"" $(CPPFLAGS) $(CFLAGS)
+CFLAGS!=echo -std=c99 `pkg-config --cflags gtk+-2.0 webkit-1.0 libsoup-2.4 gthread-2.0` -DARCH='\"gentoo\"' -lgthread-2.0 -DCOMMIT='"\""'`git log | head -n1 | sed "s/.* //"`'"\""' $(CPPFLAGS) $(CFLAGS)
 
 LDFLAGS:=$(shell pkg-config --libs gtk+-2.0 webkit-1.0 libsoup-2.4 gthread-2.0) -pthread $(LDFLAGS)
 LDFLAGS!=echo `pkg-config --libs gtk+-2.0 webkit-1.0 libsoup-2.4 gthread-2.0` -pthread $(LDFLAGS)
@@ -32,15 +32,11 @@
 	@echo LINKING object files
 	@${CC} -o $@ ${OBJ} ${LDFLAGS}
 	@echo ... done.
-	@echo Stripping binary
-	@strip $@
-	@echo ... done.
-
 
 uzbl-browser: uzbl-core
 
 # packagers, set DESTDIR to your "package directory" and PREFIX to the prefix you want to have on the end-user system
-PREFIX?=/usr/local
+PREFIX?=/usr
 INSTALLDIR?=$(DESTDIR)$(PREFIX)
 
 # the 'tests' target can never be up to date
@@ -86,14 +82,9 @@
 
 install-uzbl-core: all
 	install -d $(INSTALLDIR)/bin
-	install -d $(INSTALLDIR)/share/uzbl/docs
 	install -d $(INSTALLDIR)/share/uzbl/examples
-	cp -rp docs     $(INSTALLDIR)/share/uzbl/
-	cp -rp config.h $(INSTALLDIR)/share/uzbl/docs/
 	cp -rp examples $(INSTALLDIR)/share/uzbl/
 	install -m755 uzbl-core    $(INSTALLDIR)/bin/uzbl-core
-	install -m644 AUTHORS      $(INSTALLDIR)/share/uzbl/docs
-	install -m644 README       $(INSTALLDIR)/share/uzbl/docs
 
 install-uzbl-browser: all
 	install -d $(INSTALLDIR)/bin
