Index: project/build/dotnet.mk
===================================================================
--- project/build/dotnet.mk	(revision 1523)
+++ project/build/dotnet.mk	(working copy)
@@ -6,7 +6,11 @@
 				${DOTNET_UNMANAGED_DLL}				\
 				${DOTNET_DEMO_EXE}
 
+ifeq (${SWIG_OSNAME}, linux)
+DOTNET_BUILD		= ${DOTNET_WRAP}
+else
 DOTNET_BUILD		= ${DOTNET_WRAP}  ${DOTNET_ALL}
+endif
 DOTNET_CXX_FLAGS	= -fPIC `${SWIG_LIBRETS_CONFIG} --cflags`
 DOTNET_DEMO_EXE		= ${DOTNET_GETOBJECT_EXE}				\
 				${DOTNET_INTERLEAVED_EXE}			\
@@ -68,6 +72,7 @@
 DOTNET_UPDATE_EXE	= ${DOTNET_OBJ_DIR}/Update.exe
 DOTNET_UPDATE_SRC	= ${DOTNET_DIR}/Update.cs ${DOTNET_DIR}/Options.cs
 DOTNET_WRAP		= ${DOTNET_OBJ_DIR}/librets_wrap.cpp
+DOTNET_KEYFILE		= ${DOTNET_DIR}/librets.snk
 
 
 ${DOTNET_WRAP}: ${SWIG_FILES} 
@@ -91,7 +96,7 @@
 #
 
 ${DOTNET_MANAGED_DLL}:	${DOTNET_UNMANAGED_DLL} ${DOTNET_MANAGED_SRC}
-	${MCS} -target:library -out:${DOTNET_MANAGED_DLL} ${DOTNET_PLATFORM} ${DOTNET_MANAGED_SRC}
+	${MCS} -target:library -keyfile:${DOTNET_KEYFILE} -out:${DOTNET_MANAGED_DLL} ${DOTNET_PLATFORM} ${DOTNET_MANAGED_SRC}
 
 ${DOTNET_GETOBJECT_EXE}:	${DOTNET_GETOBJECT_SRC}
 	${MCS} -r:${DOTNET_MANAGED_DLL} ${DOTNET_PLATFORM} -out:${DOTNET_GETOBJECT_EXE} ${DOTNET_GETOBJECT_SRC}
