Index: project/build/ac-macros/swig.m4
===================================================================
--- project/build/ac-macros/swig.m4	(revision 1501)
+++ project/build/ac-macros/swig.m4	(revision 1522)
@@ -72,7 +72,7 @@
 
             if test "$generics" = "1"; then
                 if test "$GMCS" != "no"; then
-                    MCS=GMCS
+                    MCS=$GMCS
                 fi
             fi
             if test "$MCS" != "no"; then
