--- Makefile.am
+++ Makefile.am
@@ -57,6 +57,22 @@
 	NEWS \
 	README
 
+## Headers
+
+includeclhdir = $(prefix)/include/$(PACKAGE)/client/linux/handler
+includeclh_HEADERS = src/client/linux/handler/exception_handler.h
+
+includecdir = $(prefix)/include/$(PACKAGE)/common
+includec_HEADERS = src/google_breakpad/common/*.h
+
+includeclmdir = $(prefix)/include/$(PACKAGE)/client/linux/minidump_writer
+includeclm_HEADERS = src/client/linux/minidump_writer/*.h
+
+includeclcdir = $(prefix)/include/$(PACKAGE)/client/linux/crash_generation
+includeclc_HEADERS = src/client/linux/crash_generation/*.h
+
+includepdir = $(prefix)/include/$(PACKAGE)/processor
+includep_HEADERS = src/processor/scoped_ptr.h
 
 ## Libraries
 noinst_LIBRARIES =
