diff -ur OpenFOAM-1.4.1-src/applications/utilities/mesh/conversion/fluent3DMeshToFoam/fluentMeshToFoam.L OpenFOAM-1.4.1/applications/utilities/mesh/conversion/fluent3DMeshToFoam/fluentMeshToFoam.L
--- OpenFOAM-1.4.1-src/applications/utilities/mesh/conversion/fluent3DMeshToFoam/fluentMeshToFoam.L	2007-06-26 19:35:12.000000000 +0200
+++ OpenFOAM-1.4.1/applications/utilities/mesh/conversion/fluent3DMeshToFoam/fluentMeshToFoam.L	2007-10-28 18:39:30.000000000 +0100
@@ -130,11 +130,11 @@
 nl                         (\n|\r)
 spaceNl                    ({space}|{nl})*
 
-alpha                      [_[:alpha:]]
-digit                      [[:digit:]]
-decDigit                   [[:digit:]]
+alpha                      [_[:alpha:] ]
+digit                      [[:digit:] ]
+decDigit                   [[:digit:] ]
 octalDigit                 [0-7]
-hexDigit                   [[:xdigit:]]
+hexDigit                   [[:xdigit:] ]
 
 lbrac                      "("
 rbrac                      ")"
diff -ur OpenFOAM-1.4.1-src/applications/utilities/mesh/conversion/fluentMeshToFoam/fluentMeshToFoam.L OpenFOAM-1.4.1/applications/utilities/mesh/conversion/fluentMeshToFoam/fluentMeshToFoam.L
--- OpenFOAM-1.4.1-src/applications/utilities/mesh/conversion/fluentMeshToFoam/fluentMeshToFoam.L	2007-05-24 12:43:38.000000000 +0200
+++ OpenFOAM-1.4.1/applications/utilities/mesh/conversion/fluentMeshToFoam/fluentMeshToFoam.L	2007-10-28 18:38:42.000000000 +0100
@@ -115,11 +115,11 @@
 cspace                     ","{space}
 spaceNl                    ({space}|\n|\r)*
 
-alpha                      [_[:alpha:]]
-digit                      [[:digit:]]
-decDigit                   [[:digit:]]
+alpha                      [_[:alpha:] ]
+digit                      [[:digit:] ]
+decDigit                   [[:digit:] ]
 octalDigit                 [0-7]
-hexDigit                   [[:xdigit:]]
+hexDigit                   [[:xdigit:] ]
 
 lbrac                      "("
 rbrac                      ")"
diff -ur OpenFOAM-1.4.1-src/applications/utilities/mesh/conversion/gambitToFoam/gambitToFoam.L OpenFOAM-1.4.1/applications/utilities/mesh/conversion/gambitToFoam/gambitToFoam.L
--- OpenFOAM-1.4.1-src/applications/utilities/mesh/conversion/gambitToFoam/gambitToFoam.L	2007-04-22 23:09:25.000000000 +0200
+++ OpenFOAM-1.4.1/applications/utilities/mesh/conversion/gambitToFoam/gambitToFoam.L	2007-10-28 18:39:06.000000000 +0100
@@ -91,11 +91,11 @@
 cspace                ","{space}
 spaceNl               ({space}|\n)*
 
-alpha                 [_[:alpha:]]
-digit                 [[:digit:]]
-dec_digit             [[:digit:]]
+alpha                 [_[:alpha:] ]
+digit                 [[:digit:] ]
+dec_digit             [[:digit:] ]
 octal_digit           [0-7]
-hex_digit             [[:xdigit:]]
+hex_digit             [[:xdigit:] ]
 
 lbrac                      "("
 rbrac                      ")"
diff -ur OpenFOAM-1.4.1-src/applications/utilities/mesh/manipulation/patchTool/Java/Allwmake OpenFOAM-1.4.1/applications/utilities/mesh/manipulation/patchTool/Java/Allwmake
--- OpenFOAM-1.4.1-src/applications/utilities/mesh/manipulation/patchTool/Java/Allwmake	2006-04-05 23:21:20.000000000 +0200
+++ OpenFOAM-1.4.1/applications/utilities/mesh/manipulation/patchTool/Java/Allwmake	2007-10-26 13:34:42.000000000 +0200
@@ -2,5 +2,5 @@
 set -x
 
 rmclassall
-javac -classpath ".:../lib/j3d-org.jar" -d Make/classes/ PatchTool/PatchTool.java
+javac -classpath ".:../lib/j3d-org.jar:/usr/share/sun-java3d-bin/lib/vecmath.jar:/usr/share/sun-java3d-bin/lib/j3dutils.jar:/usr/share/sun-java3d-bin/lib/j3dcore.jar" -d Make/classes/ PatchTool/PatchTool.java
 jar cf ../lib/PatchTool.jar -C Make/classes .
diff -ur OpenFOAM-1.4.1-src/applications/utilities/mesh/manipulation/patchTool/Java/Make/options OpenFOAM-1.4.1/applications/utilities/mesh/manipulation/patchTool/Java/Make/options
--- OpenFOAM-1.4.1-src/applications/utilities/mesh/manipulation/patchTool/Java/Make/options	2004-03-23 13:42:24.000000000 +0100
+++ OpenFOAM-1.4.1/applications/utilities/mesh/manipulation/patchTool/Java/Make/options	2007-10-26 13:37:33.000000000 +0200
@@ -1,6 +1,6 @@
 /* EXE_INC = ../lib/j3d-org.jar */
 
-EXE_INC = -classpath ".:Make/classes:../lib/j3d-org.jar"
+EXE_INC = -classpath ".:Make/classes:../lib/j3d-org.jar:/usr/share/sun-java3d-bin/lib/vecmath.jar:/usr/share/sun-java3d-bin/lib/j3dutils.jar:/usr/share/sun-java3d-bin/lib/j3dcore.jar"
 
 LIB_LIBS = META-INF/MANIFEST.MF
 
Dateien OpenFOAM-1.4.1-src/applications/utilities/mesh/manipulation/patchTool/lib/PatchTool.jar und OpenFOAM-1.4.1/applications/utilities/mesh/manipulation/patchTool/lib/PatchTool.jar sind verschieden.
diff -ur OpenFOAM-1.4.1-src/applications/utilities/postProcessing/graphics/PVFoamReader/vtkFoam/Make/options OpenFOAM-1.4.1/applications/utilities/postProcessing/graphics/PVFoamReader/vtkFoam/Make/options
--- OpenFOAM-1.4.1-src/applications/utilities/postProcessing/graphics/PVFoamReader/vtkFoam/Make/options	2005-12-13 12:34:08.000000000 +0100
+++ OpenFOAM-1.4.1/applications/utilities/postProcessing/graphics/PVFoamReader/vtkFoam/Make/options	2007-10-25 18:58:31.000000000 +0200
@@ -1,6 +1,6 @@
 EXE_INC = \
     -I$(LIB_SRC)/finiteVolume/lnInclude \
-    -I$(ParaView_INST_DIR)/include \
+    -I$(ParaView_INST_DIR)/include/vtk-5.0 \
     -I../PVFoamReader
 
 LIB_LIBS = \
diff -ur OpenFOAM-1.4.1-src/.bashrc OpenFOAM-1.4.1/.bashrc
--- OpenFOAM-1.4.1-src/.bashrc	2007-08-01 14:40:07.000000000 +0200
+++ OpenFOAM-1.4.1/.bashrc	2007-10-28 12:47:52.000000000 +0100
@@ -56,7 +56,7 @@
 export PATH=${WM_PROJECT_DIR}/bin:$WM_PROJECT_INST_DIR/$WM_ARCH/bin:$PATH
 
 #- Location of the jobControl directory
-export FOAM_JOB_DIR=$WM_PROJECT_INST_DIR/jobControl
+export FOAM_JOB_DIR=$HOME/$WM_PROJECT/jobControl
 
 export WM_DIR=$WM_PROJECT_DIR/wmake
 export WM_LINK_LANGUAGE=c++
@@ -97,7 +97,7 @@
 WM_COMPILER_LIB=
 
 if [ "$WM_COMPILER" = "Gcc" -o "$machineTest" = "Linux" -a "$WM_COMPILER" = "" ]; then
-    export WM_COMPILER_DIR=$WM_PROJECT_INST_DIR/$WM_ARCH/gcc-4.2.1$WM_COMPILER_ARCH
+    export WM_COMPILER_DIR="/usr"
     WM_COMPILER_BIN="$WM_COMPILER_DIR/bin:$WM_COMPILER_DIR/../gdb-6.4/bin"
     WM_COMPILER_LIB=$WM_COMPILER_DIR/lib${WM_COMPILER_LIB_ARCH}:$WM_COMPILER_DIR/lib:
 fi
@@ -115,8 +115,8 @@
 # ~~~~~~~~~~~~~~~~~
 if [ "$machineTest" = "Linux" ]; then
 
-    export JAVA_HOME=$WM_PROJECT_INST_DIR/$WM_ARCH/j2sdk1.4.2_05
-
+    #export JAVA_HOME=$WM_PROJECT_INST_DIR/$WM_ARCH/j2sdk1.4.2_05
+    export JAVA_HOME=${JAVA_HOME} #Set Java jdk env variable to system $JAVA_HOME
     processor=`uname -m`
 
     if [ "$processor" = "ia64" ]; then
@@ -176,9 +176,11 @@
 
 if [ .$WM_MPLIB = .OPENMPI ]; then
 
-    export OPENMPI_VERSION=1.2.3
-    export OPENMPI_HOME=$FOAM_SRC/openmpi-$OPENMPI_VERSION
-    export OPENMPI_ARCH_PATH=$OPENMPI_HOME/platforms/$WM_OPTIONS
+#     export OPENMPI_VERSION=1.2.3
+    export OPENMPI_VERSION=`/usr/bin/ompi_info --version ompi full --parsable | grep ompi:version:full | cut -d: -f4-`
+#    export OPENMPI_HOME=$FOAM_SRC/openmpi-$OPENMPI_VERSION
+#    export OPENMPI_ARCH_PATH=$OPENMPI_HOME/platforms/$WM_OPTIONS
+    export OPENMPI_ARCH_PATH=/usr
 
     # Tell OpenMPI where to find it's install directory
     export OPAL_PREFIX=$OPENMPI_ARCH_PATH
@@ -190,26 +192,29 @@
 
 elif [ .$WM_MPLIB = .LAM ]; then
 
-    export LAM_VERSION=7.1.2
-    export LAMHOME=$FOAM_SRC/lam-$LAM_VERSION
-    export LAM_ARCH_PATH=$LAMHOME/platforms/$WM_OPTIONS
+#     export LAM_VERSION=7.1.2
+    export LAM_VERSION=`/usr/bin/laminfo -version lam full | awk '{print $$2}'`
+#     export LAMHOME=$FOAM_SRC/lam-$LAM_VERSION
+#     export LAM_ARCH_PATH=$LAMHOME/platforms/$WM_OPTIONS
+    export LAM_ARCH_PATH=/usr
 
     AddLib $LAM_ARCH_PATH/lib
     AddPath $LAM_ARCH_PATH/bin
 
-    export FOAM_MPI_LIBBIN=$FOAM_LIBBIN/lam-$LAM_VERSION
+     export FOAM_MPI_LIBBIN=$FOAM_LIBBIN/lam-$LAM_VERSION
 
 elif [ .$WM_MPLIB = .MPICH ]; then
 
-    export MPICH_VERSION=1.2.4
-    export MPICH_PATH=$FOAM_SRC/mpich-$MPICH_VERSION
-    export MPICH_ARCH_PATH=$MPICH_PATH/platforms/$WM_OPTIONS
+#     export MPICH_VERSION=1.2.4
+    export MPICH_VERSION=`/usr/bin/mpich2version --version`
+#     export MPICH_PATH=$FOAM_SRC/mpich-$MPICH_VERSION
+    export MPICH_ARCH_PATH=/usr
     export MPICH_ROOT=$MPICH_ARCH_PATH
 
     AddLib $MPICH_ARCH_PATH/lib
     AddPath $MPICH_ARCH_PATH/bin
 
-    export FOAM_MPI_LIBBIN=$FOAM_LIBBIN/mpich-$MPICH_VERSION
+     export FOAM_MPI_LIBBIN=$FOAM_LIBBIN/mpich-$MPICH_VERSION
 
 elif [ .$WM_MPLIB = .MPICH-GM ]; then
 
diff -ur OpenFOAM-1.4.1-src/.cshrc OpenFOAM-1.4.1/.cshrc
--- OpenFOAM-1.4.1-src/.cshrc	2007-08-01 14:40:15.000000000 +0200
+++ OpenFOAM-1.4.1/.cshrc	2007-10-28 12:47:28.000000000 +0100
@@ -45,7 +45,7 @@
 set path=($WM_PROJECT_DIR/bin $WM_PROJECT_INST_DIR/$WM_ARCH/bin $path)
 
 #- Location of the jobControl directory
-setenv FOAM_JOB_DIR $WM_PROJECT_INST_DIR/jobControl
+setenv FOAM_JOB_DIR $HOME/$WM_PROJECT/jobControl
 
 setenv WM_DIR $WM_PROJECT_DIR/wmake
 setenv WM_LINK_LANGUAGE c++
@@ -86,7 +86,7 @@
 set WM_COMPILER_LIB=
 
 if ($WM_COMPILER == "Gcc" || $machineTest == "Linux" && $?WM_COMPILER) then
-    setenv WM_COMPILER_DIR $WM_PROJECT_INST_DIR/$WM_ARCH/gcc-4.2.1$WM_COMPILER_ARCH
+    setenv WM_COMPILER_DIR "/usr"
     set WM_COMPILER_BIN="$WM_COMPILER_DIR/bin $WM_COMPILER_DIR/../gdb-6.4/bin"
     set WM_COMPILER_LIB=$WM_COMPILER_DIR/lib${WM_COMPILER_LIB_ARCH}:$WM_COMPILER_DIR/lib:
 endif
@@ -109,7 +109,8 @@
 # ~~~~~~~~~~~~~~~~~
 if ($machineTest == Linux) then
 
-    setenv JAVA_HOME $WM_PROJECT_INST_DIR/$WM_ARCH/j2sdk1.4.2_05
+    #setenv JAVA_HOME $WM_PROJECT_INST_DIR/$WM_ARCH/j2sdk1.4.2_05
+    setenv JAVA_HOME ${JAVA_HOME} #Set Java jdk env variable to system $JAVA_HOME
 
     set processor = `uname -m`
 
@@ -170,9 +171,11 @@
 
 if ($WM_MPLIB == OPENMPI) then
 
-    setenv OPENMPI_VERSION 1.2.3
-    setenv OPENMPI_HOME $FOAM_SRC/openmpi-$OPENMPI_VERSION
-    setenv OPENMPI_ARCH_PATH $OPENMPI_HOME/platforms/$WM_OPTIONS
+#     setenv OPENMPI_VERSION 1.2.3
+    setenv OPENMPI_VERSION=`/usr/bin/ompi_info --version ompi full --parsable | grep ompi:version:full | cut -d: -f4-`
+#     setenv OPENMPI_HOME $FOAM_SRC/openmpi-$OPENMPI_VERSION
+#     setenv OPENMPI_ARCH_PATH $OPENMPI_HOME/platforms/$WM_OPTIONS
+    setenv OPENMPI_ARCH_PATH=/usr
 
     # Tell OpenMPI where to find it's install directory
     setenv OPAL_PREFIX $OPENMPI_ARCH_PATH
@@ -184,9 +187,11 @@
 
 else if ($WM_MPLIB == LAM) then
 
-    setenv LAM_VERSION 7.1.2
-    setenv LAMHOME $FOAM_SRC/lam-$LAM_VERSION
-    setenv LAM_ARCH_PATH $LAMHOME/platforms/$WM_OPTIONS
+#     setenv LAM_VERSION 7.1.2
+    setenv LAM_VERSION=`/usr/bin/laminfo -version lam full | awk '{print $$2}'`
+#     setenv LAMHOME $FOAM_SRC/lam-$LAM_VERSION
+#     setenv LAM_ARCH_PATH $LAMHOME/platforms/$WM_OPTIONS
+    setenv LAM_ARCH_PATH=/usr
 
     AddLib $LAM_ARCH_PATH/lib
     AddPath $LAM_ARCH_PATH/bin
@@ -195,9 +200,10 @@
 
 else if ($WM_MPLIB == MPICH) then
 
-    setenv MPICH_VERSION 1.2.4
-    setenv MPICH_PATH $FOAM_SRC/mpich-$MPICH_VERSION
-    setenv MPICH_ARCH_PATH $MPICH_PATH/platforms/$WM_OPTIONS
+#     setenv MPICH_VERSION 1.2.4
+     setenv MPICH_VERSION=`/usr/bin/mpich2version --version`
+#     setenv MPICH_PATH $FOAM_SRC/mpich-$MPICH_VERSION
+    setenv MPICH_ARCH_PATH /usr
     setenv MPICH_ROOT $MPICH_ARCH_PATH
 
     AddLib $MPICH_ARCH_PATH/lib
diff -ur OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/apps/paraview/bashrc OpenFOAM-1.4.1/.OpenFOAM-1.4.1/apps/paraview/bashrc
--- OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/apps/paraview/bashrc	2007-05-08 21:05:21.000000000 +0200
+++ OpenFOAM-1.4.1/.OpenFOAM-1.4.1/apps/paraview/bashrc	2007-10-28 12:05:16.000000000 +0100
@@ -38,22 +38,25 @@
     fi
 fi
 
-export CMAKE_HOME=$WM_PROJECT_INST_DIR/$WM_ARCH/cmake-2.4.6
+# export CMAKE_HOME=$WM_PROJECT_INST_DIR/$WM_ARCH/cmake-2.4.6
 
-if [ -r $CMAKE_HOME ]; then
-    export PATH=$CMAKE_HOME/bin:$PATH
-fi
+# if [ -r $CMAKE_HOME ]; then
+#     export PATH=$CMAKE_HOME/bin:$PATH
+# fi
+
+export CMAKE_HOME=/usr
+
+export ParaView_VERSION=2.6.2
 
-export ParaView_VERSION=2.4.4
+export ParaView_INST_DIR=/usr
+export ParaView_DIR=$ParaView_INST_DIR/lib/ParaView-2.6
 
-export ParaView_INST_DIR=$WM_PROJECT_INST_DIR/$WM_ARCH/paraview-$ParaView_VERSION
-export ParaView_DIR=$ParaView_INST_DIR/lib/paraview-2.4
 
 if [ -r $ParaView_INST_DIR ]; then
     export PV_INTERFACE_PATH=$FOAM_UTILITIES/postProcessing/graphics/PVFoamReader/PVFoamReader/Make
 
-    export PATH=$ParaView_INST_DIR/bin:$PATH
-    export LD_LIBRARY_PATH=${ParaView_INST_DIR}/lib:${ParaView_INST_DIR}/lib/vtk:${LD_LIBRARY_PATH}
+#    export PATH=$ParaView_INST_DIR/bin:$PATH
+#    export LD_LIBRARY_PATH=${ParaView_INST_DIR}/lib:${ParaView_INST_DIR}/lib/vtk:${LD_LIBRARY_PATH}
 fi
 
 if [ -r $ParaView_DIR ]; then
diff -ur OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/apps/paraview/cshrc OpenFOAM-1.4.1/.OpenFOAM-1.4.1/apps/paraview/cshrc
--- OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/apps/paraview/cshrc	2007-05-08 21:05:06.000000000 +0200
+++ OpenFOAM-1.4.1/.OpenFOAM-1.4.1/apps/paraview/cshrc	2007-10-28 12:01:33.000000000 +0100
@@ -38,22 +38,24 @@
     endif
 endif
 
-setenv CMAKE_HOME $WM_PROJECT_INST_DIR/$WM_ARCH/cmake-2.4.6
+# setenv CMAKE_HOME $WM_PROJECT_INST_DIR/$WM_ARCH/cmake-2.4.6
 
-if ( -r $CMAKE_HOME ) then
-    set path=($CMAKE_HOME/bin $path)
-endif
+# if ( -r $CMAKE_HOME ) then
+#     set path=($CMAKE_HOME/bin $path)
+# endif
+
+setenv CMAKE_HOME /usr
 
-setenv ParaView_VERSION 2.4.4
+setenv ParaView_VERSION 2.6.2
 
-setenv ParaView_INST_DIR $WM_PROJECT_INST_DIR/$WM_ARCH/paraview-$ParaView_VERSION
-setenv ParaView_DIR $ParaView_INST_DIR/lib/paraview-2.4
+setenv ParaView_INST_DIR /usr
+setenv ParaView_DIR $ParaView_INST_DIR/lib/paraview-$ParaView_VERSION
 
 if ( -r $ParaView_INST_DIR ) then
     setenv PV_INTERFACE_PATH $FOAM_UTILITIES/postProcessing/graphics/PVFoamReader/PVFoamReader/Make
 
-    set path=($ParaView_INST_DIR/bin $path)
-    setenv LD_LIBRARY_PATH ${ParaView_INST_DIR}/lib:${ParaView_INST_DIR}/lib/vtk:${LD_LIBRARY_PATH}
+#     set path=($ParaView_INST_DIR/bin $path)
+#     setenv LD_LIBRARY_PATH ${ParaView_INST_DIR}/lib:${ParaView_INST_DIR}/lib/vtk:${LD_LIBRARY_PATH}
 endif
 
 if ( -r $ParaView_DIR ) then
diff -ur OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/bashrc OpenFOAM-1.4.1/.OpenFOAM-1.4.1/bashrc
--- OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/bashrc	2007-08-02 23:28:12.000000000 +0200
+++ OpenFOAM-1.4.1/.OpenFOAM-1.4.1/bashrc	2007-10-28 11:56:47.000000000 +0100
@@ -41,10 +41,11 @@
 #
 # Location of FOAM installation
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-export WM_PROJECT_INST_DIR=$HOME/$WM_PROJECT
+#export WM_PROJECT_INST_DIR=$HOME/$WM_PROJECT
 #export WM_PROJECT_INST_DIR=~$WM_PROJECT
 #export WM_PROJECT_INST_DIR=$HOME/$WM_PROJECT/$LOGNAME-$WM_PROJECT_VERSION
 #export WM_PROJECT_INST_DIR=/usr/local/$WM_PROJECT
+export WM_PROJECT_INST_DIR=/usr/lib/$WM_PROJECT
 
 # Check that the installation directory has been set and exists
 if [ "$WM_PROJECT_INST_DIR" -a -d $WM_PROJECT_INST_DIR ]; then
@@ -58,7 +59,7 @@
 
 # Compiler (if set to "" use the system compiler)
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-export WM_COMPILER=Gcc
+export WM_COMPILER=
 export WM_COMPILER_ARCH=
 export WM_COMPILER_LIB_ARCH=
 
@@ -151,8 +152,8 @@
     #export WM_COMPILER_LIB_ARCH=/mabi=64
 
     #export WM_MPLIB=
-    export WM_MPLIB=MPI
-    #export WM_MPLIB=OPENMPI
+    #export WM_MPLIB=MPI
+    export WM_MPLIB=OPENMPI
     #export WM_MPLIB=LAM
     #export WM_MPLIB=MPICH
     #export WM_MPLIB=MPICH-GM
@@ -216,8 +217,8 @@
 
 # Source user setup files for optional packages
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/ensightFoam/bashrc
-SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/paraview/bashrc
+#SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/ensightFoam/bashrc
+#SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/paraview/bashrc
 #SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/ParaView3/bashrc
 
 
diff -ur OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/cshrc OpenFOAM-1.4.1/.OpenFOAM-1.4.1/cshrc
--- OpenFOAM-1.4.1-src/.OpenFOAM-1.4.1/cshrc	2007-07-31 17:39:07.000000000 +0200
+++ OpenFOAM-1.4.1/.OpenFOAM-1.4.1/cshrc	2007-10-28 11:55:19.000000000 +0100
@@ -41,10 +41,11 @@
 #
 # Location of FOAM installation
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-setenv WM_PROJECT_INST_DIR $HOME/$WM_PROJECT
+#setenv WM_PROJECT_INST_DIR $HOME/$WM_PROJECT
 #setenv WM_PROJECT_INST_DIR ~$WM_PROJECT
 #setenv WM_PROJECT_INST_DIR $HOME/$WM_PROJECT/$LOGNAME-$WM_PROJECT_VERSION
 #setenv WM_PROJECT_INST_DIR /usr/local/$WM_PROJECT
+setenv WM_PROJECT_INST_DIR /usr/lib/$WM_PROJECT
 
 # Check that the installation directory has been set and exists
 if ( $?WM_PROJECT_INST_DIR ) then
@@ -59,7 +60,7 @@
 
 # Compiler (if set to "" use the system compiler)
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-setenv WM_COMPILER Gcc
+setenv WM_COMPILER
 setenv WM_COMPILER_ARCH
 setenv WM_COMPILER_LIB_ARCH
 
@@ -151,8 +152,8 @@
     #setenv WM_COMPILER_LIB_ARCH /mabi=64
 
     #setenv WM_MPLIB
-    setenv WM_MPLIB MPI
-    #setenv WM_MPLIB OPENMPI
+    #setenv WM_MPLIB MPI
+    setenv WM_MPLIB OPENMPI
     #setenv WM_MPLIB LAM
     #setenv WM_MPLIB MPICH
     #setenv WM_MPLIB MPICH-GM
@@ -219,8 +220,8 @@
 
 # Source user setup files for optional packages
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/ensightFoam/cshrc
-SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/paraview/cshrc
+#SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/ensightFoam/cshrc
+#SOURCE $WM_PROJECT_DIR/$FOAM_DOT_DIR/apps/paraview/cshrc
 
 
 # Again clean environment (path, PATH, MANPATH, LD_LIBRARY_PATH)
diff -ur OpenFOAM-1.4.1-src/src/Allwmake OpenFOAM-1.4.1/src/Allwmake
--- OpenFOAM-1.4.1-src/src/Allwmake	2007-07-27 17:54:28.000000000 +0200
+++ OpenFOAM-1.4.1/src/Allwmake	2007-10-18 21:04:12.000000000 +0200
@@ -3,103 +3,103 @@
 
 (cd $FOAM_SRC/OpenFOAM ; wmakeLnInclude . )
 
-wmake libso zlib-1.2.1
-
-if [ "$WM_MPLIB" = "OPENMPI" -a ! -r $OPENMPI_ARCH_PATH/lib/libmpi.a -a ! -r $OPENMPI_ARCH_PATH/lib/libmpi.so ]
-then
-
-    cd $OPENMPI_HOME
-
-        gmake distclean
-        rm -rf $OPENMPI_ARCH_PATH
-
-        ./configure \
-            --prefix=$OPENMPI_ARCH_PATH \
-            --disable-mpirun-prefix-by-default \
-            --disable-orterun-prefix-by-default \
-            --enable-shared --disable-static \
-            --disable-mpi-f77 --disable-mpi-f90 --disable-mpi-cxx \
-            --disable-mpi-profile
-
-        gmake
-        gmake install
-        gmake distclean
-
-    cd $FOAM_SRC
-fi
-
-if [ "$WM_MPLIB" = "LAM" -a ! -r $LAM_ARCH_PATH/lib/libmpi.a -a ! -r $LAM_ARCH_PATH/lib/libmpi.so ]
-then
-
-    cd $LAMHOME
-
-        gmake distclean
-        rm -rf $LAM_ARCH_PATH
-
-        ./configure \
-            --prefix=$LAM_ARCH_PATH \
-            --enable-shared \
-            --disable-static \
-            --without-romio \
-            --without-mpi2cpp \
-            --without-profiling \
-            --without-fc
-
-        gmake
-        gmake install
-        gmake distclean
-
-    cd $FOAM_SRC
-fi
-
-if [ "$WM_MPLIB" = "MPICH" -a ! -r $MPICH_ARCH_PATH/lib/libmpich.a -a ! -r $MPICH_ARCH_PATH/lib/libmpich.so ]
-then
-    
-    cd $MPICH_PATH
-
-        make distclean
-        rm -rf $MPICH_ARCH_PATH
-        rm util/machines/machines.*
-
-        ./configure \
-            --without-mpe \
-            --disable-f77 \
-            --disable-f90 \
-            --disable-f90modules \
-            --disable-c++ \
-            --disable-mpedbg \
-            --disable-devdebug \
-            --disable-debug \
-            --enable-sharedlib=$MPICH_ARCH_PATH/lib \
-            --with-device=ch_p4 \
-            -prefix=$MPICH_ARCH_PATH
-        make
-        make install
-        make distclean
-
-        if [ -r $MPICH_ARCH_PATH ]
-        then
-            cd $MPICH_ARCH_PATH/bin
-            for file in *
-            do
-                sed s%$MPICH_ARCH_PATH%'$MPICH_ARCH_PATH'%g $file > temp.$$
-                mv temp.$$ $file
-                chmod ugo+rx $file
-            done
-
-            cd $MPICH_ARCH_PATH/lib
-
-            if [ -r libmpich.so.1.0 ]
-            then
-                rm *.so
-                ln -s libmpich.so.1.0 libmpich.so
-            fi
-
-            cd $MPICH_ARCH_PATH
-        fi
-
-    cd $FOAM_SRC
-fi
+# wmake libso zlib-1.2.1
+#
+# if [ "$WM_MPLIB" = "OPENMPI" -a ! -r $OPENMPI_ARCH_PATH/lib/libmpi.a -a ! -r $OPENMPI_ARCH_PATH/lib/libmpi.so ]
+# then
+# 
+#     cd $OPENMPI_HOME
+# 
+#         gmake distclean
+#         rm -rf $OPENMPI_ARCH_PATH
+# 
+#         ./configure \
+#             --prefix=$OPENMPI_ARCH_PATH \
+#             --disable-mpirun-prefix-by-default \
+#             --disable-orterun-prefix-by-default \
+#             --enable-shared --disable-static \
+#             --disable-mpi-f77 --disable-mpi-f90 --disable-mpi-cxx \
+#             --disable-mpi-profile
+# 
+#         gmake
+#         gmake install
+#         gmake distclean
+# 
+#     cd $FOAM_SRC
+# fi
+# 
+# if [ "$WM_MPLIB" = "LAM" -a ! -r $LAM_ARCH_PATH/lib/libmpi.a -a ! -r $LAM_ARCH_PATH/lib/libmpi.so ]
+# then
+# 
+#     cd $LAMHOME
+# 
+#         gmake distclean
+#         rm -rf $LAM_ARCH_PATH
+# 
+#         ./configure \
+#             --prefix=$LAM_ARCH_PATH \
+#             --enable-shared \
+#             --disable-static \
+#             --without-romio \
+#             --without-mpi2cpp \
+#             --without-profiling \
+#             --without-fc
+# 
+#         gmake
+#         gmake install
+#         gmake distclean
+# 
+#     cd $FOAM_SRC
+# fi
+# 
+# if [ "$WM_MPLIB" = "MPICH" -a ! -r $MPICH_ARCH_PATH/lib/libmpich.a -a ! -r $MPICH_ARCH_PATH/lib/libmpich.so ]
+# then
+#     
+#     cd $MPICH_PATH
+# 
+#         make distclean
+#         rm -rf $MPICH_ARCH_PATH
+#         rm util/machines/machines.*
+# 
+#         ./configure \
+#             --without-mpe \
+#             --disable-f77 \
+#             --disable-f90 \
+#             --disable-f90modules \
+#             --disable-c++ \
+#             --disable-mpedbg \
+#             --disable-devdebug \
+#             --disable-debug \
+#             --enable-sharedlib=$MPICH_ARCH_PATH/lib \
+#             --with-device=ch_p4 \
+#             -prefix=$MPICH_ARCH_PATH
+#         make
+#         make install
+#         make distclean
+# 
+#         if [ -r $MPICH_ARCH_PATH ]
+#         then
+#             cd $MPICH_ARCH_PATH/bin
+#             for file in *
+#             do
+#                 sed s%$MPICH_ARCH_PATH%'$MPICH_ARCH_PATH'%g $file > temp.$$
+#                 mv temp.$$ $file
+#                 chmod ugo+rx $file
+#             done
+# 
+#             cd $MPICH_ARCH_PATH/lib
+# 
+#             if [ -r libmpich.so.1.0 ]
+#             then
+#                 rm *.so
+#                 ln -s libmpich.so.1.0 libmpich.so
+#             fi
+# 
+#             cd $MPICH_ARCH_PATH
+#         fi
+# 
+#     cd $FOAM_SRC
+# fi
 
 (cd Pstream ; ./Allwmake)
 wmake libso OpenFOAM
diff -ur OpenFOAM-1.4.1-src/wmake/rules/General/flex++ OpenFOAM-1.4.1/wmake/rules/General/flex++
--- OpenFOAM-1.4.1-src/wmake/rules/General/flex++	2006-08-18 17:35:12.000000000 +0200
+++ OpenFOAM-1.4.1/wmake/rules/General/flex++	2007-10-28 17:38:46.000000000 +0100
@@ -1,6 +1,6 @@
 .SUFFIXES: .L
 
-Ltoo = flex++ -f $$SOURCE ; mv lex.yy.cc $*.C ; $(CC) $(c++FLAGS) -c $*.C -o $@ 
+Ltoo = flex --c++ -f $$SOURCE ; mv lex.yy.cc $*.C ; $(CC) $(c++FLAGS) -c $*.C -o $@ 
 
 .L.dep:
 	$(MAKE_DEP)
diff -ur OpenFOAM-1.4.1-src/wmake/rules/General/flex++.test OpenFOAM-1.4.1/wmake/rules/General/flex++.test
--- OpenFOAM-1.4.1-src/wmake/rules/General/flex++.test	2007-03-13 17:27:58.000000000 +0100
+++ OpenFOAM-1.4.1/wmake/rules/General/flex++.test	2007-10-28 17:39:09.000000000 +0100
@@ -1,6 +1,6 @@
 .SUFFIXES: .L
 
-Ltoo = flex++ $$SOURCE ; mv lex.yy.cc $*.C ; $(CC) $(c++FLAGS) -c $*.C -o $@ 
+Ltoo = flex --c++ $$SOURCE ; mv lex.yy.cc $*.C ; $(CC) $(c++FLAGS) -c $*.C -o $@ 
 
 .L.dep:
 	$(MAKE_DEP)
diff -ur OpenFOAM-1.4.1-src/src/thermophysicalModels/combustion/chemistryReaders/chemkinReader/chemkinLexer.L OpenFOAM-1.4.1/src/thermophysicalModels/combustion/chemistryReaders/chemkinReader/chemkinLexer.L
--- OpenFOAM-1.4.1-src/src/thermophysicalModels/combustion/chemistryReaders/chemkinReader/chemkinLexer.L	2007-04-08 18:39:23.000000000 +0200
+++ OpenFOAM-1.4.1/src/thermophysicalModels/combustion/chemistryReaders/chemkinReader/chemkinLexer.L	2007-10-28 18:39:58.000000000 +0100
@@ -109,7 +109,7 @@
 label                 [1-9]{dec_digit}*
 zeroLabel             {digit}*
 
-word                  ([[:alnum:]]|[[:punct:]])+
+word                  ([[:alnum:] ]|[[:punct:] ])+
 string                {word}({some_space}{word})*
 
 exponent_part         [eEdD][-+]?{digit}+
diff -ur OpenFOAM-1.4.1-src/src/triSurface/triSurface/interfaces/STL/readSTLASCII.L OpenFOAM-1.4.1/src/triSurface/triSurface/interfaces/STL/readSTLASCII.L
--- OpenFOAM-1.4.1-src/src/triSurface/triSurface/interfaces/STL/readSTLASCII.L	2007-05-25 16:47:55.000000000 +0200
+++ OpenFOAM-1.4.1/src/triSurface/triSurface/interfaces/STL/readSTLASCII.L	2007-10-28 18:28:07.000000000 +0100
@@ -159,7 +159,7 @@
 zeroLabel             {digit}*
 signedInteger         [-+]?{integer}
 
-word                  ([[:alnum:]]|[[:punct:]])*
+word                  ([[:alnum:] ]|[[:punct:] ])*
 string                {word}({some_space}{word})*
 
 exponent_part         [eE][-+]?{digit}+
