diff -ur OpenFOAM-1.4.1-src/src/Allwmake OpenFOAM-1.4.1/src/Allwmake
--- OpenFOAM-1.4.1-src/src/Allwmake	2007-07-27 17:54:28.000000000 +0200
+++ OpenFOAM-1.4.1/src/Allwmake	2007-10-18 21:04:12.000000000 +0200
@@ -141,13 +141,13 @@
 
 (cd malloc ; ./Allwmake)
 
-if [ ! -r $MICO_ARCH_PATH/lib/libmico${MICO_VERSION}.a ]
-then
-    cd $MICO_PATH
-    gmake distclean
-    ./configure --prefix=$MICO_ARCH_PATH --disable-shared --without-x
-    gmake
-    gmake install
-    gmake distclean
-    cd ..
-fi
+# if [ ! -r $MICO_ARCH_PATH/lib/libmico${MICO_VERSION}.a ]
+# then
+#     cd $MICO_PATH
+#     gmake distclean
+#     ./configure --prefix=$MICO_ARCH_PATH --disable-shared --without-x
+#     gmake
+#     gmake install
+#     gmake distclean
+#     cd ..
+# fi
