diff -ur sphinx-0.9.7-rc2.orig/Makefile.am sphinx-0.9.7-rc2/Makefile.am
--- sphinx-0.9.7-rc2.orig/Makefile.am	2007-03-31 12:41:51.000000000 +0000
+++ sphinx-0.9.7-rc2/Makefile.am	2007-03-31 12:47:05.000000000 +0000
@@ -1,8 +1,8 @@
 SUBDIRS = src
 EXTRA_DIST = api storage sphinx.conf.in sphinx.sln todo.txt example.sql
-sysconf_DATA = sphinx.conf.dist example.sql
+#sysconf_DATA = sphinx.conf.dist example.sql
 
 install-data-hook:
-	mkdir -p $(localstatedir)/data && mkdir -p $(localstatedir)/log
+#	mkdir -p $(localstatedir)/data && mkdir -p $(localstatedir)/log
 
 
