--- unoconv.~1~	2010-04-08 17:05:44.000000000 +0200
+++ unoconv	2010-04-08 17:09:15.000000000 +0200
@@ -375,7 +375,7 @@
                 error(2, "Trying to launch our own listener using %s." % bin)
                 try:
                     oopid = os.spawnvp(os.P_NOWAIT, bin, [bin, "-nologo", "-nodefault", "-accept=%s" % op.connection]);
-                    time.sleep(1)
+                    time.sleep(5)
                     unocontext = resolver.resolve("uno:%s" % op.connection)
                     break
                 except Exception, e:
