--- unoconv.~1~	2007-09-01 03:23:13.000000000 +0200
+++ unoconv	2010-04-03 12:09:13.703092418 +0200
@@ -17,7 +17,7 @@
 import getopt, sys, os, glob, time
 
 #extrapaths = ('/usr/lib/openoffice/program/', '/usr/lib/openoffice.org2.0/program/')
-extrapaths = glob.glob('/usr/lib*/openoffice*/program/') + glob.glob('/usr/lib*/ooo*/program') + [ '/Applications/NeoOffice.app/Contents/program', ]
+extrapaths = glob.glob('/usr/lib*/openoffice*/basis*/program/') + glob.glob('/usr/lib*/openoffice*/program/') + glob.glob('/usr/lib*/ooo*/program') + [ '/Applications/NeoOffice.app/Contents/program', ]
 for path in extrapaths:
     try:
         sys.path.append(path)   
