--- Tests/pysvn_test_common.mak.old	2008-10-02 22:30:42.000000000 +0200
+++ Tests/pysvn_test_common.mak	2009-03-11 19:22:17.192351068 +0100
@@ -85,9 +85,14 @@ new-05: test-05.unix.new.log
 	cp  test-05.unix.new.log test-05.unix.known_good-$(SVN_VERSION_MAJ_MIN).log
 
 test-06.unix.new.log: test-06.sh test-06.unix.known_good-$(SVN_VERSION_MAJ_MIN).log
-	-rm -rf testroot-06
-	if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 3 0; then PATH=$(SVN_BIN_DIR):$(PATH) PYTHON=$(PYTHON) ./test-06.sh >test-06.unix.new.log 2>&1; fi
-	if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 3 0; then $(PYTHON) benchmark_diff.py test-06.unix.known_good-$(SVN_VERSION_MAJ_MIN).log test-06.unix.new.log; fi
+	if [[ $${UID} -ne 0 ]]; \
+	then \
+		rm -f testroot-06; \
+		if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 3 0; then PATH=$(SVN_BIN_DIR):$(PATH) PYTHON=$(PYTHON) ./test-06.sh >test-06.unix.new.log 2>&1; fi; \
+		if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 3 0; then $(PYTHON) benchmark_diff.py test-06.unix.known_good-$(SVN_VERSION_MAJ_MIN).log test-06.unix.new.log; fi; \
+	else \
+		echo "Skipping $@, Reason: executed as root"; \
+	fi
 
 clean-06:
 	-rm -f test-06.unix.new.log
