--- src/NameResolver.cc.orig	2006-08-17 22:52:24.000000000 +0200
+++ src/NameResolver.cc	2006-08-17 22:52:31.000000000 +0200
@@ -24,7 +24,7 @@
 void callback(void* arg, int status, struct hostent* host) {
   NameResolver* resolverPtr = (NameResolver*)arg;
   if(status != ARES_SUCCESS) {
-    resolverPtr->error = ares_strerror(status, 0);
+    resolverPtr->error = ares_strerror(status);
     resolverPtr->status = NameResolver::STATUS_ERROR;
     return;
   }
