--- aria2-0.7.1/src/main.cc.orig	2006-08-20 20:41:01.000000000 +0200
+++ aria2-0.7.1/src/main.cc	2006-08-20 20:41:27.000000000 +0200
@@ -33,8 +33,6 @@
 #include "DownloadEngineFactory.h"
 #include "UrlRequestInfo.h"
 #include "TorrentRequestInfo.h"
-#include "MetalinkRequestInfo.h"
-#include "Xml2MetalinkProcessor.h"
 #include <deque>
 #include <algorithm>
 #include <time.h>
@@ -46,6 +44,11 @@
 extern int optind, opterr, optopt;
 #include <getopt.h>
 
+#ifdef ENABLE_METALINK
+#include "MetalinkRequestInfo.h"
+#include "Xml2MetalinkProcessor.h"
+#endif
+
 #ifdef HAVE_LIBSSL
 // for SSL
 # include <openssl/err.h>
