diff -ru MUTE_fileSharing-0.5_UnixSource~/MUTE/Makefile MUTE_fileSharing-0.5_UnixSource/MUTE/Makefile
--- MUTE_fileSharing-0.5_UnixSource~/MUTE/Makefile	2007-01-07 00:26:40.000000000 +0100
+++ MUTE_fileSharing-0.5_UnixSource/MUTE/Makefile	2007-01-07 00:27:48.000000000 +0100
@@ -26,7 +26,7 @@
 
 
 all:
-	cd crypto; make; cd ../layers/messageRouting; make; cd ../pointToPoint; make; cd ../fileTransfer; make; cd ../../userInterface/fileTransferWebInterface; make; cd ../textBased; make; cd ../../otherApps/fileSharing; make; cd userInterface/textBased; make; cd ../../../..
+	cd crypto; cd ../layers/messageRouting; make; cd ../pointToPoint; make; cd ../fileTransfer; make; cd ../../userInterface/fileTransferWebInterface; make; cd ../textBased; make; cd ../../otherApps/fileSharing; make; cd userInterface/textBased; make; cd ../../../..
 
 clean:
 	cd crypto; make clean; cd ../layers/messageRouting; make clean; cd ../pointToPoint; make clean; cd ../fileTransfer; make clean; cd ../../userInterface/fileTransferWebInterface; make clean; cd ../textBased; make clean; cd ../../otherApps/fileSharing; make clean; cd userInterface/textBased; make clean; cd ../../../..
diff -ru MUTE_fileSharing-0.5_UnixSource~/MUTE/Makefile.common MUTE_fileSharing-0.5_UnixSource/MUTE/Makefile.common
--- MUTE_fileSharing-0.5_UnixSource~/MUTE/Makefile.common	2007-01-07 00:26:40.000000000 +0100
+++ MUTE_fileSharing-0.5_UnixSource/MUTE/Makefile.common	2007-01-07 00:27:33.000000000 +0100
@@ -37,8 +37,9 @@
 
 
 
+CXXFLAGS ?=
+COMPILE_FLAGS = -Wall ${CXXFLAGS} ${DEBUG_FLAG} ${PLATFORM_COMPILE_FLAGS} ${PROFILE_FLAG}
 
-COMPILE_FLAGS = -Wall ${DEBUG_FLAG} ${PLATFORM_COMPILE_FLAGS} ${PROFILE_FLAG}
 
 
 COMPILE = ${GXX} ${COMPILE_FLAGS} -c
diff -ru MUTE_fileSharing-0.5_UnixSource~/MUTE/Makefile.GnuLinuxX86 MUTE_fileSharing-0.5_UnixSource/MUTE/Makefile.GnuLinuxX86
--- MUTE_fileSharing-0.5_UnixSource~/MUTE/Makefile.GnuLinuxX86	2007-01-07 00:26:40.000000000 +0100
+++ MUTE_fileSharing-0.5_UnixSource/MUTE/Makefile.GnuLinuxX86	2007-01-07 00:27:33.000000000 +0100
@@ -22,7 +22,7 @@
 
 
 # All platforms but OSX support g++ and need no linker hacks
-GXX=g++ 
+GXX?=g++ 
 LINK_FLAGS = 
 
 
diff -ru MUTE_fileSharing-0.5_UnixSource~/MUTE/otherApps/fileSharing/userInterface/wxWindows/Makefile.all MUTE_fileSharing-0.5_UnixSource/MUTE/otherApps/fileSharing/userInterface/wxWindows/Makefile.all
--- MUTE_fileSharing-0.5_UnixSource~/MUTE/otherApps/fileSharing/userInterface/wxWindows/Makefile.all	2007-01-07 00:26:40.000000000 +0100
+++ MUTE_fileSharing-0.5_UnixSource/MUTE/otherApps/fileSharing/userInterface/wxWindows/Makefile.all	2007-01-07 00:27:33.000000000 +0100
@@ -47,6 +47,8 @@
 
 ROOT_PATH = ../../../../..
 
+WX_CONFIG?="/usr/bin/wx-config"
+WX_CONFIG_LOCATION=${WX_CONFIG}
 WX_LIBS = $(shell ${WX_CONFIG_LOCATION} --libs)
 WX_COMPILE_FLAGS = $(shell ${WX_CONFIG_LOCATION} --cxxflags)
 
