--- setup.py.orig	2007-03-26 10:34:43.000000000 -0400
+++ setup.py	2008-11-22 23:25:21.000000000 -0500
@@ -7,7 +7,7 @@
 
 # default settings
 library_dirs_list= []
-libraries_list = ['lapack', 'blas', 'g2c']
+libraries_list = ['lapack', 'blas', 'gfortran']
 superlu_defs = [('USE_VENDOR_BLAS',1)]
 f77_defs = []
 linky=[]
@@ -32,8 +32,8 @@
     numerix_macro = []
     numerix_include_dirs = []
 
-umfpack_defs = [('DINT', 1), ('NBLAS', 1)] # most basic configuration, no BLAS
-umfpack_libraries = []
+umfpack_defs = [('DINT', 1)] # most basic configuration, no BLAS
+umfpack_libraries = ['blas', 'atlas', 'gfortran', 'm']
 umfpack_include_dirs = ['amd', 'umfpack']
 umfpack_library_dirs = []
 
