--- ./source-trees/gridway/src/gwd/gwd.c.~1.9.2.2.~	2008-04-01 11:59:01.000000000 -0600
+++ ./source-trees/gridway/src/gwd/gwd.c	2009-08-08 22:14:25.490391881 -0600
@@ -138,7 +138,7 @@
     /* Close stds, we no longer need them                          */
     /* ----------------------------------------------------------- */
 
-    fd = open("/dev/null", O_RDWR|O_CREAT);
+    fd = open("/dev/null", O_RDWR|O_CREAT, 0640);
     	
     dup2(fd,0);
     dup2(fd,1);    
