=== modified file 'configure.in'
--- configure.in	2009-03-15 09:13:28 +0000
+++ configure.in	2009-03-15 09:13:53 +0000
@@ -198,8 +198,8 @@
 	AC_SUBST(XOSD_LIBS)
 fi
 
-CFLAGS="-Wall -Wextra -Werror -g0 -fPIC -std=gnu99"
-AC_SUBST(CFLAGS)
+DEFAULT_CFLAGS="-Wall -Wextra -Werror -g0 -fPIC -std=gnu99"
+AC_SUBST(DEFAULT_CFLAGS)
 
 AC_OUTPUT([share/Makefile share/sounds/Makefile share/pixmaps/Makefile share/en/Makefile share/ru/Makefile share/uk/Makefile share/be/Makefile share/fr/Makefile share/ro/Makefile share/kz/Makefile share/de/Makefile  etc/Makefile lib/Makefile lib/ai/Makefile lib/config/Makefile lib/misc/Makefile lib/sound/Makefile lib/x/Makefile src/Makefile po/Makefile.in Makefile xnconfig.pc])
 

=== modified file 'lib/ai/Makefile.am'
--- lib/ai/Makefile.am	2009-03-15 09:13:28 +0000
+++ lib/ai/Makefile.am	2009-03-15 09:14:07 +0000
@@ -6,6 +6,6 @@
 	detection.c		\
 	detection.h
 
-libxnai_la_CFLAGS = -I@top_srcdir@/lib/config -I@top_srcdir@/lib/misc -I@top_srcdir@/lib/x @X11_CFLAGS@ @ASPELL_CFLAGS@
+libxnai_la_CFLAGS = -I@top_srcdir@/lib/config -I@top_srcdir@/lib/misc -I@top_srcdir@/lib/x @X11_CFLAGS@ @ASPELL_CFLAGS@ @DEFAULT_CFLAGS@
 libxnai_la_LDFLAGS = -static @X11_LIBS@ @ASPELL_LIBS@
 libxnai_la_LIBADD = ../misc/libxnmisc.la

=== modified file 'lib/config/Makefile.am'
--- lib/config/Makefile.am	2009-03-15 09:13:28 +0000
+++ lib/config/Makefile.am	2009-03-15 09:14:28 +0000
@@ -14,7 +14,7 @@
 	xnconfig.c		\
 	xnconfig_memory.c
 
-libxnconfig_la_CFLAGS = -I@top_srcdir@/lib/misc -I@top_srcdir@/lib/x
+libxnconfig_la_CFLAGS = -I@top_srcdir@/lib/misc -I@top_srcdir@/lib/x @DEFAULT_CFLAGS@
 libxnconfig_la_LDFLAGS = -version-info 9:3:0
 libxnconfig_la_LIBADD = ../misc/libxnmisc.la
 

=== modified file 'lib/misc/Makefile.am'
--- lib/misc/Makefile.am	2009-03-15 09:13:28 +0000
+++ lib/misc/Makefile.am	2009-03-15 09:15:07 +0000
@@ -20,5 +20,5 @@
 	utils.c			\
 	regexp.c
 
-libxnmisc_la_CFLAGS = @PCRE_CFLAGS@
+libxnmisc_la_CFLAGS = @PCRE_CFLAGS@ @DEFAULT_CFLAGS@
 libxnmisc_la_LDFLAGS = -static @PCRE_LIBS@

=== modified file 'lib/sound/Makefile.am'
--- lib/sound/Makefile.am	2009-03-15 09:13:28 +0000
+++ lib/sound/Makefile.am	2009-03-15 09:15:23 +0000
@@ -4,6 +4,6 @@
 	sound.c			\
 	sound.h
 
-libxnsound_la_CFLAGS = -I@top_srcdir@/lib/config -I@top_srcdir@/lib/misc @X11_CFLAGS@ @FREEALUT_CFLAGS@ @GSTREAMER_CFLAGS@ 
+libxnsound_la_CFLAGS = -I@top_srcdir@/lib/config -I@top_srcdir@/lib/misc @X11_CFLAGS@ @FREEALUT_CFLAGS@ @GSTREAMER_CFLAGS@ @DEFAULT_CFLAGS@
 libxnsound_la_LDFLAGS = -static @X11_LIBS@ @FREEALUT_LIBS@ @GSTREAMER_LIBS@  @APLAY_LIBS@
 

=== modified file 'lib/x/Makefile.am'
--- lib/x/Makefile.am	2009-03-15 09:13:28 +0000
+++ lib/x/Makefile.am	2009-03-15 09:15:43 +0000
@@ -27,5 +27,5 @@
 	xOSD.c			\
 	xOSD.h
 
-libxnx_la_CFLAGS = -I@top_srcdir@/lib/config -I@top_srcdir@/lib/misc  -I@top_srcdir@/lib/sound -I@top_srcdir@/lib/ai @X11_CFLAGS@ @XPM_CFLAGS@ @IMLIB2_CFLAGS@
+libxnx_la_CFLAGS = -I@top_srcdir@/lib/config -I@top_srcdir@/lib/misc  -I@top_srcdir@/lib/sound -I@top_srcdir@/lib/ai @X11_CFLAGS@ @XPM_CFLAGS@ @IMLIB2_CFLAGS@ @DEFAULT_CFLAGS@
 libxnx_la_LDFLAGS = -static @X11_LIBS@ @XPM_LIBS@ @IMLIB2_LIBS@ @XOSD_LIBS@

=== modified file 'src/Makefile.am'
--- src/Makefile.am	2009-03-15 09:13:28 +0000
+++ src/Makefile.am	2009-03-15 09:15:58 +0000
@@ -7,7 +7,7 @@
 	newlang_creation.c	\
 	newlang_creation.h
 
-xneur_CFLAGS = -I@top_srcdir@/lib/ai -I@top_srcdir@/lib/config -I@top_srcdir@/lib/x -I@top_srcdir@/lib/misc -I@top_srcdir@/lib/sound  @X11_CFLAGS@ @XPM_CFLAGS@
+xneur_CFLAGS = -I@top_srcdir@/lib/ai -I@top_srcdir@/lib/config -I@top_srcdir@/lib/x -I@top_srcdir@/lib/misc -I@top_srcdir@/lib/sound  @X11_CFLAGS@ @XPM_CFLAGS@ @DEFAULT_CFLAGS@
 
 xneur_LDFLAGS = -L$(top_srcdir)/lib/ai -L$(top_srcdir)/lib/config -L$(top_srcdir)/lib/misc -L$(top_srcdir)/lib/x -L$(top_srcdir)/lib/sound  @X11_LIBS@ @XPM_LIBS@
 

