=== modified file 'configure.in'
--- configure.in	2010-12-06 11:25:12 +0000
+++ configure.in	2010-12-06 11:26:02 +0000
@@ -205,7 +205,8 @@
 	AC_DEFINE(WITH_PLUGINS, 1, [Define if you want plugins support])
 fi
 
-DEFAULT_CFLAGS="-Wall -Wextra -Werror -g0 -std=gnu99 -fPIC -fopenmp"
+AC_OPENMP([C])
+DEFAULT_CFLAGS="-Wall -Wextra -Werror -g0 -std=gnu99 -fPIC $OPENMP_CFLAGS"
 AC_SUBST(DEFAULT_CFLAGS)
 
 AC_OUTPUT([

