--- configure.ac	2009-09-07 19:20:06.000000000 +0200
+++ configure.ac.new	2009-09-07 19:24:40.000000000 +0200
@@ -5,23 +5,26 @@
 AC_INIT([libwiimote], [0.5.0], [http://sourceforge.net/projects/libwiimote/])
 AC_CONFIG_HEADER([config.h])
 
-
 VERSION=0.4.0
 PACKAGE=libcwiimote
 
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
 
 # Checks for programs.
+AC_PROG_CXX
 AC_PROG_CC
+AC_PROG_CPP
 AC_PROG_INSTALL
-AC_PROG_LIBTOOL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
 AC_SUBST(LIBTOOL_DEPS)
 AM_PROG_LIBTOOL
 AC_CONFIG_MACRO_DIR([m4])
 # Checks for libraries.
 AC_CHECK_LIB([bluetooth], [hci_read_remote_name],
-             [], [AC_MSG_ERROR([We require BlueZ])])
-
+             [], [AC_MSG_ERROR([BlueZ library required, available at http://www.bluez.org/])])
+AC_CHECK_LIB([m], [asin],
+             [], [AC_MSG_ERROR([libm required])])
 # Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h stdint.h stdlib.h string.h sys/ioctl.h sys/socket.h sys/time.h unistd.h])
@@ -45,7 +48,7 @@
 AC_CONFIG_FILES([Makefile
                  src/Makefile
                  test/Makefile
-		 libwiimote.pc])
+					  libwiimote.pc])
 AC_OUTPUT
 echo
 echo $PACKAGE v$VERSION
