respect flags/compiler

--- Makefile
+++ Makefile
@@ -9,13 +9,13 @@
 # CCPCRE=-I/usr/local/include -DUSEPCRE
 # LDPCRE=-L/usr/local/lib -lpcre
 
-CXX=g++
-CXXFLAGS=-Wall -O3 -g -pthread  # can override
+CXX ?= g++
+CXXFLAGS += -pthread
 RE2_CXXFLAGS=-Wno-sign-compare -c -I. $(CCPCRE)  # required
-LDFLAGS=-pthread
-AR=ar
+LDFLAGS += -pthread
+AR ?= ar
 ARFLAGS=rsc
-NM=nm
+NM ?= nm
 NMFLAGS=-p
 
 # Variables mandated by GNU, the arbiter of all good taste on the internet.
