diff -u daemonlogger-1.0/daemonlogger.c daemonlogger-1.0.new/daemonlogger.c
--- daemonlogger-1.0/daemonlogger.c	2007-10-16 20:54:45.000000000 +0300
+++ daemonlogger-1.0.new/daemonlogger.c	2007-11-09 11:46:35.000000000 +0200
@@ -108,7 +108,7 @@
 #define ERROR       1
 #define STDBUF      1024
 #define GIGABYTE    1 << 30
-#define VERSION     "0.91"
+#define VERSION     "1.0"
 
 typedef enum {
     MINUTES=1,
