--- pidgin-blinklight-0.11.1/src/pidgin-blinklight.c.old	2011-10-11 11:29:13.277577702 +0200
+++ pidgin-blinklight-0.11.1/src/pidgin-blinklight.c	2011-10-11 12:31:22.338699746 +0200
@@ -99,7 +99,7 @@
 
 	// Make /proc file writable
 	if (!fork())
-		execl("/usr/lib/pidgin-blinklight/blinklight-fixperm","blinklight-fixperm",NULL);
+		execl("/usr/libexec/blinklight-fixperm","blinklight-fixperm",NULL);
 	
 
 	char *chosen_file = blinklight_init();
