diff -r -u xwpe-1.5.30a-orig/Makefile.in xwpe-1.5.30a/Makefile.in
--- xwpe-1.5.30a-orig/Makefile.in	2005-07-07 03:53:09.000000000 +0200
+++ xwpe-1.5.30a/Makefile.in	2009-12-28 13:16:49.000000000 +0100
@@ -9,10 +9,10 @@
 #Destination Directory
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-DESTDIR=@bindir@
-LIBDIR=@libdir@
-XWPELIBDIR=	$(LIBDIR)/xwpe
-MANDIR=@mandir@
+BINDIR=$(DESTDIR)/@bindir@
+LIBDIR=$(DESTDIR)/@libdir@
+XWPELIBDIR=$(LIBDIR)/xwpe
+MANDIR=$(DESTDIR)/@mandir@
 
 LIBS=@LIBS@
 XLIBS=@X_LIBS@ @X_PRE_LIBS@ @X_EXTRA_LIBS@
@@ -115,14 +115,14 @@
 # Bug Note: Currently xwpe and xwe are installed even if X windows versions
 # are not compiled.  (They will function as wpe and we respectively.)
 install_fst:	xwpe
-	if test ! -d $(DESTDIR); then mkdir -p $(DESTDIR); fi
+	if test ! -d $(BINDIR); then mkdir -p $(BINDIR); fi
 	if test ! -d $(XWPELIBDIR); then mkdir -p $(XWPELIBDIR); fi
-	rm -f $(DESTDIR)/wpe $(DESTDIR)/xwpe $(DESTDIR)/we $(DESTDIR)/xwe
-	$(INSTALL) xwpe $(DESTDIR)/we
-	chmod ugo+x $(DESTDIR)/we
-	(cd $(DESTDIR) && $(LN_S) we xwe)
-	(cd $(DESTDIR) && $(LN_S) we xwpe)
-	(cd $(DESTDIR) && $(LN_S) we wpe)
+	rm -f $(BINDIR)/wpe $(BINDIR)/xwpe $(BINDIR)/we $(BINDIR)/xwe
+	$(INSTALL) xwpe $(BINDIR)/we
+	chmod ugo+x $(BINDIR)/we
+	(cd $(BINDIR) && $(LN_S) we xwe)
+	(cd $(BINDIR) && $(LN_S) we xwpe)
+	(cd $(BINDIR) && $(LN_S) we wpe)
 	rm -f $(XWPELIBDIR)/help.xwpe
 	if test ! -f $(XWPELIBDIR)/syntax_def; then  \
 	  $(INSTALL) syntax_def $(XWPELIBDIR)/syntax_def; \
