diff -u -r evemu-1.0.10-orig/configure.ac evemu-1.0.10/configure.ac
--- evemu-1.0.10-orig/configure.ac	2013-01-03 14:39:47.816749652 -0600
+++ evemu-1.0.10/configure.ac	2013-01-03 15:40:16.566917042 -0600
@@ -35,9 +35,17 @@
 	AC_MSG_WARN([xmlto or asciidoc not found - cannot create man pages without it])
 fi
 
-AC_SUBST(AM_CFLAGS,
-         "-Wall -Wextra -pedantic -Werror -std=gnu99 -D_GNU_SOURCE=1")
+AC_ARG_ENABLE([werror],
+	[AC_HELP_STRING([--disable-werror],
+		[build without gcc -Werror])],
+    [enable_werror=$enableval],
+    [enable_werror="yes"])
 
+AS_IF([test "x$enable_werror" = "xyes"],
+      [CFLAGS="-Werror $CFLAGS"])
+
+ 
+CFLAGS="-Wall -Wextra -pedantic -std=gnu99 -D_GNU_SOURCE=1 $CFLAGS"
 
 AC_CONFIG_FILES([Makefile
                  src/Makefile
