From: hasufell <julian.ospald@googlemail.com>
Date: Mon Apr 23 21:34:29 UTC 2012

use pkgconfig for lua detection

--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -144,10 +144,8 @@
 
 find_package(Jthread REQUIRED)
 find_package(Sqlite3 REQUIRED)
-
-# TODO: Create proper find script for Lua
-set(LUA_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/lua/src")
-set(LUA_LIBRARY "lua")
+find_package(Lua51 REQUIRED)
+
 
 configure_file(
 	"${PROJECT_SOURCE_DIR}/cmake_config.h.in"
@@ -476,9 +474,9 @@
 	add_subdirectory(sqlite)
 endif (SQLITE3_FOUND)
 
-if (LUA_FOUND)
-else (LUA_FOUND)
+if (LUA_LIBRARY AND LUA_INCLUDE_DIR)
+else (LUA_LIBRARY AND LUA_INCLUDE_DIR)
 	add_subdirectory(lua)
-endif (LUA_FOUND)
+endif (LUA_LIBRARY AND LUA_INCLUDE_DIR)
 
 #end
