--- golly-2.1-src.old/makefile-gtk	2009-09-07 02:33:06.000000000 +0200
+++ golly-2.1-src.new/makefile-gtk	2010-06-08 17:32:21.000000000 +0200
@@ -29,7 +29,7 @@
 # than 5.10 (this is necessary because boot_DynaLoader is in DynaLoader.a);
 # if using 5.10 or later we can dynamically load the Perl library because
 # it contains the boot_DynaLoader code
-PERL_LINK = `perl -MExtUtils::Embed -e '$$]<5.010 && ldopts'`
+PERL_LINK = `perl -MExtUtils::Embed -e 'ldopts'`
 
 # for Python script support
 PYTHON_INCLUDE = -I`python -c "import distutils.sysconfig; print distutils.sysconfig.get_python_inc()"`
