diff -Naur libapp-9999.orig//Makefile libapp-9999/Makefile
--- libapp-9999.orig//Makefile	2011-05-31 15:16:43.090960586 +0200
+++ libapp-9999/Makefile	2011-05-31 15:25:12.300228627 +0200
@@ -2,8 +2,7 @@
 VER=0.2.0
 OBJS=app.o list.o
 HEADERS=app.h list.h base.h
-CFLAGS=-g -fPIC -I..
-CFLAGS_T=-g -fPIC -I../..
+MY_CFLAGS=-g -fPIC -I..
 SONAME=libapp.so
 PREFIX=/usr/local
 INCDIR=$(PREFIX)/include/libapp
@@ -19,10 +18,10 @@
 list.o: base.h list.h list.c
 
 $(SONAME).$(VER):	$(OBJS)
-	gcc $(CFLAGS) -Wl,-soname=$(SONAME) -shared  -o $@ $(OBJS)
+	$(CC) $(MY_CFLAGS) $(CFLAGS) $(LDFLAGS) -Wl,-soname=$(SONAME) -shared  -o $@ $(OBJS)
 
 .c.o:
-	gcc $(CFLAGS) -c $< -o $@
+	$(CC) $(MY_CFLAGS) $(CFLAGS) -c $< -o $@
 
 clean:
 	rm -f $(SONAME).* *.o test/*.o $(TESTS)
@@ -40,7 +39,7 @@
 test: $(SONAME) $(TESTS)
 
 apptest: test/apptest.o
-	gcc $(CFLAGS) -o $@ $< -lapp
+	$(CC) $(MY_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $< -lapp
 
 listtest: test/listtest.o
-	gcc $(CFLAGS) -o $@ $< -lapp
+	$(CC) $(MY_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $< -lapp
