 wxOil/basebar2.cpp |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/wxOil/basebar2.cpp b/wxOil/basebar2.cpp
index 200c855..52b7376 100644
--- a/wxOil/basebar2.cpp
+++ b/wxOil/basebar2.cpp
@@ -4565,7 +4565,6 @@ void BaseBar::StartDrag(wxPoint point, DlgDragType TypeOfDrag)
 // Message map - maintained by Class Wizard.
 
 BEGIN_EVENT_TABLE( BaseBar, wxWindow )
-#pragma message( __LOCMSG__ "Removed BaseBar message map" )
 /*	ON_WM_NCCALCSIZE()
 	ON_WM_NCPAINT()
 	ON_WM_NCACTIVATE()
@@ -4618,7 +4617,6 @@ END_EVENT_TABLE()
 
 static void PatB( wxDC &DC, INT32 x, INT32 y, INT32 dx, INT32 dy, wxColour &rgb )
 {
-#pragma message( __LOCMSG__ "PatB - do nothing" )
 	TRACE( _T("Warning - PatB called") );
 /*	RECT rect;
 
