# fixes directory for KEYS file
--- src/gui_keys.c.orig	2009-01-12 07:19:32.000000000 +0100
+++ src/gui_keys.c	2009-01-12 07:19:47.000000000 +0100
@@ -88,7 +88,7 @@
 #ifdef G_OS_WIN32
 	helpfile = g_strdup_printf ("%s", filename);
 #else
-	helpfile = g_strdup_printf ("%s%s%s", XLOG_DATADIR, G_DIR_SEPARATOR_S, 
+	helpfile = g_strdup_printf ("%s%s%s", XLOG_DOCDIR, G_DIR_SEPARATOR_S, 
 		filename);
 #endif
 	buf = g_new0 (gchar, 100);
