--- Makefile.fip	2007-02-16 23:02:07.000000000 +0300
+++ Makefile.fip	2007-02-16 23:05:42.000000000 +0300
@@ -4,9 +4,9 @@
 include fipMakefile.srcs
 
 # General configuration variables:
-CC = gcc
-CXX = g++
-AR = ar
+#CC = gcc
+#CXX = g++
+#AR = ar
 
 INCDIR = /usr/include
 INSTALLDIR = /usr/lib
@@ -19,8 +19,8 @@
 
 MODULES = $(SRCS:.c=.o)
 MODULES := $(MODULES:.cpp=.o)
-CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
-CXXFLAGS = $(COMPILERFLAGS) -Wno-ctor-dtor-privacy $(INCLUDE)
+CFLAGS := $(CFLAGS) $(COMPILERFLAGS) -fPIC -DPIC $(INCLUDE)
+CXXFLAGS := $(CXXFLAGS) $(COMPILERFLAGS) -fPIC -DPIC -Wno-ctor-dtor-privacy $(INCLUDE)
 
 TARGET  = freeimageplus
 STATICLIB = lib$(TARGET).a
@@ -56,7 +56,7 @@
 	$(AR) r $@ $(MODULES)
 
 $(SHAREDLIB): $(MODULES)
-	$(CC) -s -shared -Wl,-soname,$(VERLIBNAME) -o $@ $(MODULES) $(LIBRARIES)
+	$(CC) $(LDFLAGS) -fPIC -s -shared -Wl,-soname,$(VERLIBNAME) -o $@ $(MODULES) $(LIBRARIES)
 
 install:
 	install -m 644 -o root -g root $(HEADER) $(INCDIR)
@@ -65,7 +65,6 @@
 	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)
 	ln -sf $(SHAREDLIB) $(INSTALLDIR)/$(VERLIBNAME)
 	ln -sf $(VERLIBNAME) $(INSTALLDIR)/$(LIBNAME)
-	ldconfig
 
 clean:
 	rm -f core Dist/*.* u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
--- Makefile.gnu	2007-02-16 23:08:06.000000000 +0300
+++ Makefile.gnu	2006-12-20 15:22:08.000000000 +0300
@@ -4,9 +4,9 @@
 include Makefile.srcs
 
 # General configuration variables:
-CC = gcc
-CXX = g++
-AR = ar
+#CC = gcc
+#CXX = g++
+#AR = ar
 
 INCDIR = /usr/include
 INSTALLDIR = /usr/lib
@@ -14,13 +14,13 @@
 # Converts cr/lf to just lf
 DOS2UNIX = dos2unix
 
-COMPILERFLAGS = -O3 -fPIC
+COMPILERFLAGS = -fPIC -DPIC
 LIBRARIES = -lstdc++
 
 MODULES = $(SRCS:.c=.o)
 MODULES := $(MODULES:.cpp=.o)
-CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
-CXXFLAGS = $(COMPILERFLAGS)  -Wno-ctor-dtor-privacy $(INCLUDE)
+CFLAGS := $(CFLAGS) $(COMPILERFLAGS) $(INCLUDE)
+CXXFLAGS := $(CXXFLAGS) $(COMPILERFLAGS) -Wno-ctor-dtor-privacy $(INCLUDE)
 
 TARGET  = freeimage
 STATICLIB = lib$(TARGET).a
