--- Makefile_old	2007-06-03 16:18:12.000000000 +0200
+++ Makefile	2007-06-03 22:59:06.000000000 +0200
@@ -42,7 +42,7 @@ modules:
 install: modules_install
 
 modules_install:
-	$(MAKE) -C $(KDIR) M=$(PWD) INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) modules_install
+	$(MAKE) -C $(KDIR) M=$(PWD) INSTALL_MOD_PATH=${DESTDIR} INSTALL_MOD_DIR=$(INSTALL_MOD_DIR) modules_install
 
 clean:
 	$(MAKE) -C $(KDIR) M=$(PWD) clean
