diff -ru miniAudicle-0.1.3.6~/makefile.alsa miniAudicle-0.1.3.6/makefile.alsa
--- miniAudicle-0.1.3.6~/makefile.alsa	2006-09-19 10:09:03.000000000 +0200
+++ miniAudicle-0.1.3.6/makefile.alsa	2006-09-19 10:13:32.000000000 +0200
@@ -5,14 +5,17 @@
 
 WXW_DIR=wxw
 BIN_DIR=wxw
+WX_CONFIG?=wx-config
 
-CC=gcc
+CC?=gcc
+CXX?=g++
 LEX=flex
 YACC=bison
 INCLUDES=-I$(CHUCK_SRC_DIR) -I$(WXW_DIR) -I$(PWD)
+CFLAGS?=-O3
 FLAGS=-D__LINUX_ALSA__ -D__LINUX__ -D__MA_WXWIDGETS__ \
-	  `wx-config --cppflags` -O3 -c -DwxABI_VERSION=20600
-LIBS=`wx-config --libs stc,adv,core,base` -lasound -lpthread \
+	  `$(WX_CONFIG) --cppflags` -c -DwxABI_VERSION=20600 $(CFLAGS)
+LIBS=`$(WX_CONFIG) --libs stc,adv,core,base` -lasound -lpthread \
 	 -lstdc++ -ldl -lm
 SF_OBJ=$(CHUCK_SRC_DIR)/util_sndfile.o
 
@@ -69,44 +72,44 @@
 	 $(WXW_DIR)/icons/replace.xpm $(WXW_DIR)/icons/miniAudicle.xpm
 
 miniAudicle: $(OBJS) $(CHUCK_OBJS)
-	g++ -o $(BIN_DIR)/miniAudicle $(OBJS) $(CHUCK_OBJS) $(LIBS)
+	$(CXX) -o $(BIN_DIR)/miniAudicle $(OBJS) $(CHUCK_OBJS) $(LIBS)
 
 # build the core chuck code
 $(CHUCK_OBJS):
 	make -C $(CHUCK_SRC_DIR) linux-alsa
 
 miniAudicle_shell.o: miniAudicle_shell.cpp miniAudicle_shell.h
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle_shell.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle_shell.cpp
 
 miniAudicle_import.o: miniAudicle_import.h miniAudicle_import.cpp
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle_import.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle_import.cpp
 
 miniAudicle.o: miniAudicle.cpp miniAudicle.h
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle.cpp
 
 $(WXW_DIR)/mAParentFrame.o: $(WXW_DIR)/mAParentFrame.cpp $(WXW_DIR)/mAParentFrame.h $(WXW_DIR)/mAView.h $(WXW_DIR)/mADocument.h $(WXW_DIR)/mAVMMonitor.h miniAudicle.h $(WXW_DIR)/mAMenuBar.h $(XPMS)
-	$(CC) -o $(WXW_DIR)/mAParentFrame.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAParentFrame.cpp
+	$(CXX) -o $(WXW_DIR)/mAParentFrame.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAParentFrame.cpp
 
 $(WXW_DIR)/mADocument.o: $(WXW_DIR)/mADocument.cpp $(WXW_DIR)/mADocument.h $(WXW_DIR)/mAView.h
-	$(CC) -o $(WXW_DIR)/mADocument.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mADocument.cpp
+	$(CXX) -o $(WXW_DIR)/mADocument.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mADocument.cpp
 
 $(WXW_DIR)/mAView.o: $(WXW_DIR)/mAView.cpp $(WXW_DIR)/mAView.h
-	$(CC) -o $(WXW_DIR)/mAView.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAView.cpp
+	$(CXX) -o $(WXW_DIR)/mAView.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAView.cpp
 
 $(WXW_DIR)/mAVMMonitor.o: $(WXW_DIR)/mAVMMonitor.cpp $(WXW_DIR)/mAVMMonitor.h $(WXW_DIR)/mAParentFrame.h miniAudicle.h
-	$(CC) -o $(WXW_DIR)/mAVMMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAVMMonitor.cpp
+	$(CXX) -o $(WXW_DIR)/mAVMMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAVMMonitor.cpp
 
 $(WXW_DIR)/mAConsoleMonitor.o: $(WXW_DIR)/mAConsoleMonitor.cpp $(WXW_DIR)/mAConsoleMonitor.h $(WXW_DIR)/mAParentFrame.h
-	$(CC) -o $(WXW_DIR)/mAConsoleMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConsoleMonitor.cpp
+	$(CXX) -o $(WXW_DIR)/mAConsoleMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConsoleMonitor.cpp
 
 $(WXW_DIR)/mAMenuBar.o: $(WXW_DIR)/mAMenuBar.cpp $(WXW_DIR)/mAMenuBar.h $(WXW_DIR)/mAEvents.h
-	$(CC) -o $(WXW_DIR)/mAMenuBar.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAMenuBar.cpp
+	$(CXX) -o $(WXW_DIR)/mAMenuBar.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAMenuBar.cpp
 
 $(WXW_DIR)/mAConfig.o: $(WXW_DIR)/mAConfig.cpp $(WXW_DIR)/mAConfig.h
-	$(CC) -o $(WXW_DIR)/mAConfig.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConfig.cpp
+	$(CXX) -o $(WXW_DIR)/mAConfig.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConfig.cpp
 
 $(WXW_DIR)/mAPreferencesWindow.o: $(WXW_DIR)/mAPreferencesWindow.cpp $(WXW_DIR)/mAPreferencesWindow.h
-	$(CC) -o $(WXW_DIR)/mAPreferencesWindow.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAPreferencesWindow.cpp
+	$(CXX) -o $(WXW_DIR)/mAPreferencesWindow.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAPreferencesWindow.cpp
 
 clean: 
 	rm -rf *.o $(WXW_DIR)/*.o $(BIN_DIR)/miniAudicle
diff -ru miniAudicle-0.1.3.6~/makefile.jack miniAudicle-0.1.3.6/makefile.jack
--- miniAudicle-0.1.3.6~/makefile.jack	2006-09-19 10:09:03.000000000 +0200
+++ miniAudicle-0.1.3.6/makefile.jack	2006-09-19 10:15:47.000000000 +0200
@@ -5,14 +5,17 @@
 
 WXW_DIR=wxw
 BIN_DIR=wxw
+WX_CONFIG?=wx-config
 
-CC=gcc
+CC?=gcc
+CXX?=g++
 LEX=flex
 YACC=bison
 INCLUDES=-I$(CHUCK_SRC_DIR) -I$(WXW_DIR) -I$(PWD)
+CFLAGS?=-O3
 FLAGS=-D__LINUX_JACK__ -D__LINUX__ -D__MA_WXWIDGETS__ \
-	  `wx-config --cppflags` -O3 -c -DwxABI_VERSION=20600
-LIBS=`wx-config --libs stc,adv,core,base` -lasound -ljack \
+	  `$(WX_CONFIG) --cppflags` -O3 -c -DwxABI_VERSION=20600 $(CFLAGS)
+LIBS=`$(WX_CONFIG) --libs stc,adv,core,base` -lasound -ljack \
 	 -lstdc++ -ldl -lm
 SF_OBJ=$(CHUCK_SRC_DIR)/util_sndfile.o
 
@@ -73,44 +76,44 @@
 	 $(WXW_DIR)/icons/replace.xpm $(WXW_DIR)/icons/miniAudicle.xpm
 
 miniAudicle: $(OBJS) $(CHUCK_OBJS)
-	g++ -o $(BIN_DIR)/miniAudicle $(OBJS) $(CHUCK_OBJS) $(LIBS)
+	$(CXX) -o $(BIN_DIR)/miniAudicle $(OBJS) $(CHUCK_OBJS) $(LIBS)
 
 # build the core chuck code
 $(CHUCK_OBJS):
 	make -C $(CHUCK_SRC_DIR) linux-jack
 
 miniAudicle_shell.o: miniAudicle_shell.cpp miniAudicle_shell.h
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle_shell.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle_shell.cpp
 
 miniAudicle_import.o: miniAudicle_import.h miniAudicle_import.cpp
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle_import.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle_import.cpp
 
 miniAudicle.o: miniAudicle.cpp miniAudicle.h
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle.cpp
 
 $(WXW_DIR)/mAParentFrame.o: $(WXW_DIR)/mAParentFrame.cpp $(WXW_DIR)/mAParentFrame.h $(WXW_DIR)/mAView.h $(WXW_DIR)/mADocument.h $(WXW_DIR)/mAVMMonitor.h miniAudicle.h $(WXW_DIR)/mAMenuBar.h $(XPMS)
-	$(CC) -o $(WXW_DIR)/mAParentFrame.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAParentFrame.cpp
+	$(CXX) -o $(WXW_DIR)/mAParentFrame.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAParentFrame.cpp
 
 $(WXW_DIR)/mADocument.o: $(WXW_DIR)/mADocument.cpp $(WXW_DIR)/mADocument.h $(WXW_DIR)/mAView.h
-	$(CC) -o $(WXW_DIR)/mADocument.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mADocument.cpp
+	$(CXX) -o $(WXW_DIR)/mADocument.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mADocument.cpp
 
 $(WXW_DIR)/mAView.o: $(WXW_DIR)/mAView.cpp $(WXW_DIR)/mAView.h
-	$(CC) -o $(WXW_DIR)/mAView.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAView.cpp
+	$(CXX) -o $(WXW_DIR)/mAView.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAView.cpp
 
 $(WXW_DIR)/mAVMMonitor.o: $(WXW_DIR)/mAVMMonitor.cpp $(WXW_DIR)/mAVMMonitor.h $(WXW_DIR)/mAParentFrame.h miniAudicle.h
-	$(CC) -o $(WXW_DIR)/mAVMMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAVMMonitor.cpp
+	$(CXX) -o $(WXW_DIR)/mAVMMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAVMMonitor.cpp
 
 $(WXW_DIR)/mAConsoleMonitor.o: $(WXW_DIR)/mAConsoleMonitor.cpp $(WXW_DIR)/mAConsoleMonitor.h $(WXW_DIR)/mAParentFrame.h
-	$(CC) -o $(WXW_DIR)/mAConsoleMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConsoleMonitor.cpp
+	$(CXX) -o $(WXW_DIR)/mAConsoleMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConsoleMonitor.cpp
 
 $(WXW_DIR)/mAMenuBar.o: $(WXW_DIR)/mAMenuBar.cpp $(WXW_DIR)/mAMenuBar.h $(WXW_DIR)/mAEvents.h
-	$(CC) -o $(WXW_DIR)/mAMenuBar.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAMenuBar.cpp
+	$(CXX) -o $(WXW_DIR)/mAMenuBar.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAMenuBar.cpp
 
 $(WXW_DIR)/mAConfig.o: $(WXW_DIR)/mAConfig.cpp $(WXW_DIR)/mAConfig.h
-	$(CC) -o $(WXW_DIR)/mAConfig.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConfig.cpp
+	$(CXX) -o $(WXW_DIR)/mAConfig.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConfig.cpp
 
 $(WXW_DIR)/mAPreferencesWindow.o: $(WXW_DIR)/mAPreferencesWindow.cpp $(WXW_DIR)/mAPreferencesWindow.h
-	$(CC) -o $(WXW_DIR)/mAPreferencesWindow.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAPreferencesWindow.cpp
+	$(CXX) -o $(WXW_DIR)/mAPreferencesWindow.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAPreferencesWindow.cpp
 
 clean: 
 	rm -rf *.o $(WXW_DIR)/*.o $(BIN_DIR)/miniAudicle
diff -ru miniAudicle-0.1.3.6~/makefile.oss miniAudicle-0.1.3.6/makefile.oss
--- miniAudicle-0.1.3.6~/makefile.oss	2006-09-19 10:09:03.000000000 +0200
+++ miniAudicle-0.1.3.6/makefile.oss	2006-09-19 10:17:57.000000000 +0200
@@ -5,14 +5,17 @@
 
 WXW_DIR=wxw
 BIN_DIR=wxw
+WX_CONFIG?=wx-config
 
-CC=gcc
+CC?=gcc
+CXX?=g++
 LEX=flex
 YACC=bison
 INCLUDES=-I$(CHUCK_SRC_DIR) -I$(WXW_DIR) -I$(PWD)
+CFLAGS?=-O3
 FLAGS=-D__LINUX_OSS___ -D__LINUX__ -D__PLATFORM_LINUX__ -D__MA_WXWIDGETS__ \
-	  `wx-config --cppflags` -O3 -c -DwxABI_VERSION=20600
-LIBS=`wx-config --libs stc,adv,core,base` -lpthread -lstdc++ -ldl -lm
+	  `$(WX_CONFIG) --cppflags` -O3 -c -DwxABI_VERSION=20600 $(CFLAGS)
+LIBS=`$(WX_CONFIG) --libs stc,adv,core,base` -lpthread -lstdc++ -ldl -lm
 SF_OBJ=$(CHUCK_SRC_DIR)/util_sndfile.o
 
 ifneq ($(CHUCK_STAT),)
@@ -73,44 +76,44 @@
 	 $(WXW_DIR)/icons/replace.xpm $(WXW_DIR)/icons/miniAudicle.xpm
 
 miniAudicle: $(OBJS) $(CHUCK_OBJS)
-	g++ -o $(BIN_DIR)/miniAudicle $(OBJS) $(CHUCK_OBJS) $(LIBS)
+	$(CXX) -o $(BIN_DIR)/miniAudicle $(OBJS) $(CHUCK_OBJS) $(LIBS)
 
 # build the core chuck code
 $(CHUCK_OBJS):
 	make -C $(CHUCK_SRC_DIR) linux-oss
 
 miniAudicle_shell.o: miniAudicle_shell.cpp miniAudicle_shell.h
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle_shell.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle_shell.cpp
 
 miniAudicle_import.o: miniAudicle_import.h miniAudicle_import.cpp
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle_import.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle_import.cpp
 
 miniAudicle.o: miniAudicle.cpp miniAudicle.h
-	$(CC) $(FLAGS) $(INCLUDES) miniAudicle.cpp
+	$(CXX) $(FLAGS) $(INCLUDES) miniAudicle.cpp
 
 $(WXW_DIR)/mAParentFrame.o: $(WXW_DIR)/mAParentFrame.cpp $(WXW_DIR)/mAParentFrame.h $(WXW_DIR)/mAView.h $(WXW_DIR)/mADocument.h $(WXW_DIR)/mAVMMonitor.h miniAudicle.h $(WXW_DIR)/mAMenuBar.h $(XPMS)
-	$(CC) -o $(WXW_DIR)/mAParentFrame.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAParentFrame.cpp
+	$(CXX) -o $(WXW_DIR)/mAParentFrame.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAParentFrame.cpp
 
 $(WXW_DIR)/mADocument.o: $(WXW_DIR)/mADocument.cpp $(WXW_DIR)/mADocument.h $(WXW_DIR)/mAView.h
-	$(CC) -o $(WXW_DIR)/mADocument.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mADocument.cpp
+	$(CXX) -o $(WXW_DIR)/mADocument.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mADocument.cpp
 
 $(WXW_DIR)/mAView.o: $(WXW_DIR)/mAView.cpp $(WXW_DIR)/mAView.h
-	$(CC) -o $(WXW_DIR)/mAView.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAView.cpp
+	$(CXX) -o $(WXW_DIR)/mAView.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAView.cpp
 
 $(WXW_DIR)/mAVMMonitor.o: $(WXW_DIR)/mAVMMonitor.cpp $(WXW_DIR)/mAVMMonitor.h $(WXW_DIR)/mAParentFrame.h miniAudicle.h
-	$(CC) -o $(WXW_DIR)/mAVMMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAVMMonitor.cpp
+	$(CXX) -o $(WXW_DIR)/mAVMMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAVMMonitor.cpp
 
 $(WXW_DIR)/mAConsoleMonitor.o: $(WXW_DIR)/mAConsoleMonitor.cpp $(WXW_DIR)/mAConsoleMonitor.h $(WXW_DIR)/mAParentFrame.h
-	$(CC) -o $(WXW_DIR)/mAConsoleMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConsoleMonitor.cpp
+	$(CXX) -o $(WXW_DIR)/mAConsoleMonitor.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConsoleMonitor.cpp
 
 $(WXW_DIR)/mAMenuBar.o: $(WXW_DIR)/mAMenuBar.cpp $(WXW_DIR)/mAMenuBar.h $(WXW_DIR)/mAEvents.h
-	$(CC) -o $(WXW_DIR)/mAMenuBar.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAMenuBar.cpp
+	$(CXX) -o $(WXW_DIR)/mAMenuBar.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAMenuBar.cpp
 
 $(WXW_DIR)/mAConfig.o: $(WXW_DIR)/mAConfig.cpp $(WXW_DIR)/mAConfig.h
-	$(CC) -o $(WXW_DIR)/mAConfig.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConfig.cpp
+	$(CXX) -o $(WXW_DIR)/mAConfig.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAConfig.cpp
 
 $(WXW_DIR)/mAPreferencesWindow.o: $(WXW_DIR)/mAPreferencesWindow.cpp $(WXW_DIR)/mAPreferencesWindow.h
-	$(CC) -o $(WXW_DIR)/mAPreferencesWindow.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAPreferencesWindow.cpp
+	$(CXX) -o $(WXW_DIR)/mAPreferencesWindow.o $(FLAGS) $(INCLUDES) $(WXW_DIR)/mAPreferencesWindow.cpp
 
 clean: 
 	rm -rf *.o $(WXW_DIR)/*.o $(BIN_DIR)/miniAudicle
