diff --git a/configure.ac b/configure.ac
index 769470c..4455c70 100644
--- a/configure.ac
+++ b/configure.ac
@@ -18,7 +18,7 @@ B64WCHARS="-_."
 AC_ARG_WITH([b64wchars], AC_HELP_STRING([--with-b64wchars=3CHARS],[change b64 web extra chars, default is '-_.']),
           [B64WCHARS=$withval], [])
 
-EXTRACFLAGS="-Isrc -Wall -Werror"
+EXTRACFLAGS="-Isrc -Wall"
 
 if test "x$enable_gcov" = "xyes";
 then
