--- s_server.c.orig	2000-10-05 22:22:52.000000000 +0400
+++ s_server.c	2003-11-23 06:34:04.000000000 +0300
@@ -464,7 +467,7 @@
 
 	if (bugs) SSL_CTX_set_options(ctx,SSL_OP_ALL);
 	if (hack) SSL_CTX_set_options(ctx,SSL_OP_NETSCAPE_DEMO_CIPHER_CHANGE_BUG);
-	if (hack) SSL_CTX_set_options(ctx,SSL_OP_NON_EXPORT_FIRST);
+	if (hack) SSL_CTX_set_options(ctx,SSL_OP_CIPHER_SERVER_PREFERENCE);
 
 	if (state) SSL_CTX_set_info_callback(ctx,apps_ssl_info_callback);
 
