Index: Makefile.in
===================================================================
--- Makefile.in	(revision 2879)
+++ Makefile.in	(working copy)
@@ -318,8 +318,8 @@
 install: all
 	$(INSTALL) -d $(DESTDIR)$(sbindir)
 	$(INSTALL) -d $(DESTDIR)$(configdir)
-	$(INSTALL) -d $(DESTDIR)$(piddir)
-	$(INSTALL) -d $(DESTDIR)$(dbdir)
+#	$(INSTALL) -d $(DESTDIR)$(piddir)
+#	$(INSTALL) -d $(DESTDIR)$(dbdir)
 	$(INSTALL) -d $(DESTDIR)$(mandir)
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man8
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man5
