--- Makefile.in	2008-08-22 06:55:32.000000000 +0200
+++ Makefile.in.new	2009-01-22 10:38:23.000000000 +0100
@@ -44,7 +44,7 @@
 linux26_procstat.cpp linux26_rendercpupagestat.cpp linux26_netstat.cpp \
 cygwin_procstat.cpp cygwin_rendercpupagestat.cpp \
 Makefile
-	$(CXX) $(CFLAGS) $(LDFLAGS) procinfo.cpp -o $@
+	$(CXX) $(CFLAGS) $(LDFLAGS) procinfo.cpp $(LIBS) -o $@
 
 #procinfo.o: procinfo.cpp procinfo.h
 #	$(XX) $(CFLAGS) procinfo.cpp -o procinfo.o
