--- SConstruct	2009-10-17 15:22:09.000000000 +0200
+++ SConstruct.new	2009-10-17 15:23:31.000000000 +0200
@@ -1240,8 +1240,6 @@
     fullInstallName = installDir + "/bin/" + name
 
     allBinaries += [ name ]
-    if solaris or linux:
-        e.AddPostAction( inst, e.Action( 'strip ' + fullInstallName ) )
 
     if linux and len( COMMAND_LINE_TARGETS ) == 1 and str( COMMAND_LINE_TARGETS[0] ) == "s3dist":
         e.AddPostAction( inst , checkGlibc )
