--- SConstruct.orig	2009-07-14 12:10:40.000000000 +0200
+++ SConstruct	2009-07-14 12:11:31.000000000 +0200
@@ -1134,8 +1134,6 @@
     fullInstallName = installDir + "/bin/" + name
 
     allBinaries += [ name ]
-    if solaris or linux:
-        e.AddPostAction( inst, e.Action( 'strip ' + fullInstallName ) )
         
     if linux and str( COMMAND_LINE_TARGETS[0] ) == "s3dist":
         e.AddPostAction( inst , checkGlibc )

