diff -u SConstruct.old SConstruct
--- SConstruct.old	2009-06-10 12:47:25.000000000 +0200
+++ SConstruct	2009-06-10 12:47:42.000000000 +0200
@@ -811,9 +811,11 @@
             Exit( 1 )
 
 def ensureTestDirs():
-    ensureDir( "/tmp/unittest/" )
-    ensureDir( "/data/" )
-    ensureDir( "/data/db/" )
+    ensureDir( "testdir/" )
+    ensureDir( "testdir/tmp/" )
+    ensureDir( "testdir/tmp/unittest/" )
+    ensureDir( "testdir/data/" )
+    ensureDir( "testdir/data/db/" )
 
 def testSetup( env , target , source ):
     ensureTestDirs()
@@ -890,7 +892,7 @@
     mongodForTestsPort = "40000"
     import os
     ensureTestDirs()
-    dirName = "/data/db/sconsTests/"
+    dirName = "testdir/data/db/sconsTests/"
     ensureDir( dirName )
     from subprocess import Popen
     mongodForTests = Popen( [ mongod[0].abspath, "--port", mongodForTestsPort, "--dbpath", dirName ] )
diff -u dbtests/dbtests.cpp.old dbtests/dbtests.cpp
--- dbtests/dbtests.cpp.old	2009-06-10 13:10:43.000000000 +0200
+++ dbtests/dbtests.cpp	2009-06-09 15:37:04.000000000 +0200
@@ -35,7 +35,7 @@
 namespace mongo {
     extern const char* dbpath;
 } // namespace mongo
-string dbpathSpec = "/tmp/unittest/";
+string dbpathSpec = "testdir/tmp/unittest/";
 
 Suite::~Suite() {
     DBDirectClient c;

