--- SConstruct.old	2009-10-17 15:26:14.000000000 +0200
+++ SConstruct	2009-10-17 15:29:52.000000000 +0200
@@ -995,15 +995,17 @@
             Exit( 1 )
 
 def ensureTestDirs():
-    ensureDir( "/tmp/unittest/" )
-    ensureDir( "/data/" )
-    ensureDir( "/data/db/" )
+    ensureDir( "testdir/" )
+    ensureDir( "testdir/tmp/" )
+    ensureDir( "testdir/tmp/unittest/" )
+    ensureDir( "testdir/data/" )
+    ensureDir( "testdir/data/db/" )
 
 def testSetup( env , target , source ):
     ensureTestDirs()
 
 if len( COMMAND_LINE_TARGETS ) == 1 and str( COMMAND_LINE_TARGETS[0] ) == "test":
-    ensureDir( "/tmp/unittest/" );
+    ensureDir( "testdir/tmp/unittest/" );
 
 addSmoketest( "smoke", [ add_exe( "test" ) ] , [ test[ 0 ].abspath ] )
 addSmoketest( "smokePerf", [ "perftest" ] , [ perftest[ 0 ].abspath ] )
@@ -1072,7 +1074,7 @@
     mongodForTestsPort = "40000"
     import os
     ensureTestDirs()
-    dirName = "/data/db/sconsTests/"
+    dirName = "testdir/data/db/sconsTests/"
     ensureDir( dirName )
     from subprocess import Popen
     mongodForTests = Popen( [ mongod[0].abspath, "--port", mongodForTestsPort, "--dbpath", dirName ] )

--- dbtests/dbtests.cpp	2009-10-17 16:40:46.000000000 +0200
+++ dbtests/dbtests.cpp.new	2009-10-17 16:40:54.000000000 +0200
@@ -22,6 +22,6 @@
 #include "dbtests.h"
 
 int main( int argc, char** argv ) {
-    return Suite::run(argc, argv, "/tmp/unittest");
+    return Suite::run(argc, argv, "testdir/tmp/unittest");
 }

